<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Graham Coop" />
  <title>Population andQuantitative Genetics</title>
  <style>
    html {
      line-height: 1.5;
      font-family: Georgia, serif;
      font-size: 20px;
      color: #1a1a1a;
      background-color: #fdfdfd;
    }
    body {
      margin: 0 auto;
      max-width: 36em;
      padding-left: 50px;
      padding-right: 50px;
      padding-top: 50px;
      padding-bottom: 50px;
      hyphens: auto;
      overflow-wrap: break-word;
      text-rendering: optimizeLegibility;
      font-kerning: normal;
    }
    @media (max-width: 600px) {
      body {
        font-size: 0.9em;
        padding: 1em;
      }
      h1 {
        font-size: 1.8em;
      }
    }
    @media print {
      body {
        background-color: transparent;
        color: black;
        font-size: 12pt;
      }
      p, h2, h3 {
        orphans: 3;
        widows: 3;
      }
      h2, h3, h4 {
        page-break-after: avoid;
      }
    }
    p {
      margin: 1em 0;
    }
    a {
      color: #1a1a1a;
    }
    a:visited {
      color: #1a1a1a;
    }
    img {
      max-width: 100%;
    }
    h1, h2, h3, h4, h5, h6 {
      margin-top: 1.4em;
    }
    h5, h6 {
      font-size: 1em;
      font-style: italic;
    }
    h6 {
      font-weight: normal;
    }
    ol, ul {
      padding-left: 1.7em;
      margin-top: 1em;
    }
    li > ol, li > ul {
      margin-top: 0;
    }
    blockquote {
      margin: 1em 0 1em 1.7em;
      padding-left: 1em;
      border-left: 2px solid #e6e6e6;
      color: #606060;
    }
    code {
      font-family: Menlo, Monaco, 'Lucida Console', Consolas, monospace;
      font-size: 85%;
      margin: 0;
    }
    pre {
      margin: 1em 0;
      overflow: auto;
    }
    pre code {
      padding: 0;
      overflow: visible;
      overflow-wrap: normal;
    }
    .sourceCode {
     background-color: transparent;
     overflow: visible;
    }
    hr {
      background-color: #1a1a1a;
      border: none;
      height: 1px;
      margin: 1em 0;
    }
    table {
      margin: 1em 0;
      border-collapse: collapse;
      width: 100%;
      overflow-x: auto;
      display: block;
      font-variant-numeric: lining-nums tabular-nums;
    }
    table caption {
      margin-bottom: 0.75em;
    }
    tbody {
      margin-top: 0.5em;
      border-top: 1px solid #1a1a1a;
      border-bottom: 1px solid #1a1a1a;
    }
    th {
      border-top: 1px solid #1a1a1a;
      padding: 0.25em 0.5em 0.25em 0.5em;
    }
    td {
      padding: 0.125em 0.5em 0.25em 0.5em;
    }
    header {
      margin-bottom: 4em;
      text-align: center;
    }
    #TOC li {
      list-style: none;
    }
    #TOC ul {
      padding-left: 1.3em;
    }
    #TOC > ul {
      padding-left: 0;
    }
    #TOC a:not(:hover) {
      text-decoration: none;
    }
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
  </style>
  <script
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js"
  type="text/javascript"></script>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Population andQuantitative Genetics</h1>
<p class="author">Graham Coop</p>
</header>
<p>[chapter]</p>
<h1 id="introduction">Introduction</h1>
<p><span class="citation"
data-cites="DobzhanskyBook">[@DobzhanskyBook]</span> Our population is
made up of a set of interbreeding individuals, the genetic composition
of which is made up of the genomes that each individual carries. The
genetic composition of the population alters due to the death of
individuals or the migration of individuals in or out of the population.
If our individuals vary in the number of children they have, this also
alters the genetic composition of the population in the next generation.
Every new individual born into the population subtly changes the genetic
composition of the population. Their genome is a unique combination of
their parents’ genomes, having been shuffled by segregation and
recombination during meioses, and possibly changed by mutation. These
individual events seem minor at the level of the population, but it is
the accumulation of small changes in aggregate across individuals and
generations that is the stuff of evolution. It is the compounding of
these small changes over tens, hundreds, and millions of generations
that drives the amazing diversity of life that has emerged on this
earth.<br />
Population genetics is the study of the genetic composition of natural
populations and its evolutionary causes and consequences. Quantitative
genetics is the study of the genetic basis of phenotypic variation and
how phenotypic changes evolve over time. Both fields are closely
conceptually aligned as we’ll see throughout these notes. They seek to
describe how the genetic and phenotypic composition of populations can
be changed over time by the forces of mutation, recombination,
selection, migration, and genetic drift. To understand how these forces
interact, it is helpful to develop simple theoretical models to help our
intuition. In these notes we will work through these models and
summarize the major areas of population- and quantitative-genetic
theory.<br />
</p>
<p>While the models we will develop will seem naı̈ve, and indeed they
are, they are nonetheless incredibly useful and powerful. Throughout the
course we will see that these simple models often yield accurate
predictions, such that much of our understanding of the process of
evolution is built on these models. We will also see how these models
are incredibly useful for understanding real patterns we see in the
evolution of phenotypes and genomes, such that much of our analysis of
evolution, in a range of areas from human medical genetics to
conservation, is based on these models. Therefore, population and
quantitative genetics are key to understanding various applied
questions, from how medical genetics identifies the genes involved in
disease to how we preserve species from extinction.<br />
Population genetics emerged from early efforts to reconcile Mendelian
genetics with Darwinian thought. Part of the power of population
genetics comes from the fact that the basic rules of transmission
genetics are simple and nearly universal. One of the truly remarkable
things about population genetics is that many of the important ideas and
mathematical models emerged before the 1940s, long before the
mechanistic-basis of inheritance (DNA) was discovered, and yet the
usefulness of these models has not diminished. This is a testament to
the fact that the models are established on a very solid foundation,
building from the basic rules of genetic transmission combined with
simple mathematical and statistical models.</p>
<p>Much of this early work traces to the ideas of R.A. Fisher, Sewall
Wright, and J.B.S. Haldane, who, along with many others, described the
early principals and mathematical models underlying our understanding of
the evolution of populations. Building on this conceptual fusion of
genetics and evolution, there followed a flourishing of evolutionary
thought, the modern evolutionary synthesis, combining these ideas with
those from the study of speciation, biodiversity, and paleontology. In
total, this work showed that both short-term evolutionary change and the
long-term evolution of biodiversity could be well understood through the
gradual accumulation of evolutionary change within and among
populations. This evolutionary synthesis continues to this day,
combining new insights from genomics, phylogenetics, ecology, and
developmental biology.</p>
<p>Population and quantitative genetics are a necessary but not
sufficient description of evolution; it is only by combining the
insights of many fields that a rich and comprehensive picture of
evolution emerges. We certainly do not need to know the genes underlying
the displays of the birds of paradise to study how the divergence of
these displays, due to sexual selection, may drive speciation. Indeed,
as we’ll see in our discussion of quantitative genetics, we can predict
how populations respond to selection, including sexual selection and
assortative mating, without any knowledge of the loci involved. Nor do
we need to know the precise selection pressures and the ordering of
genetic changes to study the emergence of the tetrapod body plan. We do
not necessarily need to know all the genetic details to appreciate the
beauty of these, and many other, evolutionary case studies. However,
every student of biology gains from understanding the basics of
population and quantitative genetics, allowing them to base their
studies on a solid bedrock of understanding of the processes that
underpin all evolutionary change.</p>
<h2 class="unnumbered"
id="genetics-eugenics-and-scientific-racism.">Genetics, Eugenics, and
Scientific Racism.</h2>
<p>The history of genetics and evolutionary biology is intertwined with
the history of eugenics and scientific racism. Francis Galton, one of
the first people to systematically study human inheritance, coined the
term “eugenics” in 1883 to describe the idea of ‘human improvement’
through controlled breeding of humans <span class="citation"
data-cites="galton1883inquiries">[@galton1883inquiries]</span>.
Historically, eugenics is much more than just the idea that selection
through breeding would work in humans; it is the idea that particular
people are “genetically inferior” and therefore “unfit” to reproduce
<span class="citation"
data-cites="paul2014wrong">[@paul2014wrong]</span>. Eugenicists’
obsession with human worth and genetic inferiority also meant that
eugenicists also often held that people from some races and ethnicities
are genetically superior to others. Thus, ideas about eugenics also
built on older racist fields of science that sought to classify humans
into a discrete racial hierarchy, while in parallel scientists in these
fields were forcing ideas from genetics and evolution into an
essentialist view of race. These deeply flawed hierarchies have
frequently been used by the powerful to justify subjugating and
disenfranchising minorities and Indigenous people.</p>
<p>Although eugenics is often correctly associated with the Nazi party
and the Holocaust, eugenic ideas and eugenic policies were also
widespread in the US and UK during the 1920s and 1930s and sometimes
aligned with progressive causes of that era <span class="citation"
data-cites="paul1984eugenics kevles1995name">[@paul1984eugenics;
@kevles1995name]</span>. Eugenic ideas were also implemented as policy
—with horrific consequences—in a number of countries. Immigration
policies based explicitly on eugenic arguments were put in place in the
US from the 1920s until their repeal in the 1950s and 60s. These
policies strongly favoured immigration from Northern Europe and were a
deliberate action to restrict or bar immigration from Asia and eastern
and southern Europe based on xenophobic, racist, and anti-Semitic views
<span class="citation"
data-cites="okrent2020guarded">[@okrent2020guarded]</span>. During the
20th Century, many US states passed eugenics sterilization laws <span
class="citation"
data-cites="reilly2015eugenics">[@reilly2015eugenics]</span>, that in
practice were often targeted against Black, Latino, and Indigenous
people <span class="citation"
data-cites="hansen2013sterilized">[@hansen2013sterilized]</span>. For
example, the state of California from 1919 to 1972 used eugenics ideas
to justify the sterilization of 20,000 people who had been labelled
unfit and mentally defective, a disproportionate number of whom were
Latino <span class="citation"
data-cites="stern2017california novak2018disproportionate">[@stern2017california;
@novak2018disproportionate]</span></p>
<p>Many early geneticists during this time were proponents of eugenics
and many supported racist views in their genetics research. One notable
example is R.A. Fisher, who we’ll encounter throughout this book. Fisher
is arguably the father of much of evolutionary genetics and modern
statistics, having made huge contributions to the foundations of both
fields. He pursued these fields in part because of his eugenic interests
and concerns about the “genetically inferiority” of the lower classes
<span class="citation"
data-cites="norton1983fisher mazumdar2005eugenics_fisher">[@norton1983fisher;
@mazumdar2005eugenics_fisher]</span>. For example, he devoted a number
of the later chapters in his classic evolutionary genetics book to
eugenics <span class="citation"
data-cites="fisher1930">[@fisher1930]</span>. He was hardly alone in his
views, with many prominent geneticists lending their voices to eugenic
and racist arguments. Indeed, many famous genetics institutions grew
from roots in eugenics. For example, the Cold Spring Harbor Laboratory
hosted a large Eugenics Record Office, and prior to 1954, the journal
Annals of Human Genetics was called Annals of Eugenics. Scientists and
their institutions strongly shaped the eugenic views and policies of
their time and at times bent science to lend support to their racist
views. Given their lasting contributions to our field, we should not shy
away from reading and discussing their work. But despite their
scientific accomplishments, we should resist the urge to celebrate or
idolize them. We should also guard against inheriting their thinking by
continually questioning the frameworks and language they put in
place.</p>
<p>From its inception, geneticists have also been central to movements
against eugenics and scientific racism on scientific as well as moral
grounds. For instance, Thomas Hunt Morgan and Lancelot Hogben were both
prominent geneticists who argued that eugenicists failed to recognize
the environmental and social causes of inequality <span class="citation"
data-cites="hogben1933nature tabery2008ra allen2011eugenics">[@hogben1933nature;
@tabery2008ra; @allen2011eugenics]</span>. These arguments thread into
later debates, where geneticists pushed back on simplistic and erroneous
claims about genetics, IQ and behavioural differences among human
populations <span class="citation"
data-cites="dobzhansky1961bogus lewontin1970race paul1994dobzhansky">[@dobzhansky1961bogus;
@lewontin1970race; @paul1994dobzhansky]</span>. Population geneticists
have also been central to the pushback against scientific racism,
highlighting the close genetic relationships among all humans due to
their recent common ancestry and the ephemeral nature of populations
<span class="citation"
data-cites="united1952race lewontin1972apportionment provine1986geneticists gannett2013theodosius">[@united1952race;
@lewontin1972apportionment; @provine1986geneticists;
@gannett2013theodosius]</span>. Racists continue to advance a selective
view of population-genetic results to further their ends. As scientists,
it is too easy to claim that we are just interested in the facts and
ignore others who seek to present a distorted view of the science to
advance their own political and social agendas. It is our job as
population geneticists to argue against misuse of our field. As human
genomics and personal genomics rise in prominence, we also need to
resist public adoption of genetic determinism and essentialist,
racialized thinking. We must question the topics we choose to
investigate, the assumptions we make, and the conversations we
prioritize as a field. Through exploring our own biases and those
embedded in the presentation and use of our field, we can help to combat
the misrepresentations of genetics and evolution that continue to cause
harm in our society.</p>
<h1 id="allele-and-genotype-frequencies.">Allele and Genotype
Frequencies.</h1>
<p>In this chapter we will work through how the basics of Mendelian
genetics play out at the population level in sexually reproducing
organisms.</p>
<p>Loci and alleles are the basic currency of population genetics–and
indeed of genetics. A locus may be an entire gene, or a single
nucleotide base pair such as A-T. At each locus, there may be multiple
genetic variants segregating in the population—these different genetic
variants are known as <em>alleles</em>. If all individuals in the
population carry the same allele, we say that the locus is
<em>monomorphic</em>; at this locus there is no genetic variability in
the population. If there are multiple alleles in the population at a
locus, we say that this locus is <em>polymorphic</em> (this is sometimes
referred to as a segregating site).</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/Drosophila_mel/Drosophila_mel_mutants.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Table <a href="#Table:ADH" data-reference-type="ref"
data-reference="Table:ADH">[Table:ADH]</a> shows a small stretch of
orthologous sequence for the ADH locus from samples from <em>Drosophila
melanogaster</em>, <em>D. simulans</em>, and <em>D. yakuba</em>. <em>D.
melanogaster</em> and <em>D. simulans</em> are sister species and <em>D.
yakuba</em> is a close outgroup to the two. Each column represents a
single haplotype from an individual (the individuals are diploid but
were inbred so they’re homozygous for their haplotype). Only sites that
differ among individuals of the three species are shown. Site <span
class="math inline">\(834\)</span> is an example of a polymorphism; some
<em>D. simulans</em> individuals carry a <span
class="math inline">\(C\)</span> allele while others have a <span
class="math inline">\(T\)</span>. <em>Fixed differences</em> are sites
that differ between the species but are monomorphic within the species.
Site <span class="math inline">\(781\)</span> is an example of a fixed
difference between <em>D. melanogaster</em> and the other two
species.</p>
<p>We can also annotate the alleles and loci in various ways. For
example, position <span class="math inline">\(781\)</span> is a
non-synonymous fixed difference. We call the less common allele at a
polymorphism the <em>minor allele</em> and the common allele the
<em>major allele</em>, e.g. at site <span
class="math inline">\(1068\)</span> the <span
class="math inline">\(T\)</span> allele is the minor allele in <em>D.
melanogaster</em>. We call the more evolutionarily recent of the two
alleles the <em>derived allele</em> and the older of the two the
<em>ancestral allele</em>. We infer that the <span
class="math inline">\(T\)</span> allele at site 1068 is the derived
allele because the <span class="math inline">\(C\)</span> is found in
both other species, suggesting that the <span
class="math inline">\(T\)</span> allele arose via a <span
class="math inline">\(C \rightarrow T\)</span> mutation.</p>
<div class="table*">
<div class="center">

</div>
</div>
<p><strong>A)</strong> How many segregating sites does the sample from
<em>D. simulans</em> have in the ADH gene?<br />
<strong>B)</strong> How many fixed differences are there between <em>D.
melanogaster</em> and <em>D. yakuba</em>?</p>
<h2 id="allele-frequencies">Allele frequencies</h2>
<p>Allele frequencies are a central unit of population genetics
analysis, but from diploid individuals we only get to observe genotype
counts. Our first task then is to calculate allele frequencies from
genotype counts. Consider a diploid autosomal locus segregating for two
alleles (<span class="math inline">\(A_1\)</span> and <span
class="math inline">\(A_2\)</span>). We’ll use these arbitrary labels
for our alleles, merely to keep this general. Let <span
class="math inline">\(N_{11}\)</span> and <span
class="math inline">\(N_{12}\)</span> be the number of <span
class="math inline">\(A_1A_1\)</span> homozygotes and <span
class="math inline">\(A_1A_2\)</span> heterozygotes, respectively.
Moreover, let <span class="math inline">\(N\)</span> be the total number
of diploid individuals in the population. We can then define the
relative frequencies of <span class="math inline">\(A_1A_1\)</span> and
<span class="math inline">\(A_1A_2\)</span> genotypes as <span
class="math inline">\(f_{11} = N_{11}/N\)</span> and <span
class="math inline">\(f_{12} = N_{12}/N\)</span>, respectively. The
frequency of allele <span class="math inline">\(A_1\)</span> in the
population is then given by</p>
<p><span class="math display">\[p = \frac{2 N_{11} + N_{12}}{2N} =
f_{11} + \frac{1}{2} f_{12}.\]</span> Note that this follows directly
from how we count alleles given individuals’ genotypes, and holds
independently of Hardy–Weinberg proportions and equilibrium (discussed
below). The frequency of the alternate allele (<span
class="math inline">\(A_2\)</span>) is then just <span
class="math inline">\(q=1-p\)</span>.</p>
<h3 id="measures-of-genetic-variability">Measures of genetic
variability</h3>
<h5 id="nucleotide-diversity-pi">Nucleotide diversity (<span
class="math inline">\(\pi\)</span>)</h5>
<p>One common measure of genetic diversity is the average number of
single nucleotide differences between haplotypes chosen at random from a
sample. This is called <em>nucleotide diversity</em> and is often
denoted by <span class="math inline">\(\pi\)</span>.</p>
<p>For example, we can calculate <span
class="math inline">\(\pi\)</span> for our ADH locus from Table <a
href="#Table:ADH" data-reference-type="ref"
data-reference="Table:ADH">[Table:ADH]</a> above: we have 6 sequences
from <em>D. simulans</em> (a-f), there’s a total of 15 ways of pairing
these sequences, and</p>
<p><span class="math display">\[\pi=\frac{1}{15} \big( (2 + 1 + 1 + 1 +
0 ) + (3 + 3 + 3 + 2 ) +(0 + 0 + 1) + (0 + 1) +
(1)  \big)=1.2\overline{6}\]</span></p>
<p>where the first bracketed term gives the pairwise differences between
a and b-f, the second bracketed term the differences between b and c-f
and so on.<br />
Our <span class="math inline">\(\pi\)</span> measure will depend on the
length of sequence it is calculated for. Therefore, <span
class="math inline">\(\pi\)</span> is usually normalized by the length
of sequence, to be a per site (or per base) measure. For example, our
ADH sequence covers <span class="math inline">\(397\)</span>bp of DNA
and so <span class="math inline">\(\pi =
1.2\overline{6}/397=0.0032\)</span> per site in <em>D. simulans</em> for
this region. Note that we could also calculate <span
class="math inline">\(\pi\)</span> per synonymous site (or
non-synonymous). For synonymous site <span
class="math inline">\(\pi\)</span>, we would count up number of
synonymous differences between our pairs of sequences, and then divide
by the total number of sites where a synonymous change could have
occurred.</p>
<h5 id="number-of-segregating-sites.">Number of segregating sites.</h5>
<p>Another measure of genetic variability is the total number of sites
that are polymorphic (segregating) in our sample. One issue is that the
number of segregating sites will grow as we sequence more individuals
(unlike <span class="math inline">\(\pi\)</span>). Later in the course,
we’ll talk about how to standardize the number of segregating sites for
the number of individuals sequenced (see eqn<a href="#watterson_theta"
data-reference-type="eqref"
data-reference="watterson_theta">[watterson_theta]</a>).</p>
<h5 id="the-frequency-spectrum.">The frequency spectrum.</h5>
<p>We also often want to compile information about the frequency of
alleles across sites. We call alleles that are found once in a sample
<em>singletons</em>, alleles that are found twice in a sample
<em>doubletons</em>, and so on. We count up the number of loci where an
allele is found <span class="math inline">\(i\)</span> times out of
<span class="math inline">\(n\)</span>, e.g. how many singletons are
there in the sample, and this is called the <em>frequency spectrum</em>.
We’ll want to do this in some consistent manner, such as calculating the
frequency spectrum of the minor allele or the derived allele.</p>
<p>How many minor-allele singletons are there in <em>D. simulans</em> in
the ADH region? [Defining minor allele just within <em>D.
simulans</em>.]</p>
<h5 id="levels-of-genetic-variability-across-species.">Levels of genetic
variability across species.</h5>
<p>Two observations have puzzled population geneticists since the
inception of molecular population genetics. The first is the relatively
high level of genetic variation observed in most obligately sexual
species. This first observation, in part, drove the development of the
Neutral theory of molecular evolution, the idea that much of this
molecular polymorphism may simply reflect a balance between genetic
drift and mutation. The second observation is the relatively narrow
range of polymorphism across species with vastly different census sizes.
This observation represented a puzzle as the Neutral theory predicts
that levels of genetic diversity should scale with population size. Much
effort in theoretical and empirical population genetics has been devoted
to trying to reconcile models with these various observations. We’ll
return to discuss these ideas throughout our course.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/Ciona_intestinalis/21016139168_2a8a57ded3_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>The first observations of molecular genetic diversity within natural
populations were made from surveys of allozyme data, but we can revisit
these general patterns with modern data. For example, <span
class="citation" data-cites="leffler:12">@leffler:12</span> compiled
data on levels of within-population, autosomal nucleotide diversity
(<span class="math inline">\(\pi\)</span>) for 167 species across 14
phyla from non-coding and synonymous sites (Figure <a href="#fig:Leffer"
data-reference-type="ref" data-reference="fig:Leffer">[fig:Leffer]</a>).
The species with the lowest levels of <span
class="math inline">\(\pi\)</span> in their survey was Lynx, with <span
class="math inline">\(\pi = 0.01\%\)</span>, i.e. only <span
class="math inline">\(1/10000\)</span> bases differed between two
sequences. In contrast, some of the highest levels of diversity were
found in <em>Ciona savignyi</em>, Sea Squirts, where a remarkable <span
class="math inline">\(1/12\)</span> bases differ between pairs of
sequences. This <span class="math inline">\(800\)</span>-fold range of
diversity seems impressive, but census population sizes have a much
larger range.</p>
<div class="figure*">
<div class="center">
<figure>
<img
src="Journal_figs/alleles_genotypes/Leffer_riddle/Leffer_riddle_diversity.png"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/Lynx/20731949565_8a065700af_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h3 id="hardyweinberg-proportions">Hardy–Weinberg proportions</h3>
<p>Imagine a population mating at random with respect to genotypes,
i.e. no inbreeding, no assortative mating, no population structure, and
no sex differences in allele frequencies. The frequency of allele <span
class="math inline">\(A_1\)</span> in the population at the time of
reproduction is <span class="math inline">\(p\)</span>. An <span
class="math inline">\(A_1A_1\)</span> genotype is made by reaching out
into our population and independently drawing two <span
class="math inline">\(A_1\)</span> allele gametes to form a zygote.
Therefore, the probability that an individual is an <span
class="math inline">\(A_1A_1\)</span> homozygote is <span
class="math inline">\(p^2\)</span>. This probability is also the
expected frequencies of the <span class="math inline">\(A_1A_1\)</span>
homozygote in the population. The expected frequency of the three
possible genotypes are</p>
<div class="center">
<table>
<thead>
<tr class="header">
<th style="text-align: center;"><span
class="math inline">\(f_{11}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(f_{12}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(f_{22}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><span
class="math inline">\(p^2\)</span></td>
<td style="text-align: center;"><span
class="math inline">\(2pq\)</span></td>
<td style="text-align: center;"><span
class="math inline">\(q^2\)</span></td>
</tr>
</tbody>
</table>
</div>
<p>i.e. their Hardy-Weinberg expectations <span class="citation"
data-cites="hardy1908mendelian weinberg1908ber">[@hardy1908mendelian;
@weinberg1908ber]</span>. Note that we only need to assume random mating
with respect to our focal allele in order for these expected frequencies
to hold in the zygotes forming the next generation. Evolutionary forces,
such as selection, change allele frequencies within generations, but do
not change this expectation for new zygotes, as long as <span
class="math inline">\(p\)</span> is the frequency of the <span
class="math inline">\(A_1\)</span> allele in the population at the time
when gametes fuse. We only need the assumptions of no migration,
selection, and mutation in order for these Hardy-Weinberg expectations
of genotypes to represent a long term equilibrium.</p>
<p>On the coastal islands of British Columbia there is a subspecies of
black bear (<em>Ursus americanus kermodei</em>, Kermode’s bear). Many
members of this black bear subspecies are white; they’re sometimes
called spirit bears. These bears aren’t hybrids with polar bears, nor
are they albinos. They are homozygotes for a recessive change at the
MC1R gene. Individuals who are <span class="math inline">\(GG\)</span>
at this SNP are white, while <span class="math inline">\(AA\)</span> and
<span class="math inline">\(AG\)</span> individuals are black.</p>
<p>Below are the genotype counts for the MC1R polymorphism in a sample
of bears from British Columbia’s island populations from <span
class="citation" data-cites="RITLAND:01">@RITLAND:01</span>.</p>
<div class="center">
<table>
<thead>
<tr class="header">
<th style="text-align: center;"><span
class="math inline">\(AA\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(AG\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(GG\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">42</td>
<td style="text-align: center;">24</td>
<td style="text-align: center;">21</td>
</tr>
</tbody>
</table>
</div>
<p>What are the expected frequencies of the three genotypes under
HW?</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/Kermode_bear/EIiK2dsWoAAmUOf.jpeg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>See Figure <a href="#fig:HWE_CEU_YRI" data-reference-type="ref"
data-reference="fig:HWE_CEU_YRI">2.1</a> for a nice empirical
demonstration of Hardy–Weinberg proportions. The mean frequency of each
genotype closely matches its HW expectations, and much of the scatter of
the dots around the expected line is due to our small sample size (<span
class="math inline">\(\sim 60\)</span> individuals). While HW often
seems like a silly model, it often holds remarkably well within
populations. This is because individuals don’t mate at random, but they
do mate at random with respect to their genotype at most of the loci in
the genome.</p>
<p>You are investigating a locus with three alleles, A, B, and C, with
allele frequencies <span class="math inline">\(p_A\)</span>, <span
class="math inline">\(p_B\)</span>, and <span
class="math inline">\(p_C\)</span>. What fraction of the population is
expected to be homozygotes under Hardy–Weinberg?</p>
<p>Microsatellites are regions of the genome where individuals vary for
the number of copies of some short DNA repeat that they carry. These
regions are often highly variable across individuals, making them a
suitable way to identify individuals from a DNA sample. This so-called
DNA fingerprinting has a range of applications from establishing
paternity and identifying human remains to matching individuals to DNA
samples from a crime scene. The FBI make use of the CODIS database. The
CODIS database contains the genotypes of over 13 million people, most of
whom have been convicted of a crime. Most of the profiles record
genotypes at 13 microsatellite loci that are tetranucleotide repeats
(since 2017, 20 sites have been genotyped).</p>
<p>The allele counts for two loci (D16S539 and TH01) are shown in table
<a href="#table:CODIS_1" data-reference-type="ref"
data-reference="table:CODIS_1">[table:CODIS_1]</a> and <a
href="#table:CODIS_2" data-reference-type="ref"
data-reference="table:CODIS_2">[table:CODIS_2]</a> for a sample of 155
people of European ancestry. You can assume these two loci are on
different chromosomes.</p>
<p><span id="table:CODIS_1" label="table:CODIS_1"></span></p>
<p><span id="table:CODIS_2" label="table:CODIS_2"></span></p>
<p><strong>Question</strong> <span id="Q:CODIS" label="Q:CODIS"></span>
You extract a DNA sample from a crime scene. The genotype is 100/80 at
the D16S539 locus and 70/93 at TH01.<br />
<strong>A)</strong> You have a suspect in custody. Assuming this suspect
is innocent and of European ancestry, what is the probability that their
genotype would match this profile by chance (a false-match
probability)?<br />
<strong>B)</strong> The FBI uses <span
class="math inline">\(\geq\)</span> 13 markers. Why is this higher
number necessary to make the match statement convincing evidence in
court?<br />
<strong>C)</strong> An early case that triggered debate among forensic
geneticists was a crime among the Abenaki, a Native American community
in Vermont <span class="citation" data-cites="lewontin:94">[see
@lewontin:94 for discussion]</span>. There was a DNA sample from the
crime scene, and the perpetrator was thought likely to be a member of
the Abenaki community. Given that allele frequencies vary among
populations, why would people be concerned about using data from a
non-Abenaki population to compute a false match probability?</p>
<div class="center">
<figure>
<img src="figures/CEU_YRI_separately_HWE.png" id="fig:HWE_CEU_YRI"
alt="Demonstrating Hardy–Weinberg proportions using 10,000 SNPs from the HapMap European (CEU) and African (YRI) populations. Within each of these populations the allele frequency against the frequency of the 3 genotypes; each SNP is represented by 3 different coloured points. The solid lines show the mean genotype frequency. The dashed lines show the predicted genotype frequency from Hardy–Weinberg equilibrium. Blog post on figure here." />
<figcaption aria-hidden="true">Demonstrating Hardy–Weinberg proportions
using 10,000 SNPs from the HapMap European (CEU) and African (YRI)
populations. Within each of these populations the allele frequency
against the frequency of the 3 genotypes; each SNP is represented by 3
different coloured points. The solid lines show the mean genotype
frequency. The dashed lines show the predicted genotype frequency from
Hardy–Weinberg equilibrium. Blog post on figure <a
href="http://gcbias.org/2011/10/13/population-genetics-course-resources-Hardy--Weinberg-eq/">here</a>.</figcaption>
</figure>
</div>
<h3 id="assortative-mating">Assortative mating</h3>
<p>One major violation of the assumptions of Hardy Weinberg is
non-random mating with respect to the genotype at a locus. One way that
individuals can mate non-randomly is if individuals choose to mate based
on a phenotype determined by (in part) the genotype at a locus. This
non-random mating can be between: 1) individuals with similar phenotype,
so called positive assortative mating or 2) individuals with dissimilar
phenotypes, negative assortative mating or disassortative mating. Here
we’ll briefly discuss a couple of real examples of assortative mating to
make sure we’re all on the same page. We’ll encounter other forms of
non-random mating, due to inbreeding and population structure, in the
next few chapters.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/alleles_genotypes/Heliconius_Merrill_assort_mating/labeled_trimmed_Heliconius_Merrill_assort_mating.pdf"
id="fig:Heliconius_Merrill_assort" />
<figcaption aria-hidden="true">Wing pattern phenotypes of top, <em>H.
cydno chioneus</em> (left), <em>H. melpomene rosina</em> (right), their
nonmimetic first-generation hybrid (center); and bottom, their sympatric
comimics H. sapho sapho (left) and H. erato demophoon (right). Figure
and caption modified from <span class="citation"
data-cites="merrill2019genetic">@merrill2019genetic</span>,
.</figcaption>
</figure>
</div>
<p>Positive assortative mating on the basis of a phenotype can create an
excess of homozygotes. Heliconius butterflies are famous for their
mimicry, where poisonous pairs of distantly related species mimic each
others’ bright colour patterns and so share the benefits of being
avoided by visual predators (Müllerian mimics). <em>H. melpomene
rosina</em> and <em>H. cydno chioneus</em> are closely related species
that co-occur in central Panama, but mimic different other co-occuring
species (Figure <a href="#fig:Heliconius_Merrill_assort"
data-reference-type="ref"
data-reference="fig:Heliconius_Merrill_assort">2.2</a> ). These
differences in colouration pattern are due to a few loci with large
phenotypic effects. The two species can hybridize and produce viable F1
hybrids. These F1 hybrids are heterozygotes at the colour loci, and
their intermediate appearance means that they’re poor mimics and so are
quickly eaten by predators. However, these heterzygote (F1) hybrids are
very rare in nature <span class="math inline">\(&lt;
\frac{1}{1000}\)</span>, as the parental species show strong positive
assortatively mating based on colour pattern, based on genetic
differences in mate preference <span class="citation"
data-cites="merrill2019genetic">@merrill2019genetic</span>.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/White_throated_sparrow/white_throated_sparrow.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Disassortative mating, mating of unlike individuals, can lead to an
excess of heterozygotes and a deficit of homozygotes. One example of
very strong disassortative mating is offered by white-throated sparrows
(<em>Zonotrichia albicollis</em>). In white-throated sparrows, there is
a white-striped and a tan-striped morph, with female and male
white-striped morphs have a much brighter white stripe and throat. There
is very strong disassortative mating in this system, with 1099 out of
1116 nesting pairs consisting of one tan- and one white-striped morph
and only 17 of these nesting pairs being different morphs <span
class="citation"
data-cites="tuttle2016divergence">[@tuttle2016divergence]</span>. The
difference between these morphs has a simple inheritance pattern, with
white being due to a single dominant allele (called 2m) and tan colour
from a recessive allele called 2. Thus strong disassortative mating has
a strong effect on the genotype frequencies:<br />
</p>
<div class="center">
<table>
<tbody>
<tr class="odd">
<td style="text-align: center;">Tan</td>
<td style="text-align: center;">White</td>
<td style="text-align: center;">(Super)White</td>
</tr>
<tr class="even">
<td style="text-align: center;">2/2</td>
<td style="text-align: center;">2/2m</td>
<td style="text-align: center;">2m/2m</td>
</tr>
<tr class="odd">
<td style="text-align: center;">978</td>
<td style="text-align: center;">1011</td>
<td style="text-align: center;">3</td>
</tr>
</tbody>
</table>
<p><br />
</p>
</div>
<p>There are almost no 2m homozygotes (so called Super white
individuals) despite the 2m allele being common in the population <span
class="citation" data-cites="tuttle2016divergence">[data from
@tuttle2016divergence table S1]</span>.</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="illustration_images/alleles_genotypes/Sparassis_crispa/Sparassis_crispa_cauliflower_mushroom.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Another important example of disassortative mating are mating type
systems, which are present in many fungi, algae, and protozoa. Gametes
of the same species can only fuse to form a zygote if they differ in
mating type. The mating type of gametes is genetically controled by a
mating type locus, and so individuals are nearly always heterozygous at
this locus. In some groups of organisms, there are just two different
alleles, in other clades these loci have tens or hundreds of
alleles.</p>
<h2
id="allele-sharing-among-related-individuals-and-identity-by-descent">Allele
sharing among related individuals and Identity by Descent</h2>
<p>All of the individuals in a population are related to each other by a
giant pedigree (family tree). For most pairs of individuals in a
population these relationships are very distant (e.g. distant cousins),
while some individuals will be more closely related (e.g. sibling/first
cousins). All individuals are related to one another by varying levels
of relatedness, or <em>kinship</em>. Related individuals can share
alleles that have both descended from the shared common ancestor. To be
shared, these alleles must be inherited through all meioses connecting
the two individuals (e.g. surviving the <span
class="math inline">\(\frac{1}{2}\)</span> probability of segregation
each meiosis). As closer relatives are separated by fewer meioses,
closer relatives share more alleles. In Figure <a
href="#fig:IBD_cousins_chr_cartoon" data-reference-type="ref"
data-reference="fig:IBD_cousins_chr_cartoon">2.3</a> we show the sharing
of chromosomal regions between two cousins. As we’ll see, many
population and quantitative genetic concepts rely on how closely related
individuals are, and thus we need some way to quantify the degree of
kinship among individuals.<br />
</p>
<div class="center">
<figure>
<img src="figures/Cousins_IBD_chromo_cartoon.png"
id="fig:IBD_cousins_chr_cartoon"
alt="First cousins sharing a stretch of chromosome identical by descent. The different grandparental diploid chromosomes are coloured so we can track them and recombinations between them across the generations. Notice that the identity by descent between the cousins persists for a long stretch of chromosome due to the limited number of generations for recombination. The squares represent males and circles females." />
<figcaption aria-hidden="true">First cousins sharing a stretch of
chromosome identical by descent. The different grandparental diploid
chromosomes are coloured so we can track them and recombinations between
them across the generations. Notice that the identity by descent between
the cousins persists for a long stretch of chromosome due to the limited
number of generations for recombination. The squares represent males and
circles females.</figcaption>
</figure>
</div>
<p>We will define two alleles to be identical by descent (IBD) if they
are identical due to transmission from a common ancestor in the past few
generations<span class="citation"
data-cites="cotterman:40 malecot:48">[@cotterman:40;
@malecot:48]</span>. For the moment, we ignore mutation, and we will be
more precise about what we mean by ‘past few generations’ later on. For
example, parent and child share exactly one allele identical by descent
at a locus, assuming that the two parents of the child are randomly
mated individuals from the population. In Figure <a
href="#fig:IBD_cousins_cartoon" data-reference-type="ref"
data-reference="fig:IBD_cousins_cartoon">[fig:IBD_cousins_cartoon]</a>,
I show a pedigree demonstrating some configurations of IBD.<br />
</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/sharing_relatives/IBD_0_1_2.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>One summary of how related two individuals (let’s call them <span
class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span>) are is the probability that our pair
of individuals share 0, 1, or 2 alleles identical by descent (see Figure
<a href="#fig:IBD_0_1_2" data-reference-type="ref"
data-reference="fig:IBD_0_1_2">[fig:IBD_0_1_2]</a>). We denote these
identity-by-descent probabilities by <span
class="math inline">\(r_0\)</span>, <span
class="math inline">\(r_1\)</span>, and <span
class="math inline">\(r_2\)</span> respectively. See Table <a
href="#table:IBDprobs" data-reference-type="ref"
data-reference="table:IBDprobs">[table:IBDprobs]</a> for some examples.
We can also interpret these probabilities as genome-wide averages. For
example, on average, at a quarter of all their autosomal loci full-sibs
share zero alleles identical by descent.<br />
One summary of relatedness that will be important is the probability
that two alleles (I &amp; J) picked at random, one from each of the two
different individuals <span class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span>, are identical by descent (<span
class="math inline">\(P(\text{I\&amp;J IBD})\)</span>). We call this
quantity the <em>coefficient of kinship</em> of individuals <span
class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span>, and denote it by <span
class="math inline">\(F_{ij}\)</span>. It is calculated as</p>
<p><span class="math display">\[\begin{aligned}
  F_{ij} = &amp; \mathbb{P}(\text{I\&amp;J IBD} )\\
  =&amp; \mathbb{P}(\text{I\&amp;J IBD} |~ \text{i\&amp;j  0 IBD})
\mathbb{P}(\text{i\&amp;j  0 IBD})  \nonumber\\
  &amp; + \mathbb{P}(\text{I\&amp;J IBD} |~ \text{i\&amp;j  1 IBD})
    \mathbb{P}(\text{i\&amp;j  1 IBD})  \nonumber\\
  &amp;+ \mathbb{P}(\text{I\&amp;J IBD} |~ \text{i\&amp;j  2 IBD})
\mathbb{P}(\text{i\&amp;j  2 IBD}) \label{eqn:coeffkinship_step}\\
   =   &amp;   0 \times r_0 + \frac{1}{4} r_1  + \frac{1}{2} r_2.
\label{eqn:coeffkinship}
\end{aligned}\]</span></p>
<p>In the above step, eqn<a href="#eqn:coeffkinship_step"
data-reference-type="eqref"
data-reference="eqn:coeffkinship_step">[eqn:coeffkinship_step]</a>,
we’re summing the conditional probability of alleles <span
class="math inline">\(I\)</span> &amp; <span
class="math inline">\(J\)</span> being IBD over whether our individuals
<span class="math inline">\(i\)</span> &amp; <span
class="math inline">\(j\)</span> share <span
class="math inline">\(0\)</span>, <span
class="math inline">\(1\)</span>, or <span
class="math inline">\(2\)</span> alleles IBD, an example of using the
Law of Total Probability (see Appendix eqn<a href="#eqn:law_tot_prob"
data-reference-type="eqref"
data-reference="eqn:law_tot_prob">[eqn:law_tot_prob]</a>). We’ve then,
in eqn<a href="#eqn:coeffkinship" data-reference-type="ref"
data-reference="eqn:coeffkinship">[eqn:coeffkinship]</a>, used the fact
that we can calculate our condition probabilities of I &amp; J being IBD
using the rules of Mendelian transmision. Consider the probability <span
class="math inline">\(P(\text{I\&amp;J IBD} |~ \text{i\&amp;j  1
IBD})\)</span>, i.e. that our pair of alleles (<span
class="math inline">\(I\)</span> &amp; <span
class="math inline">\(J\)</span>) drawn from individuals <span
class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span> are IBD given that <span
class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span> share one allele IBD, this is a <span
class="math inline">\(\frac{1}{4}\)</span> as we need to draw the allele
that is IBD from both <span class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span>, i.e. drawing both black alleles in the
middle panel of Figure <a href="#fig:IBD_0_1_2"
data-reference-type="ref"
data-reference="fig:IBD_0_1_2">[fig:IBD_0_1_2]</a>, which happens with
probability <span class="math inline">\(\frac{1}{2} \times
\frac{1}{2}\)</span>. The coefficient of kinship will appear multiple
times, in both our discussion of inbreeding and in the context of
phenotypic resemblance between relatives.<br />
</p>
<div class="table*">
<div class="center">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Relationship (i,j)<span
class="math inline">\(^{*}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(\mathbb{P}(\text{i\&amp;j  0 IBD})\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(\mathbb{P}(\text{i\&amp;j  1 IBD})\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(P(\text{i\&amp;j  2 IBD})\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(\mathbb{P}(\text{I\&amp;J IBD} )\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Relationship (i,j)<span
class="math inline">\(^{*}\)</span></td>
<td style="text-align: center;"><span
class="math inline">\(r_0\)</span></td>
<td style="text-align: center;"><span
class="math inline">\(r_1\)</span></td>
<td style="text-align: center;"><span
class="math inline">\(r_2\)</span></td>
<td style="text-align: center;"><span
class="math inline">\(F_{ij}\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">parent–child</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">full siblings</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">Monozygotic twins</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><span
class="math inline">\(1^{st}\)</span> cousins</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
</div>
</div>
<p>What are <span class="math inline">\(r_0\)</span>, <span
class="math inline">\(r_1\)</span>, and <span
class="math inline">\(r_2\)</span> for <span
class="math inline">\(\frac{1}{2}\)</span> sibs? (<span
class="math inline">\(\frac{1}{2}\)</span> sibs share one parent but not
the other).</p>
<p>Explain in words why <span
class="math inline">\(\mathbb{P}(\text{I\&amp;J IBD} |~
\text{i\&amp;j  2 IBD}) = \frac{1}{2}\)</span>.</p>
<h5 id="genotypic-sharing-between-pairs-of-individuals.">Genotypic
sharing between pairs of individuals.</h5>
<p>Our <span class="math inline">\(r\)</span> coefficients are going to
have various uses. For example, they allow us to calculate the
probability of the genotypes of a pair of relatives. Consider a
biallelic locus where allele <span class="math inline">\(A_1\)</span> is
at frequency <span class="math inline">\(p\)</span>, and two individuals
who have IBD allele sharing probabilities <span
class="math inline">\(r_0\)</span>, <span
class="math inline">\(r_1\)</span>, <span
class="math inline">\(r_2\)</span>. What is the overall probability that
these two individuals are both homozygous for allele 1? Well that’s
<span class="math display">\[\begin{aligned}
  \mathbb{P}(\textrm{both } A_1 A_1) = &amp; \mathbb{P}(\textrm{both }
A_1 A_1 | \text{0 alleles IBD}) \mathbb{P}(\text{0 alleles
IBD})  \nonumber\\
  &amp; + \mathbb{P}(\textrm{both } A_1 A_1 | \text{1 allele IBD})
\mathbb{P}(\text{1 allele IBD})  \nonumber\\
  &amp;+ \mathbb{P}(\textrm{both } A_1 A_1 | \text{2 alleles IBD})
\mathbb{P}(\text{2 alleles IBD})
\end{aligned}\]</span> Or, in our <span
class="math inline">\(r_0\)</span>, <span
class="math inline">\(r_1\)</span>, <span
class="math inline">\(r_2\)</span> notation: <span
class="math display">\[\begin{aligned}
  \mathbb{P}(\textrm{both } A_1 A_1) = &amp; \mathbb{P}(\textrm{both }
A_1 A_1 | \text{0 alleles IBD}) r_0  \nonumber\\
  &amp; + \mathbb{P}(\textrm{both } A_1 A_1 |
  \text{1 alleles IBD}) r_1  \nonumber\\
  &amp; + \mathbb{P}(\textrm{both } A_1 A_1 | \text{2 alleles IBD}) r_2
\label{eqn:initial_relly_IBD_calc}
\end{aligned}\]</span> If our pair of relatives share <span
class="math inline">\(0\)</span> alleles IBD, then the probability that
they are both homozygous is <span
class="math inline">\(\mathbb{P}(\textrm{both } A_1 A_1 | \text{0
alleles IBD}) =p^2 \times p^2\)</span>, as all four alleles represent
independent draws from the population. If they share <span
class="math inline">\(1\)</span> allele IBD, then the shared allele is
of type <span class="math inline">\(A_1\)</span> with probability <span
class="math inline">\(p\)</span>, and then the other non-IBD allele, in
both relatives, also needs to be <span
class="math inline">\(A_1\)</span> which happens with probability <span
class="math inline">\(p^2\)</span>, so <span
class="math inline">\(\mathbb{P}(\textrm{both } A_1 A_1 | \text{1
alleles IBD})=p \times p^2\)</span>. Finally, our pair of relatives can
share two alleles IBD, in which case <span
class="math inline">\(\mathbb{P}(\textrm{both } A_1 A_1 | \text{2
alleles IBD}) = p^2\)</span>, because if one of our individuals is
homozygous for the <span class="math inline">\(A_1\)</span> allele, both
individuals will be. Putting this all together our eqn <a
href="#eqn:initial_relly_IBD_calc" data-reference-type="eqref"
data-reference="eqn:initial_relly_IBD_calc">[eqn:initial_relly_IBD_calc]</a>
becomes <span class="math display">\[\mathbb{P}(\textrm{both } A_1 A_1)
= p^4 r_0 + p^3 r_1 + p^2 r_2 \label{eqn:IBD_relly_calc}\]</span> Note
that for specific cases we could also calculate this by summing over all
the possible genotypes their shared ancestor(s) had; however, that would
be much more involved and not as general as the form we have derived
here.</p>
<p>We can write out terms like eqn<a href="#eqn:IBD_relly_calc"
data-reference-type="eqref"
data-reference="eqn:IBD_relly_calc">[eqn:IBD_relly_calc]</a> for all of
the possible configurations of genotype sharing/non-sharing between a
pair of individuals. Based on this we can write down the expected number
of polymorphic sites where our individuals are observed to share 0, 1,
or 2 alleles.</p>
<p>The genotype of our suspect in Question <a href="#Q:CODIS"
data-reference-type="ref" data-reference="Q:CODIS">[Q:CODIS]</a> turns
out to be 100/80 for D16S539 and 70/80 at TH01. The suspect is not a
match to the DNA from the crime scene; however, they could be a
sibling.</p>
<p>Calculate the joint probability of observing the genotype from the
crime and our suspect:<br />
<strong>A)</strong> Assuming that they share no close
relationship.<br />
<strong>B)</strong> Assuming that they are full sibs.<br />
<strong>C)</strong> Briefly explain your findings.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/Florida_scrub_jay/8576533889_3a131ffc4c_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>There’s a variety of ways to estimate the relationships among
individuals using genetic data. An example of using allele sharing to
identify relatives is offered by the work of Nancy Chen <span
class="citation" data-cites="chen:16 Aguillon:17">[in collaboration with
Stepfanie Aguillon, see @chen:16; @Aguillon:17]</span>. <span
class="citation" data-cites="chen:16">@chen:16</span> has collected
genotyping data from thousands of Florida Scrub Jays at over ten
thousand loci. These Jays live at the Archbold field site, and have been
carefully monitored for many decades allowing the pedigree of many of
the birds to be known. Using these data, she estimates allele
frequencies at each locus. Then by equating the observed number of times
that a pair of individuals share <span class="math inline">\(0\)</span>,
<span class="math inline">\(1\)</span>, or <span
class="math inline">\(2\)</span> alleles to the theoretical expectation,
she estimates the probability of <span
class="math inline">\(r_0\)</span>, <span
class="math inline">\(r_1\)</span>, and <span
class="math inline">\(r_2\)</span> for each pair of birds. A plot of
these are shown in Figure <a href="#fig:FSJ_IBD"
data-reference-type="ref" data-reference="fig:FSJ_IBD">2.4</a>, showing
how well the estimates match those known from the pedigree.</p>
<div class="center">
<figure>
<img src="figures/FSJ_IBD.jpg" id="fig:FSJ_IBD"
alt="Estimated coefficient of kinship from Florida Scrub Jays. Each point is a pair of individuals, plotted by their estimated IBD (r_1 and r_2) from their genetic data. The points are coloured by their known pedigree relationships. Note that most pairs have low kinship, and no recent genealogical relationship, and so appear as black points in the lower left corner. Thanks to Nancy Chen for supplying the data." />
<figcaption aria-hidden="true">Estimated coefficient of kinship from
Florida Scrub Jays. Each point is a pair of individuals, plotted by
their estimated IBD (<span class="math inline">\(r_1\)</span> and <span
class="math inline">\(r_2\)</span>) from their genetic data. The points
are coloured by their known pedigree relationships. Note that most pairs
have low kinship, and no recent genealogical relationship, and so appear
as black points in the lower left corner. Thanks to Nancy Chen for
supplying the data.</figcaption>
</figure>
</div>
<h5 id="sharing-of-genomic-blocks-among-relatives.">Sharing of genomic
blocks among relatives.</h5>
<div class="center">
<figure>
<embed src="figures/sharing_relatives/First_cousin_overlap.pdf"
id="fig:first_cousin_IBD" />
<figcaption aria-hidden="true">A simulation of sharing between first
cousins. The regions of your grandmother’s 22 autosomes that you
inherited are coloured red, those that your cousins inherited are
coloured blue. In the third panel we show the overlapping genomic
regions in purple, these regions will be IBD in you and your cousin. If
you are full first cousins, you will also have shared genomic regions
from your shared grandfather, not shown here. Details about how we made
these simulations <a
href="https://gcbias.org/2013/12/02/how-many-genomic-blocks-do-you-share-with-a-cousin/">here</a>.</figcaption>
</figure>
</div>
<p>We can more directly see the sharing of the genome among close
relatives using high-density SNP genotyping arrays. In Figure <a
href="#fig:first_cousin_IBD" data-reference-type="ref"
data-reference="fig:first_cousin_IBD">2.5</a> we show a simulation of
first cousins’ genomic sharing from their shared grandmother. Colored
purple are regions where they have matching genomic material, due to
having inherited it IBD from their shared grandmother.</p>
<p>First cousins will share at least one allele of your genotype at all
of the polymorphic loci in these purple regions. There’s a range of
methods to detect such sharing. One way is to look for unusually long
stretches of the genome where two individuals are never homozygous for
different alleles. By identifying pairs of individuals who share an
unusually large number of such putative IBD blocks, we can hope to
identify unknown relatives in genotyping datasets. In fact, companies
like 23&amp;me and Ancestry.com use signals of IBD to help identify
family ties.</p>
<p>As another example, consider the case of third cousins. You share one
of eight sets of great-great-grandparents with each of your (likely
many) third cousins. On average, you and each of your third cousins each
inherit one-sixteenth of your genome from each of those two
great-great-grandparents. This turns out to imply that on average, a
little less than one percent of your and your third cousin’s genomes
(<span class="math inline">\(2 \times (1/16)^2 =0.78\%\)</span>) will be
identical by virtue of descent from those shared ancestors. A simulated
example where third cousins share blocks of their genome (on chromosome
16 and 2) due to their great-great-grandmother is shown in Figure <a
href="#fig:third_cousin_IBD" data-reference-type="ref"
data-reference="fig:third_cousin_IBD">2.6</a>.</p>
<div class="center">
<figure>
<img src="figures/sharing_relatives/Third_cousin_overlap_1.png"
id="fig:third_cousin_IBD"
alt="A simulation of sharing between third cousins, the details are the same as in Figure 2.5." />
<figcaption aria-hidden="true">A simulation of sharing between third
cousins, the details are the same as in Figure <a
href="#fig:first_cousin_IBD" data-reference-type="ref"
data-reference="fig:first_cousin_IBD">2.5</a>.</figcaption>
</figure>
</div>
<p>Note how if you compare Figure <a href="#fig:third_cousin_IBD"
data-reference-type="ref" data-reference="fig:third_cousin_IBD">2.6</a>
and Figure <a href="#fig:first_cousin_IBD" data-reference-type="ref"
data-reference="fig:first_cousin_IBD">2.5</a>, individuals inherit less
IBD from a shared great-great-grandmother than from a shared
grandmother, as they inherit from more total ancestors further back.
Also notice how the sharing occurs in shorter genomic blocks, as it has
passed through more generations of recombination during meiosis. These
blocks are still detectable, and so third cousins can be detected using
high-density genotyping chips, allowing more distant relatives to be
identified than single marker methods alone. More distant relations than
third cousins, e.g. fourth cousins, start to have a significant
probability of sharing none of their genome IBD. But you have many
fourth cousins, so you will share some of your genome IBD with some of
them; however, it gets increasingly hard to identify the degree of
relatedness from genetic data the deeper in the family tree this sharing
goes.</p>
<h3 id="inbreeding">Inbreeding</h3>
<p>We can define an inbred individual as an individual whose parents are
more closely related to each other than two random individuals drawn
from some reference population.<br />
</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/Child_first_cousins_Homozy_BD.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>When two related individuals produce an offspring, that individual
can receive two alleles that are identical by descent, i.e. they can be
homozygous by descent (sometimes termed autozygous), due to the fact
that they have two copies of an allele through different paths through
the pedigree. This increased likelihood of being homozygous relative to
an outbred individual is the most obvious effect of inbreeding. It is
also the one that will be of most interest to us, as it underlies a lot
of our ideas about inbreeding depression and population structure. For
example, in Figure <a href="#fig:IBD_cousins_cartoon"
data-reference-type="ref"
data-reference="fig:IBD_cousins_cartoon">[fig:IBD_cousins_cartoon]</a>
our offspring of first cousins is homozygous by descent having received
the same IBD allele via two different routes around an inbreeding
loop.<br />
As the offspring receives a random allele from each parent (<span
class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span>), the probability that those two
alleles are identical by descent is equal to the kinship coefficient
<span class="math inline">\(F_{ij}\)</span> of the two parents (eqn<a
href="#eqn:coeffkinship" data-reference-type="ref"
data-reference="eqn:coeffkinship">[eqn:coeffkinship]</a>). This follows
from the fact that the genotype of the offspring is made by sampling an
allele at random from each of our parents.</p>
<div class="center">
<div id="table:GeneralizedHWE">
<table>
<caption><strong>Generalized Hardy–Weinberg</strong>. Dropped the
indices <span class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span> for simplicity.</caption>
<thead>
<tr class="header">
<th style="text-align: center;"><span
class="math inline">\(f_{11}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(f_{12}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(f_{22}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\((1-F) p^2 +
F p\)</span></td>
<td style="text-align: center;"><span class="math inline">\((1-F)
2pq\)</span></td>
<td style="text-align: center;"><span class="math inline">\((1-F) q^2 +
F q\)</span></td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The only way the offspring can be heterozygous (<span
class="math inline">\(A_1 A_2\)</span>) is if their two alleles at a
locus are not IBD (otherwise they would necessarily be homozygous).
Therefore, the probability that they are heterozygous is</p>
<p><span class="math display">\[\mathbb{P}(A_1 A_2) = \mathbb{P}(A_1 A_2
|\textrm{I \&amp; J not IBD} ) \mathbb{P}(\textrm{I \&amp; J not IBD} )
= 2p q (1-F_{ij}) ,
\label{eq:hetGenHW}\]</span> The offspring can be homozygous for the
<span class="math inline">\(A_1\)</span> allele in two different ways.
They can have two non-IBD alleles that are not IBD but happen to be of
the allelic type <span class="math inline">\(A_1\)</span>, or their two
alleles can be IBD, such that they inherited allele <span
class="math inline">\(A_1\)</span> by two different routes from the same
ancestor. Thus, the probability that an offspring is homozygous for
<span class="math inline">\(A_1\)</span> is</p>
<p><span class="math display">\[\begin{aligned}
P(A_1 A_1) &amp; = \mathbb{P}(A_1 A_1 |\textrm{I \&amp; J not IBD} )
\mathbb{P}(\textrm{I\&amp;J
  not IBD} ) + \mathbb{P}(A_1 A_1 |\textrm{I\&amp;J IBD} )
\mathbb{P}(\textrm{I\&amp;J IBD}) \nonumber\\
             &amp;= p^2(1-F_{ij})  + pF_{ij}.
\end{aligned}\]</span> using the Law of Total Probability (see Appendix
eqn<a href="#eqn:law_tot_prob" data-reference-type="eqref"
data-reference="eqn:law_tot_prob">[eqn:law_tot_prob]</a>). Therefore,
the frequencies of the three possible genotypes can be written as given
in Table <a href="#table:GeneralizedHWE" data-reference-type="ref"
data-reference="table:GeneralizedHWE">2.1</a>, which provides a
generalization of the Hardy–Weinberg proportions.<br />
<strong>Question</strong> The frequency of the <span
class="math inline">\(A_1\)</span> allele is <span
class="math inline">\(p\)</span> at a biallelic locus. Assume that our
population is randomly mating and that the genotype frequencies in the
population follow from HW. We select two individuals at random to mate
from this population. We then mate the children from this cross. What is
the probability that the child from this full sib-mating is
homozygous?</p>
<h5 id="multiple-inbreeding-loops-in-a-pedigree.">Multiple inbreeding
loops in a pedigree.</h5>
<p>Up to this point we have assumed that there is at most one inbreeding
loop in the recent family history of our individuals, i.e. the parents
of our inbred individual have at most one recent genealogical
connection. However, an individual who has multiple inbreeding loops in
their pedigree can be homozygous by descent thanks to receiving IBD
alleles via multiple different different loops. To calculate inbreeding
in pedigrees of arbitrary complexity, we can extend beyond our original
relatedness coefficients <span class="math inline">\(r_0\)</span>, <span
class="math inline">\(r_1\)</span>, and <span
class="math inline">\(r_2\)</span> to account for higher order sharing
of alleles IBD among relatives. For example, we can ask, what is the
probability that <em>both</em> of the alleles in the first individual
are shared IBD with one allele in the second individual? There are nine
possible relatedness coefficients in total to completely describe
kinship between two diploid individuals, and we won’t go in to them here
as it’s a lot to keep track of. However, we will show how we can
calculate the inbreeding coefficient of an individual with multiple
inbreeding loops more directly.<br />
Let’s say the parents of our inbred individual (B and C) have <span
class="math inline">\(K\)</span> shared ancestors, i.e. individuals who
appear in both B and C’s recent family trees. We denote these shared
ancestors by <span class="math inline">\(A_1, \dots,A_K\)</span>, and we
denote by <span class="math inline">\(n\)</span> the total number of
individuals in the chain from B to C via ancestor <span
class="math inline">\(A_i\)</span>, including B, C, and <span
class="math inline">\(A_i\)</span>. For example, if B is C’s aunt, then
B and C share two ancestors, which are B’s parents and, equivalently,
C’s grandparents. In this case, there are n=4 individuals from B to C
through each of these two shared ancestor. In the general case, the
kinship coefficient of B and C, i.e. the inbreeding coefficient of their
child, is <span class="math display">\[F = \sum_{i=1}^K
\frac{1}{2^{n_i}} \big( 1+ f_{A_i} \big)
\label{eqn:inbreeding_over_ancs}\]</span> where <span
class="math inline">\(f_{A_i}\)</span> is the inbreeding coefficient of
the ancestor <span class="math inline">\(A_i\)</span>. What’s happening
here is that we sum over all the mutually-exclusive paths in the
pedigree through which B and C can share an allele IBD. With probability
<span class="math inline">\(\frac{1}{2^{n_i}}\)</span>, a pair of
alleles picked at random from B and C is descended from the same
ancestral allele in individual <span class="math inline">\(A_i\)</span>,
in which case the alleles are IBD. However, even if B inherits the
maternal allele and C inherits the paternal allele of shared ancestor
<span class="math inline">\(A_i\)</span>, if <span
class="math inline">\(A_i\)</span> was themselves inbred, with
probability <span class="math inline">\(f_{A_i}\)</span> those two
alleles are themselves IBD. Thus a shared <em>inbred</em> ancestor
further increases the kinship of B and C.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/alleles_genotypes/Charles_second_pedigree/Carlos_second_pedigree_2_trimmed.pdf"
id="fig:Carlos_second_pedigree" />
<figcaption aria-hidden="true">The pedigree of King Charles II of Spain.
Pedigree from <a
href="https://commons.wikimedia.org/wiki/File:Carlos_segundo80.png">wikimedia</a>
drawn by <a href="https://en.wikipedia.org/wiki/User:Lec_CRP1">Lec
CRP1</a>, public domain.</figcaption>
</figure>
</div>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/Carlos_second/378px-Juan_de_Miranda_Carreno_002.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Multiple inbreeding loops increase the probability that a child is
homozygous by descent at a locus, which can be calculated simply by
plugging in <span class="math inline">\(F\)</span>, the child’s
inbreeding coefficient, into our generalized HW equation.</p>
<p>As one extreme example of the impact of multiple inbreeding loops in
an individual’s pedigree, let’s consider king Charles II of Spain, the
last of the Spanish Habsburgs. Charles was the son of Philip IV of Spain
and Mariana of Austria, who were uncle and niece. If this were the only
inbreeding loop, then Charles would have had an inbreeding coefficient
of <span class="math inline">\(\frac{1}{8}\)</span>. Unfortunately for
Charles, the Spanish Habsburgs had long kept wealth and power within
their family by arranging marriages between close kin. The pedigree of
Charles II is shown in Figure <a href="#fig:Carlos_second_pedigree"
data-reference-type="ref"
data-reference="fig:Carlos_second_pedigree">2.7</a>, and multiple
inbreeding loops are apparent. For example, Phillip III, Charles II’s
grandfather and great-grandfather, was himself a child of an uncle-niece
marriage.</p>
<p><span class="citation" data-cites="alvarez:09">@alvarez:09</span>
calculated that Charles II had an inbreeding coefficient of <span
class="math inline">\(0.254\)</span>, equivalent to a full-sib mating,
thanks to all of the inbreeding loops in his pedigree. Therefore, he is
expected to have been homozygous by descent for a full quarter of his
genome. As we’ll talk about later in these notes, this means that
Charles may have been homozygous for a number of recessive disease
alleles, and indeed he was a very sickly man who left no descendants due
to his infertility. Thus plausibly the end of one of the great European
dynasties came about through inbreeding.</p>
<h3
id="calculating-inbreeding-coefficients-from-genetic-data">Calculating
inbreeding coefficients from genetic data</h3>
<p>If the observed heterozygosity in a population is <span
class="math inline">\(H_O\)</span>, and we assume that the generalized
Hardy–Weinberg proportions hold, we can set <span
class="math inline">\(H_O\)</span> equal to <span
class="math inline">\(f_{12}\)</span>, and solve Eq. <a
href="#eq:hetGenHW" data-reference-type="eqref"
data-reference="eq:hetGenHW">[eq:hetGenHW]</a> for <span
class="math inline">\(F\)</span> to obtain an estimate of the inbreeding
coefficient as</p>
<p><span class="math display">\[\hat{F} = 1-\frac{f_{12}}{2pq} =
\frac{2pq - f_{12}}{2pq}.
\label{eqn:Fhat}\]</span> As before, <span
class="math inline">\(p\)</span> is the frequency of allele <span
class="math inline">\(A_{1}\)</span> in the population. This can be
rewritten in terms of the observed heterozygosity (<span
class="math inline">\(H_O\)</span>) and the heterozygosity expected in
the absence of inbreeding, <span class="math inline">\(H_E=2pq\)</span>,
as <span class="math display">\[\hat{F} = \frac{H_E-H_O}{H_E} = 1 -
\frac{H_O}{H_E}.
\label{eqn:FhatHO}\]</span> Hence, <span
class="math inline">\(\hat{F}\)</span> quantifies the deviation due to
inbreeding of the observed heterozygosity from the one expected under
random mating, relative to the latter.</p>
<p>Suppose the following genotype frequencies were observed for an
esterase locus in a population of <em>Drosophila</em> (A denotes the
“fast" allele and B denotes the”slow" allele):</p>
<div class="center">
<table>
<thead>
<tr class="header">
<th style="text-align: center;">AA</th>
<th style="text-align: center;">AB</th>
<th style="text-align: center;">BB</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">0.6</td>
<td style="text-align: center;">0.2</td>
<td style="text-align: center;">0.2</td>
</tr>
</tbody>
</table>
</div>
<p>What is the estimate of the inbreeding coefficient at the esterase
locus?</p>
<p>If we have multiple loci, we can replace <span
class="math inline">\(H_O\)</span> and <span
class="math inline">\(H_E\)</span> by their means over loci, <span
class="math inline">\(\bar{H}_O\)</span> and <span
class="math inline">\(\bar{H}_E\)</span>, respectively. Note that, in
principle, we could also calculate <span
class="math inline">\(F\)</span> for each individual locus first, and
then take the average across loci. However, this procedure is more prone
to introducing a bias if sample sizes vary across loci, which is not
unlikely when we are dealing with real data.</p>
<p>Genetic markers are commonly used to estimate inbreeding for wild
and/or captive populations of conservation concern. As an example of
this, consider the case of the Mexican wolf (<em>Canis lupus
baileyi</em>), a sub-species of gray wolf.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/grey_wolf/5988399184_0c36a8e51c_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>They were extirpated in the wild during the mid-1900s due to hunting,
and the remaining five Mexican wolves in the wild were captured to start
a breeding program. <span class="citation"
data-cites="vonHoldt:11">@vonHoldt:11</span> estimated the current-day,
average expected heterozygosity to be <span
class="math inline">\(0.18\)</span>, based on allele frequencies at over
forty thousand SNPs. However, the average Mexican wolf individual was
only observed to be heterozygous at <span
class="math inline">\(12\%\)</span> of these SNPs. Therefore, the
average inbreeding coefficient for the Mexican wolf is <span
class="math inline">\(\hat{F} = 1 -\frac{0.12}{0.18}\)</span>,
i.e. <span class="math inline">\(\sim 33 \%\)</span> of a lobo’s genome
is homozygous due to recent inbreeding in their pedigree.</p>
<h5 id="genomic-blocks-of-homozygosity-due-to-inbreeding.">Genomic
blocks of homozygosity due to inbreeding.</h5>
<p>As we saw above, close relatives are expected to share alleles IBD in
large genomic blocks. Thus, when related individuals mate and transmit
alleles to an inbred offspring, they transmit these alleles in big
blocks through meiosis. As an example, let’s return to the case of our
hypothetical first cousins from Figure <a
href="#fig:IBD_cousins_chr_cartoon" data-reference-type="ref"
data-reference="fig:IBD_cousins_chr_cartoon">2.3</a>. If this pair of
individuals had a child, one possible pattern of genetic transmission is
shown in Figure <a href="#fig:kid_first_cousins"
data-reference-type="ref"
data-reference="fig:kid_first_cousins">2.8</a>. The child has inherited
the red stretch of chromosome via two different routes through their
predigree from the grandparents. This is an example of an autozygous
segment, where the child is homozygous by descent at all of the loci in
this red region.</p>
<div class="center">
<figure>
<embed src="figures/sharing_relatives/first_cousin_offspring.pdf"
id="fig:kid_first_cousins" />
<figcaption aria-hidden="true">A pedigree showing the offspring of first
cousins. The chromosomes of their great-grandparents are coloured
different colours so their transmission can be tracked. The child is
homozygous by descent (HBD) for a section of the red
chromosome.</figcaption>
</figure>
</div>
<p>The inbreeding coefficient of the child sets the proportion of their
genome that will be in these autozygous segments. For example, a child
of first full cousins is expected to have <span
class="math inline">\(1/16\)</span> of their genome in these segments.
The more distant the loop in the pedigree, the more meioses that
chromosomes have been through and the shorter individual blocks will be.
A child of first cousins will have longer blocks than a child of second
cousins, for example.</p>
<p>Individuals with multiple inbreeding loops in their family tree can
have a high inbreeding coefficient due to the combined effect of many
small blocks of autozygosity. For example, Charles II had an inbreeding
coefficient that is equivalent to that of the child of full-sibs, with a
quarter of his genome expected to homozygous by descent, but this would
be made up of many shorter blocks.</p>
<p>We can hope to detect these blocks by looking for unusually long
genomic runs of homozygosity (ROH) sites in an individual’s genome. One
way to estimate an individual’s inbreeding coefficient is then to total
up the proportion of an individual’s genome that falls in such ROH
regions. This estimate is called <span
class="math inline">\(F_{ROH}\)</span>.</p>
<p>An example of using <span class="math inline">\(F_{ROH}\)</span> to
study inbreeding comes from the work of <span class="citation"
data-cites="sams2018fine">@sams2018fine</span>, who identified runs of
homozygosity in 2,500 dogs, ranging from 500kb up to many megabases.</p>
<div class="center">
<figure>
<embed src="figures/sharing_relatives/dogs_FROH.pdf" id="fig:dog_FOH" />
<figcaption aria-hidden="true">The distribution of <span
class="math inline">\(F_{ROH}\)</span> of individuals from various dog
breeds from <span class="citation" data-cites="Sams:18">@Sams:18</span>,
.</figcaption>
</figure>
</div>
<p>Figure <a href="#fig:dog_FOH" data-reference-type="ref"
data-reference="fig:dog_FOH">2.9</a> shows the distribution of <span
class="math inline">\(F_{ROH}\)</span> of individuals in each dog breed
for the X and autosome. In Figure <a href="#fig:dog_FOH_dist"
data-reference-type="ref" data-reference="fig:dog_FOH_dist">2.10</a>
this is broken down by the length of ROH segments.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/english_bulldog/14752595581_4330377c97_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<div class="center">
<figure>
<embed src="figures/sharing_relatives/dog_FROH_dist.pdf"
id="fig:dog_FOH_dist" />
<figcaption aria-hidden="true">Cumulative density of ROH length,
measured in megabases (Mb) from <span class="citation"
data-cites="Sams:18">@Sams:18</span> for various dog breeds (). Note
that longer lengths of ROH are on the left of the plot.</figcaption>
</figure>
</div>
<p>Dog breeds have been subject to intense breeding that has resulted in
high levels of inbreeding. Of the population samples examined, Doberman
Pinschers have the highest levels of their genome in runs of
homozygosity (<span class="math inline">\(F_{ROH}\)</span>), somewhat
higher than English bulldogs. In <a href="#fig:dog_FOH_dist"
data-reference-type="ref" data-reference="fig:dog_FOH_dist">2.10</a> we
can see that English bulldogs have more short ROH than Doberman
Pinschers, but that Doberman Pinschers have more of their genome in very
large ROH (<span class="math inline">\(&gt;16 Mb\)</span>). This
suggests that English bulldogs have had long history of inbreeding as
they have many small blocks, but that Doberman Pinschers have a lot of
recent inbreeding as their autozygosity is contained in long blocks
relatively unbroken by recombination.</p>
<div class="ChapterSummary">
<p>This chapter developed the relationship between allele frequencies
and genotype frequencies within a generation and among relatives.</p>
<p>Under random mating, we derived expectations of the genotype
frequencies (Hardy-Weinberg), and we can identify deviations away from
these expectations.</p>
<p>Identity by descent (IBD) refers to the sharing of alleles due to a
recent shared biological relationship.</p>
<p>We can predict the probability and expected level of sharing of
alleles IBD among pairs of relatives using mendelian transmission
probabilities (as contained in coefficients <span
class="math inline">\(r_0\)</span>, <span
class="math inline">\(r_1\)</span>, and <span
class="math inline">\(r_2\)</span>). One useful summary of relatedness
for a pair of individuals is the kinship coefficient <span
class="math inline">\(F_{i,j}\)</span>.</p>
<p>We can also learn about genetic relationships from the sharing of
genomic segments among relatives, with many long shared segments
revealing a closer relationship.</p>
<p>An inbred individual has parents who are more closely related than
random draws from some reference population.</p>
<p>Inbreeding results in decreased heterozygosity and a complementary
increase in homozygosity. We can use the kinship coefficient of the
parents to estimate the distortion away from Hardy-Weinberg and the
expected level of heterozygosity.</p>
<p>Inbreeding coefficients can be calculated from genetic data, either
for multiple individuals at a single locus or for multiple loci for a
single individual.</p>
</div>
<p>Calculate <span class="math inline">\(r_{0}\)</span>, <span
class="math inline">\(r_{1}\)</span>, <span
class="math inline">\(r_{2}\)</span> and the coefficient of kinship
<span class="math inline">\(F\)</span> between:<br />
<strong>A)</strong> A grandparent and their grandchild<br />
<strong>B)</strong> A great grandparent and their great grandchild<br />
<strong>C)</strong> Full siblings<br />
<strong>D)</strong> A great aunt and her grand nephew (your great aunt =
your parent’s aunt)<br />
</p>
<p>You are studying a codominant flower colour polymorphism. Skipping
through a meadow of flowers you and compile the following data:</p>
<div class="center">
<table>
<tbody>
<tr class="odd">
<td style="text-align: center;">red</td>
<td style="text-align: center;">pink</td>
<td style="text-align: center;">white</td>
</tr>
<tr class="even">
<td style="text-align: center;">200</td>
<td style="text-align: center;">100</td>
<td style="text-align: center;">200</td>
</tr>
</tbody>
</table>
</div>
<p><strong>A)</strong> What frequencies would you expect at this locus
under Hardy-Weinberg equilibrium?<br />
<strong>B)</strong> Calculate the inbreeding coefficient at this
locus.<br />
<strong>C)</strong> Name two distinct processes that could lead to the
deviation you see, and describe how they would result in a deficit of
heterozygotes.</p>
<p>What are the relatedness coefficients of the X chromosome
between:<br />
<strong>A)</strong> Two male full siblings?<br />
<strong>B)</strong> Two female full siblings?<br />
<strong>C)</strong> What is the probability that a female offspring of a
full sib mating is homozygous by descent at a locus on her X
chromosome?</p>
<p>You are studying the wing spot polymorphism in a butterfly species.
From crosses in the lab you find that the presence of wing spots is
determined by a dominant allele.<br />
You collect 100 butterflies, 84 of them have the wing spots. What is the
frequency of the wing-spot allele? What assumption did you have to make
to come to your answer?</p>
<p>An allele has frequency of <span class="math inline">\(0.001\)</span>
in the population. What is the probability that both you and your first
(full) cousin are heterozygote for the allele?</p>
<p>The kinship coefficient of the parents is the inbreeding coefficient
of the offspring. Explain, with reference to the weighting of
relatedness coefficients in the inbreeding coefficient, why the
inbreeding coefficient is the probability that a locus is homozygous by
descent.</p>
<p>In terms of identity by descent, explain why multiple inbreeding
loops in an individual’s pedigree lead to higher levels of
inbreeding.</p>
<h1 id="population-structure-and-correlations-among-loci.">Population
Structure and Correlations Among Loci.</h1>
<p>; your parents weren’t two Bilateria plucked at random from the tree
of life. Even within species, there’s often geographically-restricted
mating among individuals. Individuals tend to mate with individuals from
the same, or closely related sets of populations. This form of
non-random mating is called population structure and can have profound
effects on the distribution of genetic variation within and among
natural populations.</p>
<p>Populations can often differ in their allele frequencies, either due
to genetic drift or selection driving differentiation among populations.
In this chapter we’ll talk through some ways to summarize and visualize
population genetic structure. Population differentiation is also a major
driver of correlations in allelic state among loci, and we’ll start our
discussion of these correlations at the end of this chapter. One reason
for talking about population structure so early in the book is that
summarizing population structure is often a key initial stage in
population genomic analyses. Thus you’ll often encounter summaries and
visualizations of population structure when we read research papers, so
it’s good to have some understanding of what they represent.</p>
<h3 id="section:F_stats">Inbreeding as a summary of population
structure.</h3>
<p>Our statements about inbreeding, and inbreeding coefficients,
represent one natural way to summarize population structure. In the
previous chapter, we defined inbreeding as having parents that are more
closely related to each other than two individuals drawn at random from
some reference population. The question that naturally arises is: Which
reference population should we use? While I might not look inbred in
comparison to allele frequencies in the United Kingdom (UK), where I am
from, my parents certainly are not two individuals drawn at random from
the world-wide population. If we estimated my inbreeding coefficient
<span class="math inline">\(F\)</span> using allele frequencies within
the UK, it would be close to zero, but would likely be larger if we used
world-wide frequencies. This is because there is a somewhat lower level
of expected heterozygosity within the UK than in the human population
across the world as a whole.<br />
Building on this idea of inbreeding coefficients estimated at various
levels, <span class="citation" data-cites="Wright:43">@Wright:43</span>
developed a set of ‘F-statistics’ (also called ‘fixation indices’) that
formalize the idea of inbreeding with respect to different levels of
population structure <span class="citation"
data-cites="Wright:43 Wright:49">[@Wright:43; @Wright:49]</span>. See
Figure <a href="#fig:FST_pig" data-reference-type="ref"
data-reference="fig:FST_pig">[fig:FST_pig]</a> for a schematic diagram.
Wright defined <span class="math inline">\(F_{\mathrm{XY}}\)</span> as
the correlation between random gametes, drawn from the same level <span
class="math inline">\(X\)</span>, relative to level <span
class="math inline">\(Y\)</span>. We will return to why <span
class="math inline">\(F\)</span>-statistics are statements about
correlations between alleles in just a moment. One commonly used <span
class="math inline">\(F\)</span>-statistic is <span
class="math inline">\(F_{\mathrm{IS}}\)</span>, which is the inbreeding
coefficient between an individual (<span
class="math inline">\(I\)</span>) and the subpopulation (<span
class="math inline">\(S\)</span>). Consider a single locus, where in a
subpopulation (<span class="math inline">\(S\)</span>) a fraction <span
class="math inline">\(H_I=f_{12}\)</span> of individuals are
heterozygous. In this subpopulation, let the frequency of allele <span
class="math inline">\(A_1\)</span> be <span
class="math inline">\(p_S\)</span>, such that the expected
heterozygosity under random mating is <span class="math inline">\(H_S =
2 p_S (1 - p_S)\)</span>. We will write <span
class="math inline">\(F_{\mathrm{IS}}\)</span> as</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/Pop_struct/FST_hierarchy.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p><span class="math display">\[F_{\mathrm{IS}}= 1-\frac{H_I}{H_S}=
1-\frac{f_{12}}{2p_Sq_S},
\label{eqn:FIS}\]</span> a direct analog of eqn<a href="#eqn:Fhat"
data-reference-type="ref" data-reference="eqn:Fhat">[eqn:Fhat]</a>.
Hence, <span class="math inline">\(F_{\mathrm{IS}}\)</span> is the
relative difference between observed and expected heterozygosity due to
a deviation from random mating within the subpopulation. We could also
compare the observed heterozygosity in individuals (<span
class="math inline">\(H_I\)</span>) to that expected in the total
population, <span class="math inline">\(H_T\)</span>. If the frequency
of allele <span class="math inline">\(A_1\)</span> in the total
population is <span class="math inline">\(p_T\)</span>, then we can
write <span class="math inline">\(F_{\mathrm{IT}}\)</span> as <span
class="math display">\[F_{\mathrm{IT}}=1-\frac{H_I}{H_T}=
1-\frac{f_{12}}{2p_Tq_T},
\label{eqn:FIT}\]</span> which compares heterozygosity in individuals to
that expected in the total population. As a simple extension of this, we
could imagine comparing the expected heterozygosity in the subpopulation
(<span class="math inline">\(H_S\)</span>) to that expected in the total
population <span class="math inline">\(H_T\)</span>, via <span
class="math inline">\(F_{\mathrm{ST}}\)</span>: <span
class="math display">\[F_{\mathrm{ST}}=
1-\frac{H_S}{H_T}=1-\frac{2p_Sq_S}{2p_Tq_T} \label{eqn:FST}.\]</span> We
can relate the three <span class="math inline">\(F\)</span>-statistics
to each other as <span class="math display">\[(1-F_{\mathrm{IT}})
=\frac{H_I}{H_S} \frac{H_S}{H_T}=(1-F_{\mathrm{IS}})(1-F_{\mathrm{ST}}).
\label{eqn:F_relationships}\]</span> Hence, the reduction in
heterozygosity within individuals compared to that expected in the total
population can be decomposed to the reduction in heterozygosity of
individuals compared to the subpopulation, and the reduction in
heterozygosity from the total population to that in the
subpopulation.<br />
If we want a summary of population structure across multiple
subpopulations, we can average <span class="math inline">\(H_I\)</span>
and/or <span class="math inline">\(H_S\)</span> across populations, and
use a <span class="math inline">\(p_T\)</span> calculated by averaging
<span class="math inline">\(p_S\)</span> across subpopulations (or our
samples from sub-populations). For example, the average <span
class="math inline">\(\bar{F_{\mathrm{ST}}}\)</span> across <span
class="math inline">\(K\)</span> subpopulations (sampled with equal
effort) is <span class="math display">\[\bar{F_{\mathrm{ST}}} = 1 -
\frac{\bar{H}_{S}}{H_T},\]</span> where <span
class="math inline">\(\bar{H}_S = \frac{1}{K} \sum_{i = 1}^{K}
H_{S}^{(i)}\)</span>, and <span class="math inline">\(H_{S}^{(i)} = 2
p_{i} q_{i}\)</span> is the expected heterozygosity in subpopulation
<span class="math inline">\(i\)</span>. It follows that the average
heterozygosity of the sub-populations <span
class="math inline">\(\bar{H}_S \leq H_T\)</span>, and so <span
class="math inline">\(\bar{F_{\mathrm{ST}}} \geq 0\)</span> and <span
class="math inline">\(\bar{F_{\mathrm{IS}}} \leq
\bar{F_{\mathrm{IT}}}\)</span>. This observation that the average
heterozygosity of the sub-populations must be less than of equal to that
of the total population is called the Wahlund effect <span
class="citation"
data-cites="wahlund1928zusammensetzung">[@wahlund1928zusammensetzung]</span>.
Furthermore, if we have multiple sites, we can replace <span
class="math inline">\(H_I\)</span>, <span
class="math inline">\(H_S\)</span>, and <span
class="math inline">\(H_T\)</span> with their averages across loci (as
above).</p>
<p>In a species of lemurs, you estimate the allele frequency to be <span
class="math inline">\(20\%\)</span>. In a particular population, you
estimate that the allele frequency is <span
class="math inline">\(10\%\)</span>. In this population, only <span
class="math inline">\(9\%\)</span> of individuals are heterozygote. What
is <span class="math inline">\(F_{IT}\)</span>, <span
class="math inline">\(F_{ST}\)</span>, and <span
class="math inline">\(F_{IS}\)</span> for this population?</p>
<p>As an example of comparing a genome-wide estimate of <span
class="math inline">\(F_{ST}\)</span> to that at individual loci we can
look at some data from blue- and golden-winged warblers (<em>Vermivora
cyanoptera</em> and <em>V. chrysoptera</em> 1-2 &amp; 5-6 in Figure <a
href="#fig:blue_golden_warblers" data-reference-type="ref"
data-reference="fig:blue_golden_warblers">[fig:blue_golden_warblers]</a>).</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/blue_golden_winged_warblers/The_warblers_of_North_America_6309257188.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>These two species are spread across eastern Northern America, with
the golden-winged warbler having a smaller, more northernly range.
They’re quite different in terms of plumage, but have long been known to
have similar songs and ecologies. The two species hybridize readily in
the wild; in fact two other previously-recognized species, Brewster’s
and Lawrence’s warbler (4 &amp; 3 in <a href="#fig:blue_golden_warblers"
data-reference-type="ref"
data-reference="fig:blue_golden_warblers">[fig:blue_golden_warblers]</a>),
are actually found to just be hybrids between theses two species. The
golden-winged warbler is listed as ‘threatened’ under the Canadian
endangered species act as its habitat is under pressure from human
activity and and due to increasing hybridization with the blue-winged
warbler, which is moving north into its range. <span class="citation"
data-cites="Toews:16">@Toews:16</span> investigated the population
genomics of these warblers, sequencing ten golden- and ten blue-winged
warblers. They found very low divergence among these species, with a
genome-wide <span class="math inline">\(F_{ST}=0.0045\)</span>. In
Figure <a href="#fig:warbler_FST" data-reference-type="ref"
data-reference="fig:warbler_FST">[fig:warbler_FST]</a>, per SNP <span
class="math inline">\(F_{ST}\)</span> is averaged in <span
class="math inline">\(2000\)</span>bp windows moving along the genome.
The average is very low, but some regions of very high <span
class="math inline">\(F_{ST}\)</span> stand out. Nearly all of these
regions correspond to large allele frequency differences at loci in, or
close, to genes known to be involved in plumage colouration differences
in other birds.</p>
<div class="figure*">
<div class="center">
<figure>
<img
src="Journal_figs/alleles_genotypes/blue_golden_winged_warblers/GW_FST_warblers.png"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>To illustrate these frequency differences <span class="citation"
data-cites="Toews:16">@Toews:16</span> genotyped a SNP in each of these
high-<span class="math inline">\(F_{ST}\)</span> regions. Here’s their
genotyping counts from the SNP, segregating for an allele 1 and 2, in
the <em>Wnt</em> region, a key regulatory gene involved in feather
development:</p>
<div class="center">
<table>
<tbody>
<tr class="odd">
<td></td>
<td style="text-align: center;">Genotypes</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Species</td>
<td style="text-align: center;">11</td>
<td>12</td>
<td>22</td>
</tr>
<tr class="odd">
<td>Blue-winged</td>
<td style="text-align: center;">2</td>
<td>21</td>
<td>31</td>
</tr>
<tr class="even">
<td>Golden-winged</td>
<td style="text-align: center;">48</td>
<td>12</td>
<td>1</td>
</tr>
</tbody>
</table>
</div>
<p>With reference to the table of <em>Wnt</em>-allele counts:<br />
<strong>A)</strong> Calculate <span
class="math inline">\(F_{IS}\)</span> in blue-winged warblers.<br />
<strong>B)</strong> Calculate <span
class="math inline">\(F_{ST}\)</span> for the sub-population of
blue-winged warblers compared to the combined sample.<br />
<strong>C)</strong> Calculate mean <span
class="math inline">\(F_{ST}\)</span> across both sub-populations.</p>
<h5 id="interpretations-of-f-statistics">Interpretations of
F-statistics</h5>
<p>Let’s now return to Wright’s definition of the <span
class="math inline">\(F\)</span>-statistics as correlations between
random gametes, drawn from the same level <span
class="math inline">\(X\)</span>, relative to level <span
class="math inline">\(Y\)</span>. Without loss of generality, we may
think about <span class="math inline">\(X\)</span> as individuals and
<span class="math inline">\(S\)</span> as the subpopulation. Rewriting
<span class="math inline">\(F_{\mathrm{IS}}\)</span> in terms of the
observed homozygote frequencies (<span
class="math inline">\(f_{11}\)</span>, <span
class="math inline">\(f_{22}\)</span>) and expected homozygosities
(<span class="math inline">\(p_{S}^2\)</span>, <span
class="math inline">\(q_{S}^2\)</span>) we find <span
class="math display">\[F_{\mathrm{IS}}= \frac{2p_Sq_S - f_{12}}{2p_Sq_S}
= \frac{f_{11}+f_{22} -
p_S^2 - q_S^2}{2p_Sq_S},
\label{eqn:Fascorr}\]</span> using the fact that <span
class="math inline">\(p^2+2pq+q^2=1\)</span>, and <span
class="math inline">\(f_{12} = 1 - f_{11} - f_{12}\)</span>. The form of
eqn. (<a href="#eqn:Fascorr" data-reference-type="ref"
data-reference="eqn:Fascorr">[eqn:Fascorr]</a>) reveals that <span
class="math inline">\(F_{\mathrm{IS}}\)</span> is the covariance between
pairs of alleles found in an individual, divided by the expected
variance under binomial sampling. Thus, <span
class="math inline">\(F\)</span>-statistics can be understood as the
correlation between alleles drawn from a population (or an individual)
above that expected by chance (i.e. drawing alleles sampled at random
from some broader population).<br />
We can also interpret <span class="math inline">\(F\)</span>-statistics
as proportions of variance explained by different levels of population
structure. To see this, let us think about <span
class="math inline">\(F_{\mathrm{ST}}\)</span> averaged over <span
class="math inline">\(K\)</span> subpopulations, whose frequencies are
<span class="math inline">\(p_1,\dots,p_K\)</span>. The frequency in the
total population is <span class="math inline">\(p_T=\bar{p} =
\frac{1}{K} \sum_{i=1}^K p_i\)</span>. Then, we can write <span
class="math display">\[\begin{aligned}
F_{\mathrm{ST}}&amp;= \frac{2 \bar{p}\bar{q} - \frac{1}{K}\sum_{i=1}^K
2p_iq_i }{2
\bar{p}\bar{q}} = \frac{ \left(\frac{1}{K} \sum_{i=1}^K p_i^2 +
\frac{1}{K} \sum_{i=1}^K q_i^2 \right) -  \bar{p}^2-\bar{q}^2 }{2
       \bar{p}\bar{q}}  \nonumber\\
       &amp;= \frac{\mathrm{Var}(p_1,\dots,p_K)}{\mathrm{Var}(\bar{p})},
\label{eqn:F_as_propvar}
\end{aligned}\]</span> which shows that <span
class="math inline">\(F_{\mathrm{ST}}\)</span> is the proportion of the
variance explained by the subpopulation labels.</p>
<h3 id="other-approaches-to-population-structure">Other approaches to
population structure</h3>
<p>There is a broad spectrum of methods to describe patterns of
population structure in population genetic datasets. We’ll briefly
discuss two broad-classes of methods that appear often in the
literature: assignment methods and principal components analysis.</p>
<h3 id="assignment-methods">Assignment Methods</h3>
<p>Here we’ll describe a simple probabilistic assignment to find the
probability that an individual of unknown population comes from one of
<span class="math inline">\(K\)</span> predefined populations. For
example, there are three broad populations of common chimpanzee (<em>Pan
troglodytes</em>) in Africa: western, central, and eastern. Imagine that
we have a chimpanzee whose population of origin is unknown (e.g. it’s
from an illegal private collection). If we have genotyped a set of
unlinked markers from a panel of individuals representative of these
populations, we can calculate the probability that our chimp comes from
each of these populations.<br />
We’ll then briefly explain how to extend this idea to cluster a set of
individuals into <span class="math inline">\(K\)</span> initially
unknown populations. This method is a simplified version of what
population genetics clustering algorithms such as STRUCTURE and
ADMIXTURE do. <span class="citation"
data-cites="pritchard:00 alexander:09">[@pritchard:00;
@alexander:09]</span></p>
<h5 id="a-simple-assignment-method">A simple assignment method</h5>
<p>We have genotype data from unlinked <span
class="math inline">\(S\)</span> biallelic loci for <span
class="math inline">\(K\)</span> populations. The allele frequency of
allele <span class="math inline">\(A_1\)</span> at locus <span
class="math inline">\(l\)</span> in population <span
class="math inline">\(k\)</span> is denoted by <span
class="math inline">\(p_{k,l}\)</span>, so that the allele frequencies
in population 1 are <span class="math inline">\(p_{1,1},\cdots
p_{1,L}\)</span> and population 2 are <span
class="math inline">\(p_{2,1},\cdots p_{2,L}\)</span> and so on.</p>
<p>You genotype a new individual from an unknown population at these
<span class="math inline">\(L\)</span> loci. This individual’s genotype
at locus <span class="math inline">\(l\)</span> is <span
class="math inline">\(g_l\)</span>, where <span
class="math inline">\(g_l\)</span> denotes the number of copies of
allele <span class="math inline">\(A_1\)</span> this individual carries
at this locus (<span class="math inline">\(g_l=0,1,2\)</span>).</p>
<p>The probability of this individual’s genotype at locus <span
class="math inline">\(l\)</span> conditional on coming from population
<span class="math inline">\(k\)</span>, i.e. their alleles being a
random HW draw from population <span class="math inline">\(k\)</span>,
is <span class="math display">\[\mathbb{P}(g_l | \textrm{pop k}) =
  \begin{cases}
(1-p_{k,l})^2  &amp; g_l=0 \\
2 p_{k,l} (1-p_{k,l}) &amp; g_l=1\\
p_{k,l}^2  &amp; g_l=2
\end{cases}\]</span></p>
<p>Assuming that the loci are independent, the probability of the
individual’s genotype across all S loci, conditional on the individual
coming from population <span class="math inline">\(k\)</span>, is <span
class="math display">\[\mathbb{P}(\textrm{ind.} | \textrm{pop k})  =
\prod_{l=1}^S \mathbb{P}(g_l | \textrm{pop k})
\label{eqn_assignment}\]</span></p>
<p>We wish to know the probability that this new individual comes from
population <span class="math inline">\(k\)</span>, i.e. <span
class="math inline">\(P(\textrm{pop k} | \textrm{ind.})\)</span>. We can
obtain this through Bayes’ rule</p>
<p><span class="math display">\[\mathbb{P}(\textrm{pop k} |
\textrm{ind.})  = \frac{\mathbb{P}(\textrm{ind.} | \textrm{pop k})
\mathbb{P}(\textrm{pop k})}{\mathbb{P}(\textrm{ind.})}\]</span> where
<span class="math display">\[\mathbb{P}(\textrm{ind.}) =
\sum_{k=1}^K  \mathbb{P}(\textrm{ind.} | \textrm{pop k})
\mathbb{P}(\textrm{pop k})\]</span> is the normalizing constant. We can
interpret <span class="math inline">\(\mathbb{P}(\textrm{pop
k})\)</span> as the prior probability of the individual coming from
population <span class="math inline">\(k\)</span>, and unless we have
some other prior knowledge we will assume that the new individual has a
equal probability of coming from each population <span
class="math inline">\(\mathbb{P}(\textrm{pop
k})=\frac{1}{K}\)</span>.</p>
<p>We interpret <span class="math display">\[\mathbb{P}(\textrm{pop k} |
\textrm{ind.})\]</span> as the posterior probability that our new
individual comes from each of our <span class="math inline">\(1,\cdots,
K\)</span> populations.</p>
<p>More sophisticated versions of this are now used to allow for
hybrids, e.g, we can have a proportion <span
class="math inline">\(q_k\)</span> of our individual’s genome come from
population <span class="math inline">\(k\)</span> and estimate the set
of <span class="math inline">\(q_k\)</span>’s.</p>
<p>Returning to our chimp example, imagine that we have genotyped a set
of individuals from the Western and Eastern populations at two SNPs
(we’ll ignore the central population to keep things simpler). The
frequency of the capital allele at two SNPs (<span
class="math inline">\(A/a\)</span> and <span
class="math inline">\(B/b\)</span>) is given by</p>
<div class="center">
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Population</th>
<th style="text-align: center;">locus A</th>
<th style="text-align: center;">locus B</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">Western</td>
<td style="text-align: center;"><span
class="math inline">\(0.1\)</span></td>
<td style="text-align: center;"><span
class="math inline">\(0.85\)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;">Eastern</td>
<td style="text-align: center;"><span
class="math inline">\(0.95\)</span></td>
<td style="text-align: center;"><span
class="math inline">\(0.2\)</span></td>
</tr>
</tbody>
</table>
</div>
<p><strong>A)</strong> Our individual, whose origin is unknown, has the
genotype <span class="math inline">\(AA\)</span> at the first locus and
<span class="math inline">\(bb\)</span> at the second. What is the
posterior probability that our individual comes from the Western
population versus Eastern chimp population?<br />
<strong>B)</strong> (Trickier) Lets assume that our individual from part
A is a hybrid (not necessarily an F1). At each locus, with probability
<span class="math inline">\(q_W\)</span> our individual draws an allele
from the Western population and with probability <span
class="math inline">\(q_E=1-q_W\)</span> they draw an allele from the
Eastern population. What is the probability of our individual’s genotype
given <span class="math inline">\(q_W\)</span>?<br />
<strong>Optional</strong> You could plot this probability as a function
of <span class="math inline">\(q_W\)</span>. How does your plot change
if our individual is heterozygous at both loci?</p>
<div class="marginfigure">
<div class="center">
<figure>
<img src="Journal_figs/alleles_genotypes/chimp/chimp.png" alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h5 id="clustering-based-on-assignment-methods">Clustering based on
assignment methods</h5>
<p>While it is great to be able to assign our individuals to a
particular population, these ideas can be pushed to learn about how best
to describe our genotype data in terms of discrete populations without
assigning any of our individuals to populations <em>a priori</em>. We
wish to cluster our individuals into <span
class="math inline">\(K\)</span> unknown populations. We begin by
assigning our individuals at random to these <span
class="math inline">\(K\)</span> populations.</p>
<ol type="1">
<li><p>Given these assignments we estimate the allele frequencies at all
of our loci in each population.</p></li>
<li><p>Given these allele frequencies we chose to reassign each
individual to a population <span class="math inline">\(k\)</span> with a
probability given by eqn<a href="#eqn_assignment"
data-reference-type="eqref"
data-reference="eqn_assignment">[eqn_assignment]</a>.</p></li>
</ol>
<p>We iterate steps 1 and 2 for many iterations (technically, this
approach is known as <em>Gibbs Sampling</em>). If the data is
sufficiently informative, the assignments and allele frequencies will
quickly converge on a set of likely population assignments and allele
frequencies for these populations.</p>
<figure>
<img src="figures/Becquet_et_al_STRUCTURE_journal_pgen_0030066_g001.png"
id="fig:chimp_structure"
alt="@becquet:07 genotyped 78 common chimpanzee and 6 bonobo at over 300 polymorphic markers (in this case microsatellites). They ran STRUCTURE to cluster the individuals using these data into K=4 populations. In @becquet:07 above figure they show each individual as a vertical bar divided into four colours depicting the estimate of the fraction of ancestry that each individual draws from each of the four estimated populations (). We can see that these four colours/populations correspond to: Red, central; blue, eastern; green, western; yellow, bonobo." />
<figcaption aria-hidden="true"><span class="citation"
data-cites="becquet:07">@becquet:07</span> genotyped 78 common
chimpanzee and 6 bonobo at over 300 polymorphic markers (in this case
microsatellites). They ran STRUCTURE to cluster the individuals using
these data into <span class="math inline">\(K=4\)</span> populations. In
<span class="citation" data-cites="becquet:07">@becquet:07</span> above
figure they show each individual as a vertical bar divided into four
colours depicting the estimate of the fraction of ancestry that each
individual draws from each of the four estimated populations (). We can
see that these four colours/populations correspond to: Red, central;
blue, eastern; green, western; yellow, bonobo.</figcaption>
</figure>
<p>To do this in a full Bayesian scheme we need to place priors on the
allele frequencies (for example, one could use a beta distribution
prior). Technically we are using the joint posterior of our allele
frequencies and assignments. Programs like STRUCTURE, use this type of
algorithm to cluster the individuals in an “unsupervised” manner (i.e.
they work out how to assign individuals to an unknown set of
populations). See Figure <a href="#fig:chimp_structure"
data-reference-type="ref" data-reference="fig:chimp_structure">3.1</a>
for an example of <span class="citation"
data-cites="becquet:07">@becquet:07</span> using STRUCTURE to determine
the population structure of chimpanzees.</p>
<p>STRUCTURE-like methods have proven incredible popular and useful in
examining population structure within species. However, the results of
these methods are open to misinterpretation; see <span class="citation"
data-cites="lawson:18">@lawson:18</span> for a recent discussion. Two
common mistakes are 1) taking the results of STRUCTURE-like approaches
for some particular value of K and taking this to represent the best way
to describe population-genetic variation. 2) Thinking that these
clusters represent ‘pure’ ancestral populations.</p>
<p>There is no right choice of K, the number of clusters to partition
into. There are methods of judging the ‘best’ K by some statistical
measure given some particular dataset, but that is not the same as
saying this is the most meaningful level on which to summarize
population structure in data. For example, running STRUCTURE on
world-wide human populations for low value of K will result in
population clusters that roughly align with continental populations
<span class="citation" data-cites="rosenberg:02">[@rosenberg:02]</span>.
However, that does not tell us that assigning ancestry at the level of
continents is a particularly meaningful way of partitioning individuals.
Running the same data for higher value of K, or within continental
regions, will result in much finer-scale partitioning of continental
groups <span class="citation"
data-cites="rosenberg:02 li:08">[@rosenberg:02; @li:08]</span>. No one
of these layers of population structure identified is privileged as
being more meaningful than another.</p>
<p>It is tempting to think of these clusters as representing ancestral
populations, which themselves are not the result of admixture. However,
that is not the case, for example, running STRUCTURE on world-wide human
data identifies a cluster that contains many European individuals,
however, on the basis of ancient DNA we know that modern Europeans are a
mixture of distinct ancestral groups.</p>
<h3 id="principal-components-analysis">Principal components
analysis</h3>
<p>Principal component analysis (PCA) is a common statistical approach
to visualize high dimensional data, and used by many fields. The idea of
PCA is to give a location to each individual data-point on each of a
small number principal component axes. These PC axes are chosen to
reflect major axes of variation in the data, with the first PC being
that which explains largest variance, the second the second most, and so
on. The use of PCA in population genetics was pioneered by
Cavalli-Sforza and colleagues and now with large genotyping datasets,
PCA has made a comeback. <span class="citation"
data-cites="menozzi:78 patterson:06">[@menozzi:78;
@patterson:06]</span></p>
<p>Consider a dataset consisting of N individuals at <span
class="math inline">\(S\)</span> biallelic SNPs. The <span
class="math inline">\(i^{th}\)</span> individual’s genotype data at
locus <span class="math inline">\(\ell\)</span> takes a value <span
class="math inline">\(g_{i,\ell}=0,1,\; \text{or} \; 2\)</span>
(corresponding to the number of copies of allele <span
class="math inline">\(A_1\)</span> an individual carries at this SNP).
We can think of this as a <span class="math inline">\(N \times
S\)</span> matrix (where usually <span class="math inline">\(N \ll
S\)</span>).</p>
<p>Denoting the sample mean allele frequency at SNP <span
class="math inline">\(\ell\)</span> by <span
class="math inline">\(p_{\ell}\)</span>, it’s common to standardize the
genotype in the following way <span
class="math display">\[\frac{g_{i,\ell} - 2 p_{\ell}}{\sqrt{2
p_{\ell}(1-p_{\ell})}} \label{eqn:std_allele_freq}\]</span> i.e. at each
SNP we center the genotypes by subtracting the mean genotype (<span
class="math inline">\(2p_{\ell}\)</span>) and divide through by the
square root of the expected variance assuming that alleles are sampled
binomially from the mean frequency (<span class="math inline">\(\sqrt{2
p_{\ell}  (1-p_{\ell})}\)</span>). Doing this to all of our genotypes,
we form a data matrix (of dimension <span class="math inline">\(N \times
S\)</span>). We can then perform principal component analysis of this
data matrix to uncover the major axes of genotype variance in our
sample. Figure <a href="#fig:chimp_PCA" data-reference-type="ref"
data-reference="fig:chimp_PCA">3.2</a> shows a PCA from <span
class="citation" data-cites="becquet:07">@becquet:07</span> using the
same chimpanzee data as in Figure <a href="#fig:chimp_structure"
data-reference-type="ref"
data-reference="fig:chimp_structure">3.1</a>.</p>
<figure>
<img src="figures/Becquet_et_al_STRUCTURE_journal_pgen_0030066_g002.png"
id="fig:chimp_PCA"
alt="Principal Component Analysis by @becquet:07 using the same chimpanzee data as in Figure 3.1. Here @becquet:07 plot the location of each individual on the first two principal components (called eigenvectors) in the left panel, and on the second and third principal components (eigenvectors) in the right panel (). In the PCA, individuals identified as all of one ancestry by STRUCTURE cluster together by population (solid circles). While the nine individuals identified by STRUCTURE as hybrids (open circles) for the most part fall at intermediate locations in the PCA. There are two individuals (red open circles) reported as being of a particular population but that but appear to be hybrids." />
<figcaption aria-hidden="true">Principal Component Analysis by <span
class="citation" data-cites="becquet:07">@becquet:07</span> using the
same chimpanzee data as in Figure <a href="#fig:chimp_structure"
data-reference-type="ref" data-reference="fig:chimp_structure">3.1</a>.
Here <span class="citation" data-cites="becquet:07">@becquet:07</span>
plot the location of each individual on the first two principal
components (called eigenvectors) in the left panel, and on the second
and third principal components (eigenvectors) in the right panel (). In
the PCA, individuals identified as all of one ancestry by STRUCTURE
cluster together by population (solid circles). While the nine
individuals identified by STRUCTURE as hybrids (open circles) for the
most part fall at intermediate locations in the PCA. There are two
individuals (red open circles) reported as being of a particular
population but that but appear to be hybrids.</figcaption>
</figure>
<p>It is worth taking a moment to delve further into what we are doing
here. There’s a number of equivalent ways of thinking about what PCA is
doing. One of these ways is to think that when we do PCA we are building
the individual by individual covariance matrix and performing an
eigenvalue decomposition of this matrix (with the eigenvectors being the
PCs). This individual by individual covariance matrix has entries the
<span class="math inline">\([i,~j]\)</span> given by <span
class="math display">\[\frac{1 }{S-1} \sum_{\ell=1}^S \frac{(g_{i,\ell}
- 2p_{\ell})(g_{j,\ell} -
  2p_{\ell})}{2 p_{\ell}(1-p_{\ell})} \label{eqn:kinship_mat}\]</span>
Note that this is the sample covariance of our standardized allele
frequencies (eqn<a href="#eqn:std_allele_freq"
data-reference-type="eqref"
data-reference="eqn:std_allele_freq">[eqn:std_allele_freq]</a>), and is
very similar to those we encountered in discussing <span
class="math inline">\(F\)</span>-statistics as correlations (eqn <a
href="#eqn:Fascorr" data-reference-type="eqref"
data-reference="eqn:Fascorr">[eqn:Fascorr]</a>), except now we are
asking about the covariance between two individuals above that expected
if they were both drawn from the total sample at random (rather than the
covariance of alleles within a single individual). So by performing PCA
on the data we are learning about the major (orthogonal) axes of the
kinship matrix.</p>
<p>As an example of the application of PCA, let’s consider the case of
the putative ring species in the greenish warbler (<em>Phylloscopus
trochiloides</em>) species complex. This set of subspecies exists in a
ring around the edge of the Himalayan plateau. <span class="citation"
data-cites="alcaide:14">@alcaide:14</span> collected <span
class="math inline">\(95\)</span> greenish warbler samples from <span
class="math inline">\(22\)</span> sites around the ring, and the
sampling locations are shown in Figure <a href="#fig:Gwarbler_geo"
data-reference-type="ref" data-reference="fig:Gwarbler_geo">3.3</a>.</p>
<div class="center">
<figure>
<img src="figures/warbler_PCA_figs/warbler_geo_map.jpg"
id="fig:Gwarbler_geo"
alt="The sampling locations of 22 populations of greenish warblers from @alcaide:14. The samples are coloured by the subspecies." />
<figcaption aria-hidden="true">The sampling locations of 22 populations
of greenish warblers from <span class="citation"
data-cites="alcaide:14">@alcaide:14</span>. The samples are coloured by
the subspecies.</figcaption>
</figure>
</div>
<p>It is thought that these warblers spread from the south, northward in
two different directions around the inhospitable Himalayan plateau,
establishing populations along the western edge (green and blue
populations) and the eastern edge (yellow and red populations). When
they came into secondary contact in Siberia, they were reproductively
isolated from one another, having evolved different songs and
accumulated other reproductive barriers from each other as they spread
independently north around the plateau, such that <em>P. t.
viridanus</em> (blue) and <em>P. t. plumbeitarsus</em> (red) populations
presently form a stable hybrid zone.</p>
<div class="marginfigure">
<figure>
<img
src="illustration_images/alleles_genotypes/greenish_warbler/10036311396_14915d1715_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
<p><span class="citation" data-cites="alcaide:14">@alcaide:14</span>
obtained sequence data for their samples at 2,334 snps. In Figure <a
href="#fig:warbler_heat" data-reference-type="ref"
data-reference="fig:warbler_heat">3.4</a> you can see the matrix of
kinship coefficients, using eqn<a href="#eqn:kinship_mat"
data-reference-type="eqref"
data-reference="eqn:kinship_mat">[eqn:kinship_mat]</a>, between all
pairs of samples. You can already see a lot about population structure
in this matrix. Note how the red and yellow samples, thought to be
derived from the Eastern route around the Himalayas, have higher kinship
with each other, and blue and the (majority) of the green samples, from
the Western route, form a similarly close group in terms of their higher
kinship.</p>
<div class="center">
<figure>
<img src="figures/warbler_PCA_figs/warbler_heatmap.png"
id="fig:warbler_heat"
alt="The matrix of kinship coefficients calcuated for the 95 samples of greenish warblers. Each cell in the matrix gives the pairwise kinship coefficient calculated for a particular pair. Hotter colours indicating higher kinship. The x and y labels of individuals are the population labels from Figure 3.3, and coloured by subspecies label as in that figure. The rows and columns have been organized to cluster individuals with high kinship." />
<figcaption aria-hidden="true">The matrix of kinship coefficients
calcuated for the 95 samples of greenish warblers. Each cell in the
matrix gives the pairwise kinship coefficient calculated for a
particular pair. Hotter colours indicating higher kinship. The <span
class="math inline">\(x\)</span> and <span
class="math inline">\(y\)</span> labels of individuals are the
population labels from Figure <a href="#fig:Gwarbler_geo"
data-reference-type="ref" data-reference="fig:Gwarbler_geo">3.3</a>, and
coloured by subspecies label as in that figure. The rows and columns
have been organized to cluster individuals with high
kinship.</figcaption>
</figure>
</div>
<p>We can then perform PCA on this kinship matrix to identify the major
axes of variation in the dataset. Figure <a href="#fig:warbler_PCA"
data-reference-type="ref" data-reference="fig:warbler_PCA">3.5</a> shows
the samples plotted on the first two PCs.</p>
<div class="center">
<figure>
<img src="figures/warbler_PCA_figs/warbler_PCAmap.jpg"
id="fig:warbler_PCA"
alt="The 95 greenish warbler samples plotted on their locations on the first two principal components. The labels of individuals are the population labels from Figure 3.3, and coloured by subspecies label as in that figure." />
<figcaption aria-hidden="true">The 95 greenish warbler samples plotted
on their locations on the first two principal components. The labels of
individuals are the population labels from Figure <a
href="#fig:Gwarbler_geo" data-reference-type="ref"
data-reference="fig:Gwarbler_geo">3.3</a>, and coloured by subspecies
label as in that figure.</figcaption>
</figure>
</div>
<p>The two major routes of expansion clearly occupy different parts of
PC space. The first principal component distinguishes populations
running North to South along the western route of expansion, while the
second principal component distinguishes among populations running North
to South along the Eastern route of expansion. Thus genetic data
supports the hypothesis that the greenish warblers speciated as they
moved around the Himalayan plateau. However, as noted by <span
class="citation" data-cites="alcaide:14">@alcaide:14</span>, it also
suggests additional complications to the traditional view of these
warblers as an unbroken ring species, a case of speciation by continuous
geographic isolation. The <em>Ludlowi</em> subspecies shows a
significant genetic break, with the southern most MN samples clustering
with the <em>Trochiloides</em> subspecies, in both the PCA and kinship
matrix (Figures <a href="#fig:warbler_PCA" data-reference-type="ref"
data-reference="fig:warbler_PCA">3.5</a> and <a href="#fig:warbler_heat"
data-reference-type="ref" data-reference="fig:warbler_heat">3.4</a>),
despite being much more geographically close to the other
<em>Ludlowi</em> samples. This suggests that genetic isolation is not
just a result of geographic distance, and other biogeographic barriers
must be considered in the case of this broken ring species.</p>
<p>Finally, while PCA is a wonderful tool for visualizing genetic data,
care must be taken in its interpretation. The U-like shape in the case
of the greenish warbler PC might be consistent with some low level of
gene flow between the red and the blue populations, pulling them
genetically closer together and helping to form a genetic ring as well
as a geographic ring. However, U-like shapes are expected to appear in
PCAs even if our populations are just arrayed along a line, and more
complex geometric arrangements of populations in PC space can result
under simple geographic models <span class="citation"
data-cites="novembre:08">[@novembre:08]</span>. Inferring the
geographical and population-genetic history of species requires the
application of a range of tools; see <span class="citation"
data-cites="alcaide:14">@alcaide:14</span> and <span class="citation"
data-cites="bradburd:16">@bradburd:16</span> for more discussion of the
greenish warblers.</p>
<h3
id="correlations-between-loci-linkage-disequilibrium-and-recombination">Correlations
between loci, linkage disequilibrium, and recombination</h3>
<p>Up to now we have been interested in correlations between alleles at
the same locus, e.g. correlations within individuals (inbreeding) or
between individuals (relatedness). We have seen how relatedness between
parents affects the extent to which their offspring is inbred. We now
turn to correlations between alleles at different loci.<br />
</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/LD_decay/recom_cartoon.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h5 id="recombination">Recombination</h5>
<p>To understand correlations between loci we need to understand
recombination a bit more carefully. Let us consider a heterozygous
individual, containing <span class="math inline">\(AB\)</span> and <span
class="math inline">\(ab\)</span> haplotypes. If no recombination occurs
between our two loci in this individual, then these two haplotypes will
be transmitted intact to the next generation. While if a recombination
(i.e. an odd number of crossing over events) occurs between the two
parental haplotypes, then <span
class="math inline">\(\frac{1}{2}\)</span> the time the child receives
an <span class="math inline">\(Ab\)</span> haplotype and <span
class="math inline">\(\frac{1}{2}\)</span> the time the child receives
an <span class="math inline">\(aB\)</span> haplotype. See Figure <a
href="#fig:recom_cartoon" data-reference-type="ref"
data-reference="fig:recom_cartoon">[fig:recom_cartoon]</a>. Effectively,
recombination breaks up the association between loci. For linked markers
we’ll define the recombination fraction (<span
class="math inline">\(x\)</span>) to be the probability of an odd number
of crossing over events between our loci in a single meiosis. The
recombination fraction between a pair of loci can range from <span
class="math inline">\(0\)</span> to <span
class="math inline">\(\frac{1}{2}\)</span>, with <span
class="math inline">\(c=\frac{1}{2}\)</span> corresponding markers far
enough apart on a chromosome that many recombination events occur
between them (loci on different automosomes also have a <span
class="math inline">\(c=\frac{1}{2}\)</span>). In practice we’ll often
be interested in relatively short regions such that recombination is
relatively rare, and so we might think that <span
class="math inline">\(c=c_{BP}L \ll \frac{1}{2}\)</span>, where <span
class="math inline">\(c_{BP}\)</span> is the average recombination rate
(in Morgans) per base pair (typically <span class="math inline">\(\sim
10^{-8}\)</span> ) and L is the number of base pairs separating our two
loci.<br />
</p>
<h5 id="linkage-disequilibrium">Linkage disequilibrium</h5>
<p>The (horrible) phrase linkage disequilibrium (LD) refers to the
statistical non-independence (i.e. a correlation) of alleles in a
population at different loci <span class="citation"
data-cites="lewontin1960evolutionary slatkin2008linkage">[@lewontin1960evolutionary;
@slatkin2008linkage]</span>. It’s a fantastically useful concept; LD is
key to our understanding of diverse topics, from sexual selection and
speciation to the limits of genome-wide association studies.</p>
<p>Our two biallelic loci, which segregate alleles <span
class="math inline">\(A/a\)</span> and <span
class="math inline">\(B/b\)</span>, have allele frequencies of <span
class="math inline">\(p_A\)</span> and <span
class="math inline">\(p_B\)</span> respectively. The frequency of the
two locus haplotype AB is <span class="math inline">\(p_{AB}\)</span>,
and likewise for our other three combinations. If our loci were
statistically independent then <span class="math inline">\(p_{AB} =
p_Ap_B\)</span>, otherwise <span class="math inline">\(p_{AB} \neq
p_Ap_B\)</span> We can define a covariance between the <span
class="math inline">\(A\)</span> and <span
class="math inline">\(B\)</span> alleles at our two loci as <span
class="math display">\[D_{AB} = p_{AB} -
p_Ap_B  \label{eqn:LD_def}\]</span> and likewise for our other
combinations at our two loci (<span
class="math inline">\(D_{Ab},~D_{aB},~D_{ab}\)</span>). Gametes with two
similar case alleles (e.g. A and B, or a and b) are known as
<em>coupling</em> gametes, and those with different case alleles are
known as <em>repulsion</em> gametes (e.g. a and B, or A and b). Then, we
can think of <span class="math inline">\(D\)</span> as measuring the
<em>excess</em> of coupling to repulsion gametes. These <span
class="math inline">\(D\)</span> statistics are all closely related to
each other as <span class="math inline">\(D_{AB} = - D_{Ab}\)</span> and
so on. Thus we only need to specify one <span
class="math inline">\(D_{AB}\)</span> to know them all, so we’ll drop
the subscript and just refer to <span class="math inline">\(D\)</span>.
Also a handy result is that we can rewrite our haplotype frequency <span
class="math inline">\(p_{AB}\)</span> as <span
class="math display">\[p_{AB} = p_Ap_B+D. \label{eqn:ABviaD}\]</span> If
<span class="math inline">\(D=0\)</span> we’ll say the two loci are in
linkage equilibrium, while if <span
class="math inline">\(D&gt;0\)</span> or <span
class="math inline">\(D&lt;0\)</span> we’ll say that the loci are in
linkage disequilibrium (we’ll perhaps want to test whether <span
class="math inline">\(D\)</span> is statistically different from <span
class="math inline">\(0\)</span> before making this choice). Linkage
disequilibrium is a horrible phrase, as it risks muddling the concepts
of genetic linkage and linkage disequilibrium. Genetic linkage refers to
the linkage of multiple loci due to the fact that they are transmitted
through meiosis together (most often because the loci are on the same
chromosome). Linkage disequilibrium merely refers to the covariance
between the alleles at different loci; this may in part be due to the
genetic linkage of these loci but does not necessarily imply this
(e.g. genetically unlinked loci can be in LD due to population
structure).<br />
You genotype 2 bi-allelic loci (A &amp; B) segregating in two mouse
subspecies (1 &amp; 2) which mate randomly among themselves, but have
not historically interbreed since they speciated. The frequencies of
haplotypes in each population are:</p>
<div class="center">
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Pop</th>
<th style="text-align: center;"><span
class="math inline">\(p_{AB}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(p_{Ab}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(p_{aB}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(p_{ab}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">1</td>
<td style="text-align: center;">.02</td>
<td style="text-align: center;">.18</td>
<td style="text-align: center;">.08</td>
<td style="text-align: center;">.72</td>
</tr>
<tr class="even">
<td style="text-align: center;">2</td>
<td style="text-align: center;">.72</td>
<td style="text-align: center;">.18</td>
<td style="text-align: center;">.08</td>
<td style="text-align: center;">.02</td>
</tr>
</tbody>
</table>
</div>
<p><strong>A)</strong> How much LD is there within species?
(i.e. estimate D)<br />
<strong>B)</strong> If we mixed individuals from the two species
together in equal proportions, we could form a new population with <span
class="math inline">\(p_{AB}\)</span> equal to the average frequency of
<span class="math inline">\(p_{AB}\)</span> across species 1 and 2. What
value would D take in this new population before any mating has had the
chance to occur?<br />
</p>
<div class="center">
<figure>
<img src="Journal_figs/alleles_genotypes/TAPS_hotspot/Taps_hotspot.png"
id="fig:TAPS_hotspot"
alt="LD across the TAP2 gene region in a sample of Humans and Chimps, from @ptak2004absence, . The rows and columns are consecutive SNPs, with each cell giving the absolute D^{\prime} value between a pair of SNPs. Note that these are different sets of SNPs in the two species, as shared polymorphisms are very rare." />
<figcaption aria-hidden="true">LD across the TAP2 gene region in a
sample of Humans and Chimps, from <span class="citation"
data-cites="ptak2004absence">@ptak2004absence</span>, . The rows and
columns are consecutive SNPs, with each cell giving the absolute <span
class="math inline">\(D^{\prime}\)</span> value between a pair of SNPs.
Note that these are different sets of SNPs in the two species, as shared
polymorphisms are very rare.</figcaption>
</figure>
</div>
<p>Our linkage disequilibrium statistic <span
class="math inline">\(D\)</span> depends strongly on the allele
frequencies of the two loci involved. One common way to partially remove
this dependence, and make it more comparable across loci, is to divide
<span class="math inline">\(D\)</span> through by its maximum possible
value given the frequency of the loci. This normalized statistic is
called <span class="math inline">\(D^{\prime}\)</span> and varies
between <span class="math inline">\(+1\)</span> and <span
class="math inline">\(-1\)</span>. In Figure <a href="#fig:TAPS_hotspot"
data-reference-type="ref" data-reference="fig:TAPS_hotspot">3.6</a>
there’s an example of LD across the TAP2 region in human and chimp.
Notice how physically close SNPs, i.e. those close to the diagonal, have
higher absolute values of <span
class="math inline">\(D^{\prime}\)</span> as closely linked alleles are
separated by recombination less often allowing high levels of LD to
accumulate. Over large physical distances, away from the diagonal, there
is lower <span class="math inline">\(D^{\prime}\)</span>. This is
especially notable in humans as there is an intense, human-specific
recombination hotspot in this region, which is breaking down LD between
opposite sides of this region.</p>
<p>Another common statistic for summarizing LD is <span
class="math inline">\(r^2\)</span> which we write as <span
class="math display">\[r^2 = \frac{D^2}{p_A(1-p_A) p_B(1-p_B) }\]</span>
As <span class="math inline">\(D\)</span> is a covariance, and <span
class="math inline">\(p_A(1-p_A)\)</span> is the variance of an allele
drawn at random from locus <span class="math inline">\(A\)</span>, <span
class="math inline">\(r^2\)</span> is the squared correlation
coefficient.<br />
fraction.<br />
</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/Mus_musculus/20746324002_e014b4fcc6_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Figure <a href="#fig:Mouse_LD" data-reference-type="ref"
data-reference="fig:Mouse_LD">3.7</a> shows <span
class="math inline">\(r^2\)</span> for pairs of SNPs at various physical
distances in two population samples of <em>Mus musculus domesticus</em>.
Again LD is highest between physically close markers as LD is being
generated faster than it can decay via recombination; more distant
markers have much lower LD as here recombination is winning out. Note
the decay of LD is much slower in the advanced-generation cross
population than in the natural wild-caught population. This persistence
of LD across megabases is due to the limited number of generations for
recombination since the cross was created.</p>
<div class="center">
<figure>
<img
src="Journal_figs/alleles_genotypes/mouse_LD/mouse_LD_Laurie_et_al.png"
id="fig:Mouse_LD"
alt="The decay of LD for autosomal SNPin Mus musculus domesticus, as measured by r^2, in a wild-caught mouse population from Arizona and a set of advanced-generation crosses between inbred lines of lab mice. Each dot gives the r^2 for a pair of SNPs a given physical distance apart, for a total of \sim  3000 SNPs. The solid black line gives the mean, the jagged red line the 95^{th} percentile, and the flat red line a cutoff for significant LD. From @Laurie:07, ." />
<figcaption aria-hidden="true">The decay of LD for autosomal SNPin
<em>Mus musculus domesticus</em>, as measured by <span
class="math inline">\(r^2\)</span>, in a wild-caught mouse population
from Arizona and a set of advanced-generation crosses between inbred
lines of lab mice. Each dot gives the <span
class="math inline">\(r^2\)</span> for a pair of SNPs a given physical
distance apart, for a total of <span
class="math inline">\(\sim  3000\)</span> SNPs. The solid black line
gives the mean, the jagged red line the <span
class="math inline">\(95^{th}\)</span> percentile, and the flat red line
a cutoff for significant <span class="math inline">\(LD\)</span>. From
<span class="citation" data-cites="Laurie:07">@Laurie:07</span>,
.</figcaption>
</figure>
</div>
<h5 id="the-generation-of-ld.">The generation of LD.</h5>
<p>Various population genetic forces can generate LD <span
class="citation"
data-cites="slatkin2008linkage">[@slatkin2008linkage]</span>. Selection
can generate LD by favouring particular combinations of alleles. Genetic
drift will also generate LD, not because particular combinations of
alleles are favoured, but simply because at random particular haplotypes
can by chance drift up in frequency. Mixing between divergent
populations can also generate LD <span class="citation"
data-cites="nei1973linkage">[@nei1973linkage]</span>, as we saw in the
mouse question above.</p>
<h5 id="the-decay-of-ld-due-to-recombination">The decay of LD due to
recombination</h5>
<p>We will now examine what happens to LD over the generations if, in a
very large population (i.e. no genetic drift and frequencies of our loci
thus follow their expectations), we only allow recombination to occur.
To do so, consider the frequency of our <span
class="math inline">\(AB\)</span> haplotype in the next generation,
<span class="math inline">\(p_{AB}^{\prime}\)</span>. We lose a fraction
<span class="math inline">\(c\)</span> of our <span
class="math inline">\(AB\)</span> haplotypes to recombination ripping
our alleles apart but gain a fraction <span class="math inline">\(cp_A
p_B\)</span> per generation from other haplotypes recombining together
to form <span class="math inline">\(AB\)</span> haplotypes. Thus in the
next generation <span class="math display">\[p_{AB}^{\prime} =
(1-c)p_{AB} + cp_Ap_B \label{new_hap_freq}\]</span> The last term above,
in eqn<a href="#new_hap_freq" data-reference-type="ref"
data-reference="new_hap_freq">[new_hap_freq]</a>, is <span
class="math inline">\(c(p_{AB}+p_{Ab})(p_{AB}+p_{aB})\)</span>
simplified, which is the probability of recombination in the different
diploid genotypes that could generate a <span
class="math inline">\(p_{AB}\)</span> haplotype.<br />
We can then write the change in the frequency of the <span
class="math inline">\(p_{AB}\)</span> haplotype as <span
class="math display">\[\Delta p_{AB} = p_{AB}^{\prime} -p_{AB} = -c
p_{AB} + cp_Ap_B = - c D\]</span></p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/LD_decay/LD_decay_time.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/LD_decay/LD_decay_recom.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>So recombination will cause a decrease in the frequency of <span
class="math inline">\(p_{AB}\)</span> if there is an excess of <span
class="math inline">\(AB\)</span> haplotypes within the population
(<span class="math inline">\(D&gt;0\)</span>), and an increase if there
is a deficit of <span class="math inline">\(AB\)</span> haplotypes
within the population (<span class="math inline">\(D&lt;0\)</span>). Our
LD in the next generation is <span
class="math display">\[\begin{aligned}
D^{\prime} &amp; = p_{AB}^{\prime} - p&#39;_{A}p&#39;_{B} \nonumber\\
&amp; = (p_{AB} + \Delta p_{AB}) - (p_{A} + \Delta p_{A})(p_{B} + \Delta
p_{B}) \nonumber\\
&amp; = p_{AB} + \Delta p_{AB} - p_{A}p_{B} \nonumber\\
&amp; = (1-c) D
\end{aligned}\]</span> where we can cancel out <span
class="math inline">\(\Delta p_{A}\)</span> and <span
class="math inline">\(\Delta p_{B}\)</span> above because recombination
only changes haplotype, not allele, frequencies. So if the level of LD
in generation <span class="math inline">\(0\)</span> is <span
class="math inline">\(D_0\)</span>, the level <span
class="math inline">\(t\)</span> generations later (<span
class="math inline">\(D_t\)</span>) is <span
class="math display">\[D_t=  (1-c)^t D_0\]</span> Recombination is
acting to decrease LD, and it does so geometrically at a rate given by
<span class="math inline">\((1-c)\)</span> <span class="citation"
data-cites="weinberg1909vererbungsgesetze jennings1917numerical">[@weinberg1909vererbungsgesetze;
@jennings1917numerical]</span>. If <span class="math inline">\(c \ll
1\)</span> then we can approximate this by an exponential and say that
<span class="math display">\[D_t \approx  D_0
e^{-ct}  \label{eqn_LD_decay}\]</span><br />
which follows from a Taylor series expansion, see Appendix eqn<a
href="#eqn:Taylor_geo" data-reference-type="eqref"
data-reference="eqn:Taylor_geo">[eqn:Taylor_geo]</a>.
<strong>Question</strong> You find a hybrid population between the two
mouse subspecies described in the question above, which appears to be
comprised of equal proportions (<span
class="math inline">\(50/50\)</span>) of ancestry from the two
subspecies. You estimate LD between the two markers to be <span
class="math inline">\(D=0.0723\)</span>. On the basis of previous work
you estimate that the two loci are separated by a recombination fraction
of 0.1. Assuming that this hybrid population is large and was formed by
a single mixture event, can you estimate how long ago this population
formed?<br />
</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/alleles_genotypes/Neanderthal/14800262343_f21929987a_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>A particularly striking example of the decay of LD generated by the
mixing of populations is offered by the LD created by the interbreeding
between humans and Neanderthals <span class="citation"
data-cites="Sankararaman:12">[@Sankararaman:12]</span>. Neanderthals and
modern humans diverged from each other likely over half a million years
ago, allowing time for allele frequency differences to accumulate
between the Neanderthal and modern human populations. The two
populations spread back into secondary contact when humans moved out of
Africa over the past hundred thousand years or so. One of the most
exciting findings from the sequencing of the Neanderthal genome was that
modern-day people with Eurasian ancestry carry a few percent of their
genome derived from the Neanderthal genome, via interbreeding during
this secondary contact <span class="citation"
data-cites="green2010draft">[@green2010draft]</span>. To date the timing
of this interbreeding, <span class="citation"
data-cites="Sankararaman:12">@Sankararaman:12</span> looked at the LD in
modern humans between pairs of alleles found to be derived from the
Neanderthal genome (and nearly absent from African populations). In
Figure <a href="#fig:LD_Neanderthal" data-reference-type="ref"
data-reference="fig:LD_Neanderthal">3.8</a> we show the average LD
between these loci as a function of the genetic distance (<span
class="math inline">\(c\)</span>) between them, from the work of <span
class="citation"
data-cites="Sankararaman:12">@Sankararaman:12</span>.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/alleles_genotypes/Neanderthal_LD/European_neanderthal_LD.pdf"
id="fig:LD_Neanderthal" />
<figcaption aria-hidden="true">The LD between putative-Neanderthal
alleles in a modern European population (the CEU sample from the 1000
Genomes Project). Each point represents the average D statistic between
a pair of alleles at loci at a given genetic distance apart (as given on
the x-axis and measured in centiMorgans (cM)). The putative Neanderthal
alleles are alleles where the Neanderthal genome has a derived allele
that is at very low frequency in a modern-human West African population
sample (thought to have little admixture from Neanderthals). The red
line is the fit of an exponential decay of LD, using non-linear least
squared (nls in R).</figcaption>
</figure>
</div>
<p>Assuming a recombination rate <span class="math inline">\(r\)</span>,
we can fit the exponential decay of LD predicted by eqn<a
href="#eqn_LD_decay" data-reference-type="eqref"
data-reference="eqn_LD_decay">[eqn_LD_decay]</a> to the data points in
this figure; the fit is shown as a red line. Doing this we estimate
<span class="math inline">\(t=1200\)</span> generations, or about 35
thousand years (using a human generation time of 29 years). Thus the LD
in modern Eurasians, between alleles derived from the interbreeding with
Neanderthals, represents over thirty thousand years of recombination
slowly breaking down these old associations.</p>
<div class="ChapterSummary">
<p>Individuals often mate non-randomly, e.g. by geographical location,
this generates population genetic structure that can be thought of as a
form of inbreeding. This inbreeding at a population level leads to a
reduction in heterozygosity within sub-populations as compared to the
total population (if allele frequencies differ across populations).</p>
<p>Wright’s <span class="math inline">\(F\)</span> statistics can be
used to measure the extent of population structure, describing the
reduction in heterozygosity at various scales, for example the
individual compared to the sub-population (<span
class="math inline">\(F_{IS}\)</span>) or the sub-population compared to
the total population (<span class="math inline">\(F_{ST}\)</span>). We
can calculate these statistics either genome-wide or at individual
loci.</p>
<p>These <span class="math inline">\(F\)</span> statistics can be
understood as expressing a correlation between alleles drawn from the
same level of population structure, or the proportion of genetic
variance explained by population structure.</p>
<p>Other ways to visualize population structure include STRUCTURE-like
approaches, which are based on assigning individuals to populations
based on the likelihood of their genotype given allele frequencies
(assignment methods) and learning the assignment of individuals to
discrete populations. Another common approach relies on identifying
major axes of variation in relatedness via Principal components
analysis.</p>
<p>We’ll often be interested in covariances and correlations among
alleles at different loci, linkage disequilibrium (LD).</p>
<p>Covariance between loci (LD) can arise between loci for a variety of
reasons, notably population structure and admixture as described in the
chapter.</p>
<p>The decay of LD due recombination can be modelled and potentially
used to date when LD was generated (e.g. via admixture).</p>
</div>
<p>The loss of heterozygosity due to inbreeding can be partitioned
across F statistics at multiple levels. For example we can partition the
total inbreeding coefficient of a individuals (<span
class="math inline">\(F_{IT}\)</span>) compared to a population between
<span class="math inline">\(F_{IS}\)</span> and <span
class="math inline">\(F_{ST}\)</span>. For the following example
scenarios, do you expect <span class="math inline">\(F_{IS}\)</span> to
be larger or smaller than <span class="math inline">\(F_{ST}\)</span>?
Explain your answer.<br />
<strong>A)</strong> Charles II, where the subpopulation is Spain and the
total population is Europeans.<br />
<strong>B)</strong> Subpopulations of plants living on a moutainside,
where pollen disperses long distances via wind, butindividuals
self-pollinate about 50% of the time,<br />
<strong>C)</strong> Fish that live in lakes with very few accessible
waterways between lakes, but where the fish swim freely within lakes.
Each lake is a subpopulation and the entire lake basin is the total
population.</p>
<p>In a species of beetle, the colour and shape of the wings are
controlled by two distinct polymorphisms (with alleles big/small and
red/yellow respectively). In a museum collection you estimate the
frequency of the four haplotypes to be:<br />
</p>
<div class="center">
<table>
<tbody>
<tr class="odd">
<td style="text-align: center;">big/red</td>
<td style="text-align: center;">big/yellow</td>
<td style="text-align: center;">small/red</td>
<td style="text-align: center;">small/yellow</td>
</tr>
<tr class="even">
<td style="text-align: center;">0.69</td>
<td style="text-align: center;">0.00</td>
<td style="text-align: center;">0.09</td>
<td style="text-align: center;">0.22</td>
</tr>
</tbody>
</table>
</div>
<p>This collection is from 60 years ago. In present day populations you
estimate the frequencies of the haplotypes to be:<br />
</p>
<div class="center">
<table>
<tbody>
<tr class="odd">
<td style="text-align: center;">0.5452</td>
<td style="text-align: center;">0.1448</td>
<td style="text-align: center;">0.2348</td>
<td style="text-align: center;">0.0752</td>
</tr>
</tbody>
</table>
</div>
<p><strong>A)</strong> Assuming one generation per year, what is the
recombination fraction between these loci?<br />
<strong>B)</strong> Qualitatively, how would your answer change if you
determined that crossing over only occurred in females and not in
males?</p>
<h1 id="Chapter:Drift">Genetic Drift and Neutral Diversity.</h1>
<p>, from the lucky birds blown off course to colonize some new oceanic
island, to which mutations arise first in the HIV strain infecting an
individual taking anti-retroviral drugs. One major source of
stochasticity in evolutionary biology is genetic drift. Genetic drift
occurs because more or less copies of an allele by chance can be
transmitted to the next generation <span class="citation"
data-cites="wright1931evolution">[@wright1931evolution]</span>. This can
occur because, by chance, the individuals carrying a particular allele
can leave more or less offspring in the next generation. In a sexual
population, genetic drift also occurs because Mendelian transmission
means that only one of the two alleles in an individual, chosen at
random at a locus, is transmitted to the offspring.</p>
<p>Genetic drift can play a role in the dynamics of all alleles in all
populations, but it will play the biggest role for neutral alleles. A
neutral polymorphism occurs when the segregating alleles at a
polymorphic site have no discernible differences in their effect on
fitness. We’ll make clear what we mean by "discernible" later, but for
the moment think of this as "no effect" on fitness.</p>
<h5 id="the-neutral-theory-of-molecular-evolution.">The neutral theory
of molecular evolution.</h5>
<p>The role of genetic drift in molecular evolution has been hotly
debated since the 60s when the Neutral theory of molecular evolution was
proposed [see <span class="citation"
data-cites="ohta1996development">@ohta1996development</span> for a
history]<span class="citation"
data-cites="kimura:68 king:69 kimura:83">[@kimura:68; @king:69;
@kimura:83]</span>. The central premise of Neutral theory is that
patterns of molecular polymorphism within species and substitution
between species can be well understood by supposing that the vast
majority of these molecular polymorphisms and substitutions were neutral
alleles, whose dynamics were just subject to the vagaries of genetic
drift and mutation. Early proponents of this view suggested that the
vast majority of new mutations are either neutral or highly deleterious
(e.g. mutations that disrupt important protein functions). This latter
class of mutations are too deleterious to contribute much to common
polymorphisms or substitutions between species, because they are quickly
weeded out of the population by selection.</p>
<p>Neutral theory can sound strange given that much of the time our
first brush with evolution often focuses on adaptation and phenotypic
evolution. However, proponents of this world-view didn’t deny the
existence of advantageous mutations, they simply thought that beneficial
mutations are rare enough that their contribution to the bulk of
polymorphism or divergence can be largely ignored. They also often
thought that much of phenotypic evolution may well be adaptive, but
again the loci responsible for these phenotypes are a small fraction of
all the molecular change that occur. The neutral theory of molecular
evolution was originally proposed to explain protein polymorphism.
However, we can apply it more broadly to think about neutral evolution
genome-wide. With that in mind, what types of molecular changes could be
neutral? Perhaps:</p>
<ol type="1">
<li><p>Changes in non-coding DNA that don’t disrupt regulatory
sequences. For example, in the human genome only about 2% of the genome
codes for proteins. The rest is mostly made up of old transposable
element and retrovirus insertions, repeats, pseudo-genes, and general
genomic clutter. Current estimates suggest that, even counting
conserved, functional, non-coding regions, less than 10% of our genome
is subject to evolutionary constraint <span class="citation"
data-cites="rands:14">[@rands:14]</span>.</p></li>
<li><p>Synonymous changes in coding regions, i.e. those that don’t
change the amino-acid encoded by a codon.</p></li>
<li><p>Non-synonymous changes that don’t have a strong effect on the
functional properties of the amino acid encoded, e.g. changes that don’t
change the size, charge, or hydrophobic properties of the amino acid too
much.</p></li>
<li><p>An amino-acid change with phenotypic consequences, but little
relevance to fitness, e.g. a mutation that causes your ears to be a
slightly different shape, or that prevents an organism from living past
50 in a species where most individuals reproduce and die by their
20s.</p></li>
</ol>
<p>There are counter examples to all of these ideas, e.g. synonymous
changes can affect the translation speed and accuracy of proteins and so
are subject to selection. However, the list above hopefully convinces
you that the general thinking that some portion of molecular change may
not be subject to selection isn’t as daft as it may have initially
sounded.</p>
<p>Various features of molecular polymorphism and divergence have been
viewed as consistent with the neutral theory of molecular evolution. In
this chapter we’ll focus on the prediction of a high level of molecular
polymorphism in many species (see for example Figure <a
href="#fig:Leffer" data-reference-type="ref"
data-reference="fig:Leffer">[fig:Leffer]</a>). In a subsequent chapter
we’ll talk about the prediction of a molecular clock. We’ll see that
various aspects of the original neutral theory have merit in describing
some features and types of molecular change, but we’ll also see that it
is demonstrably wrong in some cases. We’ll also see the primary utility
of the neutral theory isn’t whether it is right or wrong, but that it
serves as a simple null model that can be tested and in some cases
rejected, and subsequently built on. The broader debate currently in the
field of molecular evolution is the balance of neutral, adaptive, and
deleterious changes that drive different types of evolutionary
change.</p>
<h2 id="LossofHet">Loss of heterozygosity due to drift.</h2>
<p>Genetic drift will, in the absence of new mutations, slowly purge our
population of neutral genetic diversity, as alleles slowly drift to high
or low frequencies and are lost or fixed over time.<br />
Imagine a randomly mating population of a constant size <span
class="math inline">\(N\)</span> diploid individuals, and that we are
examining a locus segregating for two alleles that are neutral with
respect to each other. This population is randomly mating with respect
to the alleles at this locus. See Figures <a
href="#fig:LossHet_two_alleles" data-reference-type="ref"
data-reference="fig:LossHet_two_alleles">4.1</a> and <a
href="#fig:LossHet_many_alleles" data-reference-type="ref"
data-reference="fig:LossHet_many_alleles">4.2</a> to see how genetic
drift proceeds, by tracking alleles within a small population.<br />
In generation <span class="math inline">\(t\)</span> our current level
of heterozygosity is <span class="math inline">\(H_t\)</span>, i.e. the
probability that two randomly sampled alleles in generation <span
class="math inline">\(t\)</span> are non-identical is <span
class="math inline">\(H_t\)</span>. Assuming that the mutation rate is
zero (or vanishingly small), what is our level of heterozygosity in
generation <span class="math inline">\(t+1\)</span>?<br />
</p>
<div class="center">
<figure>
<img src="figures/Loss_of_he_col_two_alleles.png"
id="fig:LossHet_two_alleles"
alt="Loss of heterozygosity over time, in the absence of new mutations. A diploid population of 5 individuals over the generations, with lines showing transmission. In the first generation every individual is a heterozygote." />
<figcaption aria-hidden="true">Loss of heterozygosity over time, in the
absence of new mutations. A diploid population of 5 individuals over the
generations, with lines showing transmission. In the first generation
every individual is a heterozygote.</figcaption>
</figure>
</div>
<div class="center">
<figure>
<img src="figures/Loss_of_het_2_many_alleles.png"
id="fig:LossHet_many_alleles"
alt="Loss of heterozygosity over time, in the absence of new mutations. A diploid population of 5 individuals. In the first generation I colour every allele a different colour so we can track their descendants." />
<figcaption aria-hidden="true">Loss of heterozygosity over time, in the
absence of new mutations. A diploid population of 5 individuals. In the
first generation I colour every allele a different colour so we can
track their descendants.</figcaption>
</figure>
</div>
<p>In the next generation (<span class="math inline">\(t+1\)</span>) we
are looking at the alleles in the offspring of generation <span
class="math inline">\(t\)</span>. If we randomly sample two alleles in
generation <span class="math inline">\(t+1\)</span> which had different
parental alleles in generation <span class="math inline">\(t\)</span>,
that is just like drawing two random alleles from generation <span
class="math inline">\(t\)</span>. So the probability that these two
alleles in generation <span class="math inline">\(t+1\)</span>, that
have different parental alleles in generation <span
class="math inline">\(t\)</span>, are non-identical is <span
class="math inline">\(H_t\)</span>.<br />
Conversely, if the two alleles in our pair had the same parental allele
in the proceeding generation (i.e. the alleles are identical by descent
one generation back) then these two alleles must be identical (as we are
not allowing for any mutation).<br />
In a diploid population of size <span class="math inline">\(N\)</span>
individuals there are <span class="math inline">\(2N\)</span> alleles.
The probability that our two alleles have the same parental allele in
the proceeding generation is <span
class="math inline">\(\frac{1}{(2N)}\)</span> and the probability that
they have different parental alleles is is <span
class="math inline">\(1-\frac{1}{(2N)}\)</span>. So by the above
argument, the expected heterozygosity in generation <span
class="math inline">\(t+1\)</span> is <span
class="math display">\[H_{t+1} = \frac{1}{2N} \times 0 +
\left(1-\frac{1}{2N} \right)H_t\]</span> Thus, if the heterozygosity in
generation <span class="math inline">\(0\)</span> is <span
class="math inline">\(H_0\)</span>, our expected heterozygosity in
generation <span class="math inline">\(t\)</span> is <span
class="math display">\[H_t = \left(1-\frac{1}{2N}
\right)^tH_0  \label{eqn:loss_het_discrete}\]</span></p>
<p>i.e. the expected heterozygosity within our population is decaying
geometrically with each passing generation. If we assume that <span
class="math inline">\(\frac{1}{(2N)} \ll 1\)</span> then we can
approximate this geometric decay by an exponential decay (see Question
<a href="#geoquestion" data-reference-type="ref"
data-reference="geoquestion">[geoquestion]</a> below), such that <span
class="math display">\[H_t =H_{0} e^{ - \frac{t}{(2N)} }\]</span>
i.e. heterozygosity decays exponentially at a rate <span
class="math inline">\(\frac{1}{(2N)}\)</span>.</p>
<p>In Figure <a href="#fig:LossHet_WF_N50" data-reference-type="ref"
data-reference="fig:LossHet_WF_N50">4.3</a> we show trajectories through
time for 40 independently simulated loci drifting in a population of 50
individuals. Each population was started from a frequency of <span
class="math inline">\(30\%\)</span>. Some drift up and some drift down,
eventually being lost or fixed from the population, but, on average
across simulations, the allele frequency doesn’t change. We also track
heterozygosity, you can see that heterozygosity sometimes goes up, and
sometimes goes down, but on average we are losing heterozygosity, and
this rate of loss is well predicted by eqn. <a
href="#eqn:loss_het_discrete" data-reference-type="eqref"
data-reference="eqn:loss_het_discrete">[eqn:loss_het_discrete]</a>.</p>
<div class="center">
<figure>
<embed src="figures/WF_loss_het/WF_loss_het_N50.pdf"
id="fig:LossHet_WF_N50" />
<figcaption aria-hidden="true">Change in allele frequency and loss of
heterozygosity over time for 40 replicates. Simulations of genetic drift
in a diploid population of 50 individuals, in the absence of new
mutations. We start 40 independent, biallelic loci each with an initial
allele at 30% frequency. The left panel shows the allele frequency over
time and the right panel shows the heterozygosity over time, with the
mean decay matching eqn. <a href="#eqn:loss_het_discrete"
data-reference-type="eqref"
data-reference="eqn:loss_het_discrete">[eqn:loss_het_discrete]</a>.</figcaption>
</figure>
</div>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Genetic_drift/smelt/20497452375_9be855d9ff_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p><strong>Question</strong> You are in charge of maintaining a
population of delta smelt in the Sacramento River delta. Using a large
set of microsatellites you estimate that the mean level of
heterozygosity in this population is 0.005. You set yourself a goal of
maintaining a level of heterozygosity of at least 0.0049 for the next
two hundred years. Assuming that the smelt have a generation time of 3
years, and that only genetic drift affects these loci, what is the
smallest fully outbreeding population that you would need to maintain to
meet this goal?</p>
<p>Note how this picture of decreasing heterozygosity stands in contrast
to the consistency of Hardy-Weinberg equilibrium from the previous
chapter. However, our Hardy-Weinberg <em>proportions</em> still hold in
forming each new generation. As the offspring genotypes in the next
generation (<span class="math inline">\(t+1\)</span>) represent a random
draw from the previous generation (<span
class="math inline">\(t\)</span>), if the parental frequency is <span
class="math inline">\(p_t\)</span>, we <em>expect</em> a proportion
<span class="math inline">\(2p_t(1-p_t)\)</span> of our offspring to be
heterozygotes (and HW proportions for our homozygotes). However, because
population size is finite, the observed genotype frequencies in the
offspring will (likely) not match exactly with our expectations. As our
genotype frequencies likely change slightly due to sampling,
biologically this reflects random variation in family size and Mendelian
segregation, the allele frequency will changed. Therefore, while each
generation represents a sample from Hardy-Weinberg proportions based on
the generation before, our genotype proportions are not at an
equilibrium (an unchanging state) as the underlying allele frequency
changes over the generations. We’ll develop some mathematical models for
these allele frequency changes later on. For now, we’ll simply note that
under our simple model of drift (formally the Wright-Fisher model), our
allele count in the <span class="math inline">\(t+1^{th}\)</span>
generation represents a binomial sample (of size <span
class="math inline">\(2N\)</span>) from the population frequency <span
class="math inline">\(p_t\)</span> in the previous generation. If you’ve
read to here, please email Prof Coop a picture of JBS Haldane in a
striped suit with the title "I’m reading the chapter 3 notes”. (It’s
well worth googling JBS Haldane and to read more about his life; he’s a
true character and one of the last great polymaths. )</p>
<div class="center">
<figure>
<embed
src="Journal_figs/genetic_drift/black_footed_ferrets/black_footed_ferrets_He.pdf"
id="fig:LossHet_ferrets" />
<figcaption aria-hidden="true">Loss of heterozygosity in the
Black-footed Ferrets in their declining population. Numbers in brackets
give estimated number of individuals alive at that time. Data from <span
class="citation" data-cites="Wisely:02">@Wisely:02</span>.</figcaption>
</figure>
</div>
<p>To see how a decline in population size can affect levels of
heterozygosity, let’s consider the case of black-footed ferrets
(<em>Mustela nigripes</em>). The black-footed ferret population has
declined dramatically through the twentieth century due to destruction
of their habitat and sylvatic plague. In 1979, when the last known
black-footed ferret died in captivity, they were thought to be extinct.
In 1981, a very small wild population was rediscovered (<span
class="math inline">\(40\)</span> individuals), but in 1985 this
population suffered a number of disease outbreaks.</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="illustration_images/Genetic_drift/Black_footed_ferrets/Black_footed_ferret.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>At that point of the <span class="math inline">\(18\)</span>
remaining wild individuals were brought into captivity, 7 of which
reproduced. Thanks to intense captive breeding efforts and conservation
work, a wild population of over 300 individuals has been established
since. However, because all of these individuals are descended from
those 7 individuals who survived the bottleneck, diversity levels remain
low. <span class="citation" data-cites="Wisely:02">@Wisely:02</span>
measured heterozygosity at a number of microsatellites in individuals
from museum collections, showing the sharp drop in diversity as
population sizes crashed (see Figure <a href="#fig:LossHet_ferrets"
data-reference-type="ref"
data-reference="fig:LossHet_ferrets">4.4</a>).</p>
<p><span id="geoquestion" label="geoquestion"></span> In mathematical
population genetics, a commonly used approximation is <span
class="math inline">\((1-x) \approx e^{-x}\)</span> for <span
class="math inline">\(x &lt;&lt; 1\)</span> (formally, this follows from
the Taylor series expansion of <span
class="math inline">\(\exp(-x)\)</span>, ignoring second order and
higher terms of <span class="math inline">\(x\)</span>, see Appendix
eqn<a href="#eqn:Taylor_geo" data-reference-type="eqref"
data-reference="eqn:Taylor_geo">[eqn:Taylor_geo]</a>). This
approximation is especially useful for approximating a geometric decay
process by an exponential decay process, e.g. <span
class="math inline">\((1 - x)^t \approx e^{-xt}\)</span>. Using your
calculator, or R, check how well this expression approximates the exact
expression for two values of <span class="math inline">\(x\)</span>,
<span class="math inline">\(x = 0.1\)</span>, and <span
class="math inline">\(0.01\)</span>, across two different values of t,
<span class="math inline">\(t=5\)</span> and <span
class="math inline">\(t=50\)</span>. Briefly comment on your
results.</p>
<h3 id="DriftMutationBalance">Levels of diversity maintained by a
balance between mutation and drift</h3>
<p>Next we’re going to consider the amount of neutral polymorphism that
can be maintained in a population as a balance between genetic drift
removing variation and mutation introducing new neutral variation, see
Figure <a href="#fig:Mut_Sel_balance" data-reference-type="ref"
data-reference="fig:Mut_Sel_balance">4.5</a> for an example. Note in our
example, how no single allele is maintained at a stable equilibrium,
rather an equilibrium level of polymorphism is maintained by a
constantly shifting set of alleles.</p>
<div class="center">
<figure>
<img src="figures/Mut_drift_balance.png" id="fig:Mut_Sel_balance"
alt="Mutation-drift balance. A diploid population of 5 individuals. In the first generation everyone has the same allele (black). Each generation the transmitted allele can mutate and we generate a new colour. In the bottom plot, I trace the frequency of alleles in our population over time. The mutation rate we use is very high, simply to maintain diversity in this small population." />
<figcaption aria-hidden="true">Mutation-drift balance. A diploid
population of 5 individuals. In the first generation everyone has the
same allele (black). Each generation the transmitted allele can mutate
and we generate a new colour. In the bottom plot, I trace the frequency
of alleles in our population over time. The mutation rate we use is very
high, simply to maintain diversity in this small
population.</figcaption>
</figure>
</div>
<h5 id="the-neutral-mutation-rate.">The neutral mutation rate.</h5>
<p>We’ll first want to consider the rate at which neutral mutations
arise in the population.Thinking back to our discussion of the neutral
theory of molecular evolution, let’s suppose that there are only two
classes of mutation that can arise in our genomic region of interest:
neutral mutations and highly deleterious mutations. The total mutation
rate at our locus is <span class="math inline">\(\mu\)</span> per
generation, i.e. per transmission from parent to child. A fraction <span
class="math inline">\(C\)</span> of our mutations are new alleles that
are highly deleterious and so quickly removed from the population. We’ll
call this <span class="math inline">\(C\)</span> parameter the
constraint, and it will differ according to the genomic region we
consider. The remaining fraction <span
class="math inline">\((1-C)\)</span> are our neutral mutations, such
that our neutral mutation rate is <span
class="math inline">\((1-C)\mu\)</span>. This is the per generation
rate. In the rest of the chapter for simplicity we’ll assume that <span
class="math inline">\(C=0\)</span> and use a neutral mutation rate of
<span class="math inline">\(\mu\)</span>. However, we’ll return to this
discussion of constaint when we discuss molecular divergence in a
subsequent chapter.</p>
<p>It’s worth taking a minute to get familiar with both how rare, and
how common, mutation is. The per base pair mutation rate in humans is
around <span class="math inline">\(1.5\)</span> <span
class="math inline">\(\times\)</span> <span
class="math inline">\(10^{-8}\)</span> per generation. That means, on
average, we have to monitor a site for <span class="math inline">\(\sim
66.6\)</span> million transmissions from parent to child to see a
mutation. Yet populations and genomes are big places, so mutations are
common at these levels.<br />
<strong>A)</strong> Your autosomal genome is <span
class="math inline">\(\sim\)</span> 3 billion base pairs long (<span
class="math inline">\(3\)</span> <span
class="math inline">\(\times\)</span> <span
class="math inline">\(10^9\)</span>). You have two copies, the one you
received from your mum and one from your dad. What is the average
(i.e. the expected) number of mutations that occurred in the
transmission from your mum and your dad to you?<br />
<strong>B)</strong> The current human population size is <span
class="math inline">\(\sim\)</span><!-- -->7 billion individuals. How
many times, at the level of the entire human population, is a single
base-pair mutated in the transmission from one generation to the
next?</p>
<h5
id="levels-of-heterozygosity-maintained-as-a-balance-between-mutation-and-drift.">Levels
of heterozygosity maintained as a balance between mutation and
drift.</h5>
<p>Looking backwards in time from one generation to the previous
generation, we are going to say that two alleles which have the same
parental allele (i.e. find their common ancestor) in the preceding
generation have <em>coalesced</em>, and refer to this event as a
<em>coalescent event</em>. If our pairs of alleles are to be different
from each other in the present day, a mutation must have occured more
recently on one or other lineage before they found a common
ancestor.</p>
<p>The probability that our pair of randomly sampled alleles have
coalesced in the preceding generation is <span
class="math inline">\(\frac{1}{(2N)}\)</span>, and the probability that
our pair of alleles fail to coalesce is <span
class="math inline">\(1-\frac{1}{(2N)}\)</span>.</p>
<p>The probability that a mutation changes the identity of the
transmitted allele is <span class="math inline">\(\mu\)</span> per
generation. So the probability of no mutation occurring is <span
class="math inline">\((1-\mu)\)</span>. We’ll assume that when a
mutation occurs it creates some new allelic type which is not present in
the population. This assumption (commonly called the
infinitely-many-alleles model) makes the math slightly cleaner, and also
is not too bad an assumption biologically. See Figure <a
href="#fig:Mut_Sel_balance" data-reference-type="ref"
data-reference="fig:Mut_Sel_balance">4.5</a> for a depiction of
mutation-drift balance in this model over the generations.</p>
<p>This model lets us calculate when our two alleles last shared a
common ancestor and whether these alleles are identical as a result of
failing to mutate since this shared ancestor. For example, we can work
out the probability that our two randomly sampled alleles coalesce <span
class="math inline">\(2\)</span> generations in the past (i.e. they fail
to coalesce in generation <span class="math inline">\(1\)</span> and
then coalesce in generation <span class="math inline">\(2\)</span>), and
that they are identical as <span class="math display">\[\left(1-
\frac{1}{2N} \right) \frac{1}{2N} (1-\mu)^4\]</span> Note the power of
<span class="math inline">\(4\)</span> is because our two alleles have
to have failed to mutate through <span class="math inline">\(2\)</span>
meioses each.</p>
<p>More generally, the probability that our alleles coalesce in
generation <span class="math inline">\(t+1\)</span> (counting backwards
in time) and are identical due to no mutation to either allele in the
subsequent generations is <span
class="math display">\[\mathbb{P}(\textrm{coal. in t+1 \&amp; no
mutations}) =  \frac{1}{2N} \left(1- \frac{1}{2N} \right)^t \left(1-\mu
\right)^{2(t+1)}\]</span> To make this slightly easier on ourselves
let’s further assume that <span class="math inline">\(t \approx
t+1\)</span> and so rewrite this as: <span
class="math display">\[\mathbb{P}(\textrm{coal. in t+1 \&amp; no
mutations}) \approx \frac{1}{2N} \left(1- \frac{1}{2N} \right)^t
\left(1-\mu \right)^{2t}\]</span></p>
<p>This gives us the approximate probability that two alleles will
coalesce in the <span class="math inline">\((t+1)^\text{th}\)</span>
generation. In general, we may not know when two alleles may coalesce:
they could coalesce in generation <span class="math inline">\(t=1, t=2,
\ldots\)</span>, and so on. Thus, to calculate the probability that two
alleles coalesce in <em>any</em> generation before mutating, we can
write: <span class="math display">\[\begin{aligned}
  \mathbb{P}(\textrm{coal. in any generation \&amp; no mutations})
\approx &amp; \mathbb{P}(\textrm{coal. in} \; t=1 \; \textrm{\&amp; no
mutations}) \; + \nonumber\\
&amp;  \mathbb{P}(\textrm{coal. in} \; t=2 \; \textrm{\&amp; no
mutations}) + \ldots \nonumber\\
  %P(\textrm{coal. in} \; t=3 \; \textrm{\&amp; no mutations})  +\ldots
\nonumber\\
  = &amp; \sum_{t=1}^\infty \mathbb{P}(\textrm{coal. in } \; t \;
\textrm{generations \&amp; no mutation})
\end{aligned}\]</span></p>
<p>an example of using the Law of Total Probability, see Appendix eqn <a
href="#eqn:law_tot_prob" data-reference-type="eqref"
data-reference="eqn:law_tot_prob">[eqn:law_tot_prob]</a>, combined with
the fact that coalescing in a particular generation is mutually
exclusive with coalescing in a different generation.</p>
<p>While we could calculate a value for this sum given <span
class="math inline">\(N\)</span> and <span
class="math inline">\(\mu\)</span>, it’s difficult to get a sense of
what’s going on with such a complicated expression. Here, we turn to a
common approximation in population genetics (and all applied
mathematics), where we assume that <span
class="math inline">\(\frac{1}{(2N)} \ll 1\)</span> and <span
class="math inline">\(\mu \ll 1\)</span>. This allows us to approximate
the geometric decay as an exponential decay (see Appendix eqn <a
href="#eqn:Taylor_exp" data-reference-type="eqref"
data-reference="eqn:Taylor_exp">[eqn:Taylor_exp]</a>). Then, the
probability two alleles coalesce in generation <span
class="math inline">\(t+1\)</span> and don’t mutate can be written as:
<span class="math display">\[\begin{aligned}
\mathbb{P}(\textrm{coal. in t+1 \&amp; no mutations}) &amp;\approx
\frac{1}{2N}
\left(1- \frac{1}{2N} \right)^t \left(1-\mu \right)^{2t} \\
&amp; \approx \frac{1}{2N} e^{-t/(2N)} e^{-2\mu t } \\
&amp;=\frac{1}{2N} e^{-t(2\mu+1/(2N))}
\end{aligned}\]</span> Then we can approximate the summation by an
integral, giving us:</p>
<p><span class="math display">\[\frac{1}{2N} \int_0^{\infty}
e^{-t(2\mu+1/(2N))} dt =
\frac{1/(2N)}{1/(2N)+2\mu} \label{eqn:coal_no_mut}\]</span></p>
<p>The equation above gives us the probability that our two alleles
coalesce at some point in time, and do not mutate before reaching their
common ancestor. Equivalently, this can be thought of as the probability
our two alleles coalesce <em>before</em> mutating, i.e. that they are
homozygous.</p>
<p>Then, the complementary probability that our pair of alleles are
non-identical (or heterozygous) is simply one minus this. The following
equation gives the equilibrium heterozygosity in a population at
equilibrium between mutation and drift:</p>
<p><span class="math display">\[H = \frac{2\mu}{1/(2N)+2\mu}  =
\frac{4N\mu}{1+4N\mu} \label{eqn:hetero}\]</span> The compound parameter
<span class="math inline">\(4N\mu\)</span>, the population-scaled
mutation rate, will come up a number of times so we’ll give it its own
name: <span class="math display">\[\theta = 4N\mu\]</span> What’s the
intuition of our eqn<a href="#eqn:hetero" data-reference-type="eqref"
data-reference="eqn:hetero">[eqn:hetero]</a>, well the probability that
any event happens in a particular generation is <span
class="math inline">\(\mathbb{P}(\textrm{mutation or coalescence})
\approx \frac{1}{(2N)}+2\mu\)</span>, so conditional on an event
happening the probability that it is a mutation is <span
class="math inline">\(\mathbb{P}(\textrm{mutation} \mid \textrm{mutation
or coalescence}) = \frac{2\mu}{\left(\frac{1}{(2N)}+2\mu
\right)}\)</span>.</p>
<p>So all else being equal, species with larger population sizes should
have proportionally higher levels of neutral polymorphism. Indeed,
populations of animals, e.g. birds, on small islands have lower levels
of diversity than closely related species on the mainland with larger
ranges. More generally, we do see higher levels of heterozygosity in
larger census population sizes across animals Figure <a
href="#fig:allozyme_N" data-reference-type="ref"
data-reference="fig:allozyme_N">4.6</a>. However, while census
population sizes vary over many orders of magnitude, levels of diversity
vary much less than that. So, if levels of diversity in natural
populations represent a balance between genetic drift and mutation,
levels of genetic drift in large populations must be a lot faster than
their census population size suggests. In the next section we’ll talk
about some possible reasons why.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/genetic_drift/Allozyme_pop_size/bird_allozyme_pop_size.pdf"
id="fig:allozyme_N" />
<figcaption aria-hidden="true">Average basepair heterozygosity plotted
against the log of range size for endemic island and mainland bird
populations <span class="citation"
data-cites="leroy2020endemic">[@leroy2020endemic]</span>. Average
allozyme heterozygosity plotted against the log of census population
size (N) for animals. Data from <span class="citation"
data-cites="soule1976allozyme frankham1996relationship">@soule1976allozyme
[@frankham1996relationship]</span></figcaption>
</figure>
</div>
<h3 id="the-effective-population-size">The effective population
size</h3>
<p>In practice, populations rarely conform to our assumptions of being
constant in size with low variance in reproductive success. Real
populations experience dramatic fluctuations in size, and there is often
high variance in reproductive success. Thus rates of drift in natural
populations are often a lot higher than the census population size would
imply. See Figure <a href="#fig:LossHet_varying_pop"
data-reference-type="ref"
data-reference="fig:LossHet_varying_pop">4.7</a> for a depiction of a
repeatedly bottlenecked population losing diversity at a fast rate.</p>
<div class="center">
<figure>
<img src="figures/Loss_of_he_col_alleles_varying_pop_dark.png"
id="fig:LossHet_varying_pop"
alt="Loss of heterozygosity over time in a bottlenecking population. A diploid population of 10 individuals, that bottlenecks down to three individuals repeatedly. In the first generation, I colour every allele a different colour so we can track their descendants. There are no new mutations." />
<figcaption aria-hidden="true">Loss of heterozygosity over time in a
bottlenecking population. A diploid population of 10 individuals, that
bottlenecks down to three individuals repeatedly. In the first
generation, I colour every allele a different colour so we can track
their descendants. There are no new mutations.</figcaption>
</figure>
</div>
<p>To cope with this discrepancy, population geneticists often invoke
the concept of an <em>effective population size</em> (<span
class="math inline">\(N_e\)</span>). In many situations (but not all),
departures from model assumptions can be captured by substituting <span
class="math inline">\(N_e\)</span> for <span
class="math inline">\(N\)</span>.<br />
If population sizes vary rapidly in size, we can (if certain conditions
are met) replace our population size by the harmonic mean population
size. Consider a diploid population of variable size, whose size is
<span class="math inline">\(N_t\)</span> <span
class="math inline">\(t\)</span> generations into the past. The
probability our pairs of alleles have not coalesced by generation <span
class="math inline">\(t\)</span> is given by <span
class="math display">\[\prod_{i=1}^{t} \left(1-\frac{1}{2N_i} \right)
\label{eqn:var_pop_coal}\]</span> Note that this simply collapses to our
original expression <span class="math inline">\(\left(1-\frac{1}{2N }
\right)^t\)</span> if <span class="math inline">\(N_i\)</span> is
constant. Under this model, the rate of loss of heterozygosity in this
population is equivalent to a population of effective size <span
class="math display">\[N_e =\frac{1}{\frac{1}{t} \sum_{i=1}^{t}
\frac{1}{N_i} }. \label{eq:Ne_harmonic}\]</span> This is the harmonic
mean of the varying population size.</p>
<p>Thus our effective population size, the size of an idealized constant
population which matches the rate of genetic drift, is the harmonic mean
true population size over time. The harmonic mean is very strongly
affected by small values, such that if our population size is one
million <span class="math inline">\(99\%\)</span> of the time but drops
to <span class="math inline">\(1000\)</span> every hundred or so
generations, <span class="math inline">\(N_e\)</span> will be much
closer to <span class="math inline">\(1000\)</span> than a
million.<br />
</p>
<div class="center">
<figure>
<img src="figures/Loss_of_he_col_alleles_varying_RS.png"
id="fig:LossHet_varying_RS"
alt="High variance on reproductive success increases the rate of genetic drift. A diploid population of 10 individuals, where the circled individuals have much higher reproductive success. In the first generation I colour every allele a different colour so we can track their descendants, there are no new mutations." />
<figcaption aria-hidden="true">High variance on reproductive success
increases the rate of genetic drift. A diploid population of 10
individuals, where the circled individuals have much higher reproductive
success. In the first generation I colour every allele a different
colour so we can track their descendants, there are no new
mutations.</figcaption>
</figure>
</div>
<p>Variance in reproductive success will also affect our effective
population size. Even if our population has a large constant size <span
class="math inline">\(N\)</span> individuals, if only small proportion
of them get to reproduce, then the rate of drift will reflect this much
smaller number of reproducing individuals. See Figure <a
href="#fig:LossHet_varying_RS" data-reference-type="ref"
data-reference="fig:LossHet_varying_RS">4.8</a> for a depiction of the
higher rate of drift in a population where there is high variance in
reproductive success.</p>
<p>To see one example of this, consider the case where <span
class="math inline">\(N_F\)</span> of females get to reproduce and <span
class="math inline">\(N_M\)</span> males get reproduce. While every
individual has a biological mother and father, not every individual gets
to be a parent. In practice, in many animal species far more females get
to reproduce than males, i.e. <span class="math inline">\(N_M
&lt;N_F\)</span>, as a few males get many mating opportunities and many
males get no/few mating opportunities <span class="citation"
data-cites="janicke:16">[see @janicke:16 for a broad analysis, and note
that there a certainly many exceptions to this general pattern]</span>.
When our two alleles pick an ancestor, <span
class="math inline">\(25\%\)</span> of the time our alleles were both in
a female ancestor, in which case they are IBD with probability <span
class="math inline">\(1/(2N_F)\)</span>, and <span
class="math inline">\(25\%\)</span> of the time they are both in a male
ancestor, in which case they coalesce with probability <span
class="math inline">\(1/(2N_M)\)</span>. The remaining <span
class="math inline">\(50\%\)</span> of the time, our alleles trace back
to two individuals of different sexes in the prior generation and so
cannot coalesce. Therefore, our probability of coalescence in the
preceding generation is <span
class="math display">\[\frac{1}{4}\left(\frac{1}{2N_M}
\right)+\frac{1}{4}\left(\frac{1}{2N_F} \right) %=
%\frac{1}{8}\frac{N_F+N_M}{N_FN_M}\]</span> i.e. the rate of coalescence
is the harmonic mean of the two sexes’ population sizes, equating this
to <span class="math inline">\(\frac{1}{2N_e}\)</span> we find</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="illustration_images/Genetic_drift/Hamadryas_baboon/Hamadryas_baboon.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p><span class="math display">\[N_e = \frac{4N_FN_M}{N_F+N_M}\]</span>
Thus if reproductive success is very skewed in one sex (e.g. <span
class="math inline">\(N_M \ll N/2\)</span>), our autosomal effective
population size will be much reduced as a result. For more on how
different evolutionary forces affect the rate of genetic drift, and
their impact on the effective population size, see <span
class="citation"
data-cites="charlesworth:09">@charlesworth:09</span>.<br />
<strong>Question</strong> You are studying a population of 500 male and
500 female Hamadryas baboons. Assume that all of the females but only
1/10 of the males get to mate. What is the effective population size for
the autosome?<br />
</p>
<p>Variance in male and female reproductive success can have very
different effects on chromosomes with differing modes of inheritance
such as the X chromosome, mitochondria, and Y chromosome. The
mitochondria (mtDNA) and Y chromosome are haploid and only inherited
through the females and males respectively, so they have a haploid
effective population sizes of <span class="math inline">\(N_M\)</span>
and <span class="math inline">\(N_F\)</span>.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/genetic_drift/Scythian_horses/Scythian_horses.pdf"
id="fig:Scythian_horses_pi" />
<figcaption aria-hidden="true">Levels of genome-wide diversity in
Scythian horses from <span class="math inline">\(2300\)</span> year old
Scythian horses and Modern horses (Nordic). The numbers next to each
column given the fraction of diversity remaining in the present day,
Data from <span class="citation"
data-cites="librado2017ancient">@librado2017ancient</span>.
.</figcaption>
</figure>
</div>
<p>To see the impact of differential variance in male and female
reproductive success, let’s look at how levels of genetic diversity over
thousands of years in domestic horses.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Genetic_drift/Scythian_rider/616px-Szkíta.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p><span class="citation"
data-cites="librado2017ancient">@librado2017ancient</span> sequenced
ancient DNA from 13 sacrificed stallions from an <span
class="math inline">\(2300\)</span> year old Scythian burial mound in
Kazakhstan. The Scythian were a nomadic people whose Russian Steppe
empire stretched from the Black Sea to the borders of China. They were
among the first people to master horseback warfare with both men and
women riding armed with short bows. </p>
<p>By comparing these data to modern horses, <span class="citation"
data-cites="librado2017ancient">@librado2017ancient</span> found that
levels of diversity had been substantially reduced on the autosomes and
greatly reduced on the Y chromosome. This constrasts with the mtDNA
where levels of diversity have decreased only slightly. This pattern
likely reflects the fact that much of modern horse breeding relies on a
breeding a small number of stallions to a large number of mares, and so
the effective population size of the Y chromosome has been much smaller
than the mtDNA leading to a much higher rate of loss of diversity on the
Y than on other chromosomes.</p>
<p>Using the data on the reduction in horse genetic diversity in Figure
<a href="#fig:Scythian_horses_pi" data-reference-type="ref"
data-reference="fig:Scythian_horses_pi">4.9</a>:<br />
<strong>A)</strong> Estimate the effective number of stallions and mares
contributing to the horse population using the mtDNA and Y chromosome
data. B) Predict what the reduction in diversity over the <span
class="math inline">\(2300\)</span> years should be on the autosomes
using these numbers?<br />
Assume a horse generation time of <span class="math inline">\(8\)</span>
years. Assume no new mutations during this time interval.</p>
<p>One of the highest levels of genetic diversity is seen in the diploid
split-gill fungus, <em>Schizophyllum commune</em>. Populations in the
USA have a sequence-level heterozygosity of <span
class="math inline">\(0.13\)</span> per synonymous base <span
class="citation"
data-cites="baranova2015extraordinary">[@baranova2015extraordinary]</span>.
<span class="citation"
data-cites="baranova2015extraordinary">@baranova2015extraordinary</span>
sequenced parents and multiple offspring to estimate that <span
class="math inline">\(\mu= 2 \times 10^{-8} bp^{-1}\)</span> per
generation. What is your estimate of the effective population size of
<em>S. commune</em>?</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Genetic_drift/splitgill_mushroom/splitgill_mushrooms.png"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h2 id="the-coalescent-and-patterns-of-neutral-diversity">The Coalescent
and patterns of neutral diversity</h2>
<blockquote>
<p>“Life can only be understood backwards; but it must be lived
forwards” – Kierkegaard</p>
</blockquote>
<h5
id="pairwise-coalescent-time-distribution-and-the-number-of-pairwise-differences.">Pairwise
Coalescent time distribution and the number of pairwise
differences.</h5>
<p>Thinking back to our calculations we made about the loss of neutral
heterozygosity and equilibrium levels of diversity (in Sections <a
href="#LossofHet" data-reference-type="ref"
data-reference="LossofHet">4.1</a> and <a href="#DriftMutationBalance"
data-reference-type="ref"
data-reference="DriftMutationBalance">4.1.1</a>), you’ll note that we
could first specify which generation a pair of sequences coalesce in,
and then calculate some properties of heterozygosity based on that.
That’s because neutral mutations do not affect the probability that an
individual transmits an allele, and so don’t affect the way in which we
can trace ancestral lineages back through the generations.<br />
</p>
<p>As such, it will often be helpful to consider the time to the common
ancestor of a pair of sequences (<span
class="math inline">\(T_2\)</span>), and then think of the impact of
that time to coalescence on patterns of diversity. See Figure <a
href="#fig:Coalescent_simulation" data-reference-type="ref"
data-reference="fig:Coalescent_simulation">4.10</a> for an example of
this.</p>
<div class="center">
<figure>
<img src="figures/Coalescent.png" id="fig:Coalescent_simulation"
alt="A simple demonstration of the coalescent process. The simulation consists of a diploid population of 10 individuals (20 alleles). In each generation, each individual is equally likely to be the parent of an offspring (and the allele transmitted is indicated by a light grey line). We track a pair of alleles, chosen in the present day, back 14 generations until they find a common ancestor. Deeper in time than 14 generations those two alleles have the same ancestral lineage and completely share their history, e.g. the mutations that occur on that lineage." />
<figcaption aria-hidden="true">A simple demonstration of the coalescent
process. The simulation consists of a diploid population of 10
individuals (20 alleles). In each generation, each individual is equally
likely to be the parent of an offspring (and the allele transmitted is
indicated by a light grey line). We track a pair of alleles, chosen in
the present day, back 14 generations until they find a common ancestor.
Deeper in time than 14 generations those two alleles have the same
ancestral lineage and completely share their history, e.g. the mutations
that occur on that lineage.</figcaption>
</figure>
</div>
<p>The probability that a pair of alleles have failed to coalesce in
<span class="math inline">\(t\)</span> generations and then coalesce in
the <span class="math inline">\(t+1\)</span> generation back is <span
class="math display">\[\mathbb{P}(T_2=t+1) = \frac{1}{2N} \left(1-
\frac{1}{2N} \right)^{t} \label{eqn:coal_time_dist}\]</span> For
example, the probability that a pair of sequences coalesce three
generations back is the probability that they fail to coalesce in
generation 1 and 2, which is <span class="math inline">\(\left(1-
\frac{1}{2N} \right) \times \left(1- \frac{1}{2N} \right)\)</span>,
multipled by the probability that they find a common ancestor, i.e.
coalesce, in the third generation, which happens with probability <span
class="math inline">\(\frac{1}{2N}\)</span>.</p>
<p>From the form of eqn <a href="#eqn:coal_time_dist"
data-reference-type="eqref"
data-reference="eqn:coal_time_dist">[eqn:coal_time_dist]</a> we can see
that the coalescent time of our pair of alleles is a Geometrically
distributed random variable, where the probability of success is <span
class="math inline">\(p=\frac{1}{2N}\)</span>. The waiting time for a
pair of lineages to coalesce is like the number of tails thrown while
waiting for a head on a coin with the probability of a head is <span
class="math inline">\(\frac{1}{2N}\)</span>, i.e. if the population is
large we might be waiting for a long time for our pair to coalesce.
We’ll denote this geometric distribution by <span
class="math inline">\(T_2 \sim \text{Geo}(1/(2N))\)</span>. The expected
(i.e. the mean over many replicates) coalescent time of a pair of
alleles is then <span class="math display">\[\mathbb{E}(T_2) =
2N\]</span> generations. This form to the expectation follows from the
fact that the mean of an geometric random variable is <span
class="math inline">\(\frac{1}{p}\)</span>.<br />
Conditional on a pair of alleles coalescing <span
class="math inline">\(t\)</span> generations ago, there are <span
class="math inline">\(2t\)</span> generations in which a mutation could
occur. See Figure <a href="#fig:pair_coal_muts"
data-reference-type="ref"
data-reference="fig:pair_coal_muts">[fig:pair_coal_muts]</a> for an
example. If the per generation mutation rate is <span
class="math inline">\(\mu\)</span>, then the expected number of
mutations between a pair of alleles coalescing <span
class="math inline">\(t\)</span> generations ago is <span
class="math inline">\(2 t\mu\)</span> (the alleles have gone through a
total of <span class="math inline">\(2t\)</span> meioses since they last
shared a common ancestor).</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/Coalescent/Coal_two_lineages_muts.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>So we can write the expected number of mutations (<span
class="math inline">\(S_2\)</span>) separating two alleles drawn at
random from the population as <span
class="math display">\[\begin{aligned}
\mathbb{E}(S_2) &amp;= \sum_{t=0}^{\infty} \mathbb{E}(S_2 | T_2=t)
P(T_2=t) \nonumber\\
&amp; =\sum_{t=0}^{\infty} 2 \mu t P(T_2=t) \nonumber\\
&amp; =2\mu \mathbb{E}(T_2)  \nonumber\\
&amp; = 4 \mu N
\end{aligned}\]</span> this makes use of the law of total expectation
(see Appendix eqn <a href="#eqn:tot_exptation_def"
data-reference-type="eqref"
data-reference="eqn:tot_exptation_def">[eqn:tot_exptation_def]</a>) to
average which generation our pair of sequences coalesce in. We’ll assume
that mutation is rare enough that it never happens at the same basepair
twice, i.e. no multiple hits, such that we get to see all of the
mutation events that separate our pair of sequences. This is assumption
that repeat mutation is vanishingly rare at a basepair is called the
<em>i</em>nfinitely-many-sites assumption, which should hold if <span
class="math inline">\(N\mu_{BP} \ll 1\)</span>, where <span
class="math inline">\(\mu_{BP}\)</span> is the mutation rate per
basepair. Thus the number of mutations between a pair of sites is the
observed number of differences between a pair of sequences. In the
previous chapter we denote the observed number of pairwise differences
at putatively neutral sites separating a pair of sequences as <span
class="math inline">\(\pi\)</span> (we usually average this over a
number of pairs of sequences for a region). Therefore, under our simple,
neutral, constant population-size model we expect <span
class="math display">\[\mathbb{E}(\pi) = 4 N \mu = \theta
\label{eqn:pi_expectation}\]</span> So we can get an empirical estimate
of <span class="math inline">\(\theta\)</span> from <span
class="math inline">\(\pi\)</span>, let’s call this <span
class="math inline">\(\widehat{\theta}_{\pi}\)</span>, by setting <span
class="math inline">\(\widehat{\theta}_{\pi}=\pi\)</span>, i.e. our
observed level of pairwise genetic diversity. If we have an independent
estimate of <span class="math inline">\(\mu\)</span>, then from setting
<span class="math inline">\(\pi =\widehat{\theta}_{\pi} = 4N\mu\)</span>
we can furthermore obtain an estimate of the population size <span
class="math inline">\(N\)</span> that is consistent with our levels of
neutral polymorphism. If we estimate the population size this way, we
should call it the effective coalescent population size (<span
class="math inline">\(N_e\)</span>). It’s best to think about <span
class="math inline">\(N_{e}\)</span> estimated from neutral diversity as
a long-term effective population size for the species, but there are
many caveats that come along with that assumption. For example, past
bottlenecks and population expansions are all subsumed into a single
number and so this estimated <span class="math inline">\(N_{e}\)</span>
may not be very representative of the population size at any time. That
said, it’s not a bad place to start when thinking about the rate of
genetic drift for neutral diversity in our population over long
time-periods.</p>
<p>Let’s take a moment to distinguish our expected heterozygosity (eqn.
<a href="#eqn:hetero" data-reference-type="ref"
data-reference="eqn:hetero">[eqn:hetero]</a>) from our expected number
of pairwise differences (<span class="math inline">\(\pi\)</span>). Our
expected heterozygosity is the probability that two alleles at a locus,
sampled from a population at random, are different from each other. If
one or more mutations have occurred since a pair of alleles last shared
a common ancestor, then our sequences will be different from each other.
On the other hand, our <span class="math inline">\(\pi\)</span> measure
keeps track of the average total number of differences between our loci.
As such, <span class="math inline">\(\pi\)</span> is often a more useful
measure, as it records the number of differences between the sequences,
not just whether they are different from each other (however, for
certain types of loci, e.g. microsatellites, heterozygosity is often
used as we cannot usually count up the minimum number of mutations in a
sensible way). In the case where our locus is a single basepair, the two
measures will usually be close to one another, as <span
class="math inline">\(H \approx \theta\)</span> for small values of
<span class="math inline">\(\theta\)</span>. For example, comparing two
sequences at random in humans, <span class="math inline">\(\pi \approx
1/1000\)</span> per basepair, and the probability that a specific base
pair differs between two sequences is <span
class="math inline">\(\approx 1/1000\)</span>. However, these two
quantities start to differ from each other when we consider regions with
higher mutation rates. For example, if we consider a 10kb region, our
mutation rate will 10,000 times larger than a single base pair. For this
length of sequence the probability that two randomly chosen haplotypes
differ is quite different from the number of mutational differences
between them. (Try a mutation rate of <span
class="math inline">\(10^{-8}\)</span> per base and a population size of
<span class="math inline">\(10,000\)</span> in our calculations of <span
class="math inline">\(\mathbb{E}[\pi]\)</span> and H to see this.)</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/Grey_fox/14770789583_4db7ec5164_o.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p><span class="citation" data-cites="robinson:16">@robinson:16</span>
found that the endangered Californian Channel Island fox on San Nicolas
had very low levels of diversity (<span class="math inline">\(\pi
=0.000014 \text{bp}^{-1}\)</span>) compared to its close relative the
California mainland gray fox (<span
class="math inline">\(0.0012\text{bp}^{-1}\)</span>).<br />
<strong>A)</strong> Assuming a mutation rate of <span
class="math inline">\(2\times 10^{-8}\)</span> per bp, what effective
population sizes do you estimate for these two populations?<br />
<strong>B)</strong> Why is the effective population size of the Channel
Island fox so low? [Hint: quickly google Channel island foxes to read up
on their history, also to see how ridiculously cute they are.]</p>
<p>In your own words describe why the coalescent time of a pair of
lineages scales linearly with the (effective) population size.</p>
<h5
id="more-details-on-the-pairwise-coalescent-and-the-randomness-of-mutation.">More
details on the pairwise coalescent and the randomness of mutation.</h5>
<p>We found that our pairwise coalescent times followed a Geometric
distribution, eqn <a href="#eqn:coal_time_dist"
data-reference-type="eqref"
data-reference="eqn:coal_time_dist">[eqn:coal_time_dist]</a>. However,
that assumes discrete generations, and we’ll often was to think about
populations that lack discrete generations (i.e. individuals reproducing
at random times with some mean generation time). Using our exponential
approximation, we can see that is <span class="math display">\[\approx
\frac{1}{2N} e^{-t/(2N)}\]</span> and so think of a continuous random
variable, i.e. we could say that the coalescent time of a pair of
sequences (<span class="math inline">\(T_2\)</span>) is approximately
exponentially distributed with a rate <span
class="math inline">\(1/(2N)\)</span>, i.e. <span
class="math inline">\(T_2 \sim \text{Exp}\left( 1/(2N) \right)\)</span>.
Formally we can do this by taking the limit of the discrete process more
carefully. See Appendix eqn <a href="#eqn:exp_rv_def"
data-reference-type="eqref"
data-reference="eqn:exp_rv_def">[eqn:exp_rv_def]</a> for more on
exponential random variables.</p>
<p>We’ve derived the expected number of differences between a pair of
sequences and talked about the variability of the coalescent time for a
pair of sequences. The mutation process is also very variable; even if
two sequences coalesce in the very distant past by chance, they may
still be identical in the present if there was no mutation during that
time.</p>
<p>Conditional on the coalescent time <span
class="math inline">\(t\)</span>, the probability that our pair of
alleles are separated by <span class="math inline">\(S_2\)</span>
mutations since they last shared a common ancestor is bionomially
distributed <span class="math display">\[\mathbb{P}(S_2 | T_2 = t ) =
{2t \choose j} \mu^{j} (1-\mu)^{2t-j}\]</span> i.e. mutations happen in
<span class="math inline">\(j\)</span> generations and do not happen in
<span class="math inline">\(2t-j\)</span> generations (with <span
class="math inline">\({2t \choose j}\)</span> ways this combination of
events can possibly happen). See Appendix eqn <a
href="#eqn:binomial_dist" data-reference-type="eqref"
data-reference="eqn:binomial_dist">[eqn:binomial_dist]</a> for
discussion of the binomial distribution. Assuming that <span
class="math inline">\(\mu \ll 1\)</span> and that <span
class="math inline">\(2t-j \approx 2t\)</span>, then we can approximate
the probability that we have <span class="math inline">\(S_2\)</span>
mutations as a Poisson distribution: <span
class="math display">\[\mathbb{P}(S_2 | T_2 = t ) = \frac{ (2 \mu t
)^{j} e^{-2\mu t}}{j!}\]</span> i.e. a Poisson with mean <span
class="math inline">\(2\mu t\)</span>. This is an example of taking the
binomial distribution to its Poisson distribution limit, see Appendix
eqn <a href="#eqn:bionom_to_poiss" data-reference-type="eqref"
data-reference="eqn:bionom_to_poiss">[eqn:bionom_to_poiss]</a> for more
details. We’ll not make much use of this result, but it is very useful
in thinking about how to simulate the process of mutation.<br />
</p>
<h2 id="the-coalescent-process-of-a-sample-of-alleles.">The coalescent
process of a sample of alleles.</h2>
<p>Usually we are not just interested in pairs of alleles, or the
average pairwise diversity. Generally we are interested in the
properties of diversity in samples of a number of alleles drawn from the
population. Instead of just following a pair of lineages back until they
coalesce, we can follow the history of a sample of alleles back through
the population.</p>
<p>Consider first sampling three alleles at random from the population.
The probability that all three alleles choose exactly the same ancestral
allele one generation back is <span
class="math inline">\(\frac{1}{(2N)^2}\)</span>. If <span
class="math inline">\(N\)</span> is reasonably large, then this is a
very small probability. As such, it is very unlikely that our three
alleles coalesce all at once, and in a moment we’ll see that it is safe
to ignore such unlikely events.<br />
</p>
<div class="center">
<figure>
<embed src="figures/Coalescent/Coal_three_lineages.pdf"
id="fig:Coalescent_simulation_3" />
<figcaption aria-hidden="true">A simple simulation of the coalescent
process for three lineages. We track the ancestry of three modern-day
alleles, the first pair (red and purple) coalesce four generations back,
after which there are only two independent lineages we are tracking.
This pair then coalesces twelve generations in the past. Note that
different random realizations of this process will differ from each
other a lot. The <span class="math inline">\(T_{MRCA}\)</span> is <span
class="math inline">\(T_3+T_2\)</span>. The total time in the tree is
<span class="math inline">\(T_{tot}=3T_3 + 2T_2= 25\)</span>
generations.</figcaption>
</figure>
</div>
<p>The probability that a specific pair of alleles find a common
ancestor in the preceding generation is still <span
class="math inline">\(\frac{1}{(2N)}\)</span>. There are three possible
pairs of alleles, so the probability that no pair finds a common
ancestor in the preceding generation is <span
class="math display">\[\left(1-\frac{1}{2N} \right)^3 \approx \left( 1-
\frac{3}{2N} \right)\]</span> In making this approximation we are
multiplying out the right hand-side and ignoring terms of <span
class="math inline">\(1/N^2\)</span> and higher (a Taylor approximation,
see Appendix eqn <a href="#eqn:Taylor_exp" data-reference-type="eqref"
data-reference="eqn:Taylor_exp">[eqn:Taylor_exp]</a>). See Figure <a
href="#fig:Coalescent_simulation_3" data-reference-type="ref"
data-reference="fig:Coalescent_simulation_3">4.11</a> for a random
realization of this process.<br />
More generally, when we sample <span class="math inline">\(i\)</span>
alleles there are <span class="math inline">\({i \choose 2}\)</span>
pairs, i.e. <span class="math inline">\(i(i-1)/2\)</span> pairs. Thus,
the probability that no pair of alleles in a sample of size <span
class="math inline">\(i\)</span> coalesces in the preceding generation
is <span class="math display">\[\left(1-\frac{1}{(2N)} \right)^{{i
\choose
2}} \approx \left( 1- \frac{{i \choose
2}}{2N}\right)\]</span> while the probability any pair coalesces is
<span class="math inline">\(\approx \frac{{i \choose  2}}{2N}\)</span>,
again using eqn <a href="#eqn:Taylor_exp" data-reference-type="eqref"
data-reference="eqn:Taylor_exp">[eqn:Taylor_exp]</a>.<br />
We can ignore the possibility that more than pairs of alleles (e.g.
tripletons) simultaneously coalesce at once as terms of <span
class="math inline">\(\frac{1}{N^2}\)</span> and higher can be ignored
as they are vanishingly rare. Obviously in reasonable sample sizes there
are many more triples (<span class="math inline">\({i \choose
3}\)</span>) and higher order combinations than there are pairs (<span
class="math inline">\({i \choose 2}\)</span>), but if <span
class="math inline">\(i \ll N\)</span> then we are safe to ignore these
terms.</p>
<p>When there are <span class="math inline">\(i\)</span> alleles, the
probability that we wait until the <span
class="math inline">\(t+1\)</span> generation before any pair of alleles
coalesces is <span class="math display">\[\mathbb{P}(T_i =t+1) =
\frac{{i \choose
2}}{2N}\left( 1- \frac{{i \choose
2}}{2N}\right)^{t} \label{eqn:T_i}\]</span> Thus the waiting time to the
first coalescent event while there are <span
class="math inline">\(i\)</span> lineages is a geometrically distributed
random variable with probability of success <span
class="math inline">\(p=\frac{{i \choose 2}}{2N}\)</span>, which we
denote by <span class="math display">\[T_i \sim \text{Geo}
\left(  \frac{{i \choose
      2}}{2N} \right).\]</span> The mean waiting time till any of pair
within our sample coalesces is <span class="math display">\[\mathbb{E}(
T_i) = \frac{2N}{{i \choose  2}}  \label{eqn:E_T_i}\]</span> which again
follows from the mean of a geometric random variable being <span
class="math inline">\(\frac{1}{p}\)</span>.</p>
<p>After a pair of alleles first finds a common ancestral allele some
number of generations back in the past, we only have to keep track of
that common ancestral allele for the pair when looking further into the
past. In our example coalescent genealogy for our 3 alleles, shown in
Figure <a href="#fig:Coalescent_simulation_3" data-reference-type="ref"
data-reference="fig:Coalescent_simulation_3">4.11</a>, we start by
tracking the 3 lineages, then by chance the blue and purple coalesce in
the four generations back. Then we’re tracking just two lineages, the
red lineage and the ancestral lineage of the blue and purple alleles;
then those two coalesce and we’ve found our most recent common ancestor
of our sample. Another example with four tips is shown in Figure <a
href="#fig:Coal_w_muts" data-reference-type="ref"
data-reference="fig:Coal_w_muts">4.12</a>; we’re track four lineages,
then a pair coalesce, then we tracking three lineages, then a pair
coalesce, then we’re tracking two lineages, then this final pair
coalesce and we’ve found the most recent common ancestor of our sample
(fin, end scene).</p>
<p>More generally, when a pair of alleles in our sample of <span
class="math inline">\(i\)</span> alleles coalesces, we then switch to
having to follow <span class="math inline">\(i-1\)</span> alleles back
in time. Then when a pair of these <span
class="math inline">\(i-1\)</span> alleles coalesce, we then only have
to follow <span class="math inline">\(i-2\)</span> alleles back. This
process continues until we coalesce back to a sample of two, and from
there to a single most recent common ancestor (MRCA).<br />
</p>
<h5 id="simulating-a-coalescent-genealogy">Simulating a coalescent
genealogy</h5>
<p>To simulate a coalescent genealogy at a locus for a sample of <span
class="math inline">\(n\)</span> alleles we therefore simply follow the
following algorithm:</p>
<ol type="1">
<li><p>Set <span class="math inline">\(i=n\)</span>.</p></li>
<li><p>Simulate a random variable to be the time <span
class="math inline">\(T_i\)</span> to the next coalescent event from
<span class="math inline">\(T_i \sim  \text{Exp}\left(\frac{{i
\choose  2}}{2N} \right)\)</span></p></li>
<li><p>Choose a pair of alleles to coalesce at random from all possible
pairs.</p></li>
<li><p>Set <span class="math inline">\(i=i-1\)</span></p></li>
<li><p>Continue looping steps 2-4 until <span
class="math inline">\(i=1\)</span>, i.e. the most recent common ancestor
of the sample is found.</p></li>
</ol>
<p>By following this algorithm we are generating realizations of the
genealogy of our sample.<br />
</p>
<h3
id="expected-properties-of-coalescent-genealogies-and-mutations.">Expected
properties of coalescent genealogies and mutations.</h3>
<div class="center">
<figure>
<embed src="figures/Coalescent/Coal_w_muts.pdf" id="fig:Coal_w_muts" />
<figcaption aria-hidden="true">A simple coalescent tree from a single
coalescent simulation, tracing the genealogy of 4 alleles with
mutational changes marked with dashes showing transitions away from the
MRCA sequence (AGTTT) . The <span
class="math inline">\(T_{MRCA}\)</span> is <span
class="math inline">\(T_4+T_3+T_2\)</span>. The total time in the tree
is <span class="math inline">\(T_{tot}=4 T_4+3T_3 + 2T_2= 54\)</span>
generations.</figcaption>
</figure>
</div>
<h5 id="the-expected-time-to-the-most-recent-common-ancestor.">The
expected time to the most recent common ancestor.</h5>
<p>We will first consider the time to the most recent common ancestor of
the entire sample (<span class="math inline">\(T_{MRCA}\)</span>). This
is <span class="math display">\[T_{MRCA} = \sum_{i=n}^2 T_i\]</span>
generations back, where we are summing from <span
class="math inline">\(i=n\)</span> alleles counting backwards to <span
class="math inline">\(i=2\)</span> alleles (see Figure <a
href="#fig:Coal_w_muts" data-reference-type="ref"
data-reference="fig:Coal_w_muts">4.12</a> for example). As our
coalescent times for different <span class="math inline">\(i\)</span>
are independent, the expected time to the most recent common ancestor is
<span class="math display">\[\mathbb{E}(T_{MRCA}) = \sum_{i=n}^2
\mathbb{E}(T_i) = \sum_{i=n}^2  2N/{i \choose
2}\]</span> Using the fact that <span
class="math inline">\(\frac{1}{i(i-1)}=\frac{1}{i-1} -
\frac{1}{i}\)</span> and a bit of rearrangement, we can rewrite this as
<span class="math display">\[\mathbb{E}(T_{MRCA}) = 4N\left(1-
\frac{1}{n} \right) \label{TMRCA_neutral}\]</span> So the average <span
class="math inline">\(T_{MRCA}\)</span> scales linearly with population
size <span class="math inline">\(N\)</span>. Interestingly, as we move
to larger and larger samples (i.e. <span class="math inline">\(n \gg
1\)</span>), the average time to the most recent common ancestor
converges on <span class="math inline">\(4N\)</span>. What’s happening
here is that in large samples our lineages typically coalesce rapidly at
the start and very soon coalesce down to a much smaller number of
lineages.<br />
<strong>Question</strong> Assume an autosomal effective population of
10,000 individuals (roughly the long-term human estimate) and a
generation time of 30 years. What is the expected time to the most
recent common ancestor of a sample of 20 people? What is this time for a
sample of 500 people?</p>
<h5
id="the-expected-total-time-in-a-genealogy-and-the-number-of-segregating-sites.">The
expected total time in a genealogy and the number of segregating
sites.</h5>
<p>Mutations fall on specific lineages of the coalescent genealogy and
are transmitted to all descendants of their lineage. Furthermore, under
the infinitely-many-sites assumption, each mutation creates a new
segregating site. The mutation process is a <em>Poisson process</em>,
and the longer a particular lineage, i.e. the more generations of
meioses it represents, the more mutations that can accumulate on it. The
total number of segregating sites in a sample is thus a function of the
<em>total</em> amount of time in the genealogy of the sample, or the sum
of all the branch lengths on the genealogical tree, <span
class="math inline">\(T_{tot}\)</span>. Our total amount of time in the
genealogy is</p>
<p><span class="math display">\[T_{tot} = \sum_{i=n}^2 iT_i\]</span> as
when there are <span class="math inline">\(i\)</span> lineages, each
contributes a time <span class="math inline">\(T_i\)</span> to the total
time (see Figure <a href="#fig:Coal_w_muts" data-reference-type="ref"
data-reference="fig:Coal_w_muts">4.12</a> for an example). Taking the
expectation of the total time in the genealogy, <span
class="math display">\[\mathbb{E}(T_{tot}) = \sum_{i=n}^2 i \frac{2N}{{i
\choose
2} } = \sum_{i=n}^2 \frac{4N}{i -1} =\sum_{i=n-1}^1 \frac{4N}{i}
\label{eqn:E_T_tot}\]</span> we see that our expected total amount of
time in the genealogy scales linearly with our population size <span
class="math inline">\(N\)</span>. Our expected total amount of time is
also increasing with sample size <span class="math inline">\(n\)</span>,
but is doing so very slowly. This again follows from the fact that in
large samples, the initial coalescence usually happens very rapidly, so
that extra samples add little to the total amount of time in the
genealogical tree.<br />
We saw above that the number of mutational differences between a pair of
alleles that coalescence <span class="math inline">\(T_2\)</span>
generations ago was Poisson with a mean of <span class="math inline">\(2
\mu T_2\)</span>, where <span class="math inline">\(2T_{2}\)</span> is
the total branch length in this simple 2-sample genealogical tree. A
mutation that occurs on any branch of our genealogy will cause a
segregating polymorphism in the sample (meeting our
infinitely-many-sites assumption). Thus, if the total time in the
genealogy is <span class="math inline">\(T_{tot}\)</span>, there are
<span class="math inline">\(T_{tot}\)</span> generations for mutations.
So the total number of mutations segregating in our sample (<span
class="math inline">\(S\)</span>) is Poisson with mean <span
class="math inline">\(\mu T_{tot}\)</span>. Thus the expected number of
segregating sites in a sample of size <span
class="math inline">\(n\)</span> is <span
class="math display">\[\mathbb{E}(S) = \mu \mathbb{E}(T_{tot}) =
\sum_{i=n-1}^1 \frac{4N\mu }{i} = \theta
\sum_{i=n-1}^1 \frac{1}{i} \label{eqn:seg_sites}\]</span> Note that this
is growing with the sample size <span class="math inline">\(n\)</span>,
albeit very slowly (roughly at the rate of the <span
class="math inline">\(\log\)</span> of the sample size). We can use this
formula to derive another estimate of the population scaled mutation
rate <span class="math inline">\(\theta\)</span>, by setting our
observed number of segregating sites in a sample (<span
class="math inline">\(S\)</span>) equal to this expectation. We’ll call
this estimator <span class="math inline">\(\widehat{\theta}_W\)</span>:
<span class="math display">\[\widehat{\theta}_W =\frac{
S}{\sum_{i=n-1}^1 \frac{1}{i}}   \label{watterson_theta}\]</span> This
estimator of <span class="math inline">\(\theta\)</span> was devised by
<span class="citation" data-cites="watterson:75">@watterson:75</span>,
hence the <span class="math inline">\(W\)</span>.</p>
<h5 id="the-neutral-site-frequency-spectrum.">The neutral site-frequency
spectrum.</h5>
<p>We can use our coalescent process to find the expected number of
derived alleles present <span class="math inline">\(i\)</span> times out
of a sample size <span class="math inline">\(n\)</span>, e.g. how many
singletons (<span class="math inline">\(i = 1\)</span>) do we expect to
find in our sample? For example, in Figure <a href="#fig:Coal_w_muts"
data-reference-type="ref" data-reference="fig:Coal_w_muts">4.12</a> in
our sample of four sequences, there are 3 singletons and 2 doubletons.
The number of sites with these different allele frequencies depends on
the lengths of specific genealogical branches. A mutation that falls on
a branch with <span class="math inline">\(i\)</span> descendants will
create a derived allele with frequency <span
class="math inline">\(i\)</span>. For example, in our example tree in
Figure <a href="#fig:Coal_w_muts" data-reference-type="ref"
data-reference="fig:Coal_w_muts">4.12</a>, the total number of
generations where a mutation could arise and be a doubleton is <span
class="math inline">\(T_3+2T_2\)</span>, the total length of the branch
ancestral to just the orange and red allele <span
class="math inline">\((T_3+T_2)\)</span> plus the branch ancestral to
just the blue and purple allele <span
class="math inline">\((T_2)\)</span>.</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/Genetic_drift/freq_spec_tree.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>To see how we could go about working this out, let’s start by
considering the simple coalescent tree, shown in Figure <a
href="#fig:freq_coal" data-reference-type="ref"
data-reference="fig:freq_coal">[fig:freq_coal]</a>, for sample of <span
class="math inline">\(3\)</span> alleles drawn from a population.
Mutations that fall on the branches coloured in black will be derived
singletons, while mutations that fall along the orange branch will be
doubletons in the sample. The total number of generations where a
singleton mutation could arise is <span class="math inline">\(3 T_3 +
T_2\)</span>. Note that we only count the time where there are two
lineages <span class="math inline">\((T_{2})\)</span> once. So our
expected number of singletons, using eqn <a href="#eqn:E_T_i"
data-reference-type="eqref" data-reference="eqn:E_T_i">[eqn:E_T_i]</a>,
is <span class="math display">\[\mathbb{E}(S_i) = \mu \left(
3\mathbb{E}(T_3) +  \mathbb{E}(T_2) \right) = \mu \left( 3
  \frac{2N}{3}+ 2N \right) = \theta\]</span> By similar logic, the time
where doubletons could arise is <span class="math inline">\(T_2\)</span>
and our expected number of doubletons is <span
class="math inline">\(\mathbb{E}(S_i) =\theta/2\)</span>. Thus, there
are on average half as many doubletons as singletons.</p>
<p>Extending this logic to larger samples might be doable, but is
tedious (I mean really tedious: for 10 alleles there are thousands of
possible tree shapes and the task quickly gets impossible even
computationally). A nice, relatively simple proof of the neutral site
frequency spectrum is given by <span class="citation"
data-cites="Hudson:15">[@Hudson:15]</span>, but we won’t give this here.
The general form is: <span class="math display">\[\mathbb{E}(S_i) =
\frac{\theta }{i}   \label{eqn:neutral_freq_spec}\]</span> i.e. there
are twice as many singletons as doubletons, three times as many
singletons as tripletons, and so on. The other thing that will be
helpful for us to know is that neutral alleles at intermediate frequency
tend to be old, and those that are rare in the sample are on average
young. We expect to see a lot more rare alleles in our sample than
common alleles.</p>
<p>There are two possible tree shapes that could relate four samples.
Draw both of them and separately colour (or otherwise mark) the branches
by where singletons, doubletons, and tripleton derived alleles could
arise.</p>
<p>We can also ask the probability of observing a derived allele
segregating at frequency <span class="math inline">\(i/n\)</span> given
that the site is polymorphic in our sample of size <span
class="math inline">\(n\)</span> (i.e. given that <span
class="math inline">\(0&lt;i&lt;n\)</span> ). We can obtain this
probability by dividing the expected number of sites segregating for an
allele at frequency <span class="math inline">\(i\)</span> by the
expected number segregating at all of the possible allele frequencies
for polymorphisms in our sample <span
class="math display">\[\begin{aligned}
\mathbb{P}(i |0&lt;i&lt;n) &amp;=\frac{\mathbb{E}(S_i)}{\sum_{j=1}^{n-1}
\mathbb{E}(S_j)} = \frac{\frac{1}{i}}{\sum_{j=1}^{n-1} \frac{1}{j}}.
\end{aligned}\]</span> We can interpret this probability as the fraction
of polymorphic sites we expect to find at a frequency <span
class="math inline">\(i/n\)</span>.</p>
<h5 id="tests-based-on-the-site-frequency-spectrum">Tests based on the
site frequency spectrum</h5>
<p>Population geneticists have proposed a variety of ways to test
whether an observed site frequency spectrum conforms to its neutral,
constant-size expectations. These tests are useful for detecting
population size changes using data across many loci, or for detecting
the signal of selection at individual loci. One of the first tests was
proposed by <span class="citation"
data-cites="tajima:89">[@tajima:89]</span>, and is called Tajima’s <span
class="math inline">\(D\)</span>. Tajima’s <span
class="math inline">\(D\)</span> is <span class="math display">\[D =
\frac{\hat{\theta}_{\pi}-\hat{\theta}_{W}}{C}
\label{eqn_Tajimas_D}\]</span> where the numerator is the difference
between the estimate of <span class="math inline">\(\theta\)</span>
based on pairwise differences and that based on segregating sites. As
these two estimators both have expectation <span
class="math inline">\(\theta\)</span> under the neutral, constant-size
model, the expectation of <span class="math inline">\(D\)</span> is
zero. The denominator <span class="math inline">\(C\)</span> is a
positive constant; it’s the square-root of an estimator of the variance
of this difference under the constant population size, neutral model.
This constant was chosen for <span class="math inline">\(D\)</span> to
have mean zero and variance <span class="math inline">\(1\)</span> under
the null model, so we can test for departures from this simple null
model.<br />
An excess of rare alleles compared to the constant-size, neutral model
will result in a negative Tajima’s <span
class="math inline">\(D\)</span>, because each additional rare allele
increases the number of segregating sites by <span
class="math inline">\(1\)</span>, but only has a small effect on the
number of pairwise differences between samples. In contrast, a positive
Tajima’s <span class="math inline">\(D\)</span> reflects an excess of
intermediate frequency alleles relative to the constant-size, neutral
expectation. Alleles at intermediate-frequency increase pairwise
diversity more per segregating site than typical, thus increasing <span
class="math inline">\(\theta_{\pi}\)</span> more than <span
class="math inline">\(\theta_{W}\)</span>. In the next section we’ll see
how long-term changes in population size systematically change the site
frequency spectrum and so are detectable by statistics such as Tajima’s
<span class="math inline">\(D\)</span>.</p>
<h3 id="demography-and-the-coalescent">Demography and the
coalescent</h3>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="Journal_figs/genetic_drift/human_pop_growth/Nelson_pop_growth.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>We’ve already seen how changes in population size can change the rate
at which heterozygosity is lost from the population (see the discussion
around eqn. <a href="#eqn:var_pop_coal" data-reference-type="eqref"
data-reference="eqn:var_pop_coal">[eqn:var_pop_coal]</a>). If the
population size in generation <span class="math inline">\(i\)</span> is
<span class="math inline">\(N_i\)</span>, the probability that a pair of
lineages coalesce is <span
class="math inline">\(\frac{1}{(2N_i)}\)</span>; this conforms to our
intuition that if the population size is small, the rate at which pairs
of lineages find their common ancestor is faster. We can potentially
accommodate rapid random fluctuations in population size by simply using
the effective population size <span class="math inline">\(N_e\)</span>
in place of <span class="math inline">\(N\)</span>. However,
longer-term, more systematic changes in population size will distort the
coalescent genealogies, and hence patterns of diversity, in more
systematic ways.</p>
<p>We can see how demography potentially distorts the observed frequency
spectrum away from the neutral expectation in a very large sample of
humans shown in Figure <a href="#fig:Human_growth"
data-reference-type="ref"
data-reference="fig:Human_growth">[fig:Human_growth]</a>. For
comparison, the neutral frequency spectrum, eqn <a
href="#eqn:neutral_freq_spec" data-reference-type="eqref"
data-reference="eqn:neutral_freq_spec">[eqn:neutral_freq_spec]</a>, is
shown as a red line. There are vastly more rare alleles than expected
under our neutral, constant-size-size model, but the neutral prediction
and reality agree somewhat more for alleles that are more common.</p>
<div class="figure*">
<div class="center">
<figure>
<embed src="figures/Genetic_drift/Demography/Growth_genealogy.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Why is this? Well, these patterns are likely the result of the very
recent explosive growth in human populations. If the population has
grown rapidly, then the pairwise-coalescent rate in the past may be much
higher than the coalescent rate closer to the present. (see Figure <a
href="#fig:Genealogy_growth" data-reference-type="ref"
data-reference="fig:Genealogy_growth">[fig:Genealogy_growth]</a>).</p>
<p>One consequence of a recent population expansion is that there is
much less genetic diversity in the population than you’d predict using
the census population size. Humans are one example of this effect; there
are <span class="math inline">\(7\)</span> billion of us alive today,
but this is due to very rapid population growth over the past thousand
to tens of thousands of years. Our level of genetic diversity is very
much lower than you’d predict given our census size, reflecting our much
smaller ancestral population. A second consequence of recent population
expansion is that the deeper coalescent branches are much more squished
together in time compared to those in a constant-sized population.
Mutations on deeper branches are the source of alleles at more
intermediate frequencies, and so there are even fewer
intermediate-frequency alleles in growing populations. That’s why there
are so many rare alleles, especially singletons, in this large sample of
Europeans.</p>
<p>Another common demographic scenario is a population bottleneck. In a
bottleneck, the population size crashes dramatically, and subsequently
recovers. For example, our population may have had size <span
class="math inline">\(N_{\textrm{Big}}\)</span> and crashed down to
<span class="math inline">\(N_{\textrm{Small}}\)</span>. One example of
a bottleneck is shown in Figure <a href="#fig:Genealogy_crash"
data-reference-type="ref"
data-reference="fig:Genealogy_crash">[fig:Genealogy_crash]</a>.</p>
<div class="figure*">
<div class="center">
<figure>
<embed src="figures/Genetic_drift/Demography/Crash_genealogy.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Looking at a sample of lineages drawn from the population today, if
the bottleneck was somewhat recent (<span class="math inline">\(\ll
N_{\textrm{Big}}\)</span> generations in the past) many of our lineages
will not have coalesced before reaching the bottleneck, moving backward
in time. But during the bottleneck our lineages coalesce at a much
higher rate, such that many of our lineages will coalesce if the
bottleneck lasts long enough (<span class="math inline">\(\sim
N_{\textrm{Small}}\)</span> generations). If the bottleneck is very
strong, then all of our lineages will coalesce during the bottleneck,
and the resulting site frequency spectrum may look very much like our
population growth model (i.e. an excess of rare alleles). However, if
some pairs of lineages escape coalescing during the bottleneck, they
will coalesce much more deeply in time (e.g. the blue and orange
ancestral lineages in <a href="#fig:Genealogy_crash"
data-reference-type="ref"
data-reference="fig:Genealogy_crash">[fig:Genealogy_crash]</a>).</p>
<div class="center">
<figure>
<embed
src="figures/Genetic_drift/Demography/Mimulus_coalescent_times.pdf"
id="fig:Mimulus_bottleneck" />
<figcaption aria-hidden="true">Diversity along a region of the Mimulus
genome. Black dots give <span class="math inline">\(\pi\)</span> in 1kb
windows between chromosomes sampled from two individuals, the red line
is a moving average (data from <span class="citation"
data-cites="brandvain:14">@brandvain:14</span>). Pairwise coalescent
times (<span class="math inline">\(t\)</span>) estimated assuming <span
class="math inline">\(t= \frac{\pi}{2 \mu}\)</span> using <span
class="math inline">\(\mu_{BP}=10^{-9}\)</span>.</figcaption>
</figure>
</div>
<p>An example of this is shown Figure <a href="#fig:Mimulus_bottleneck"
data-reference-type="ref"
data-reference="fig:Mimulus_bottleneck">4.13</a>, data from <span
class="citation" data-cites="brandvain:14">@brandvain:14</span>.
<em>Mimulus nasutus</em> is a selfing species that arose recently from
an out-crossing progenitor <em>M. guttatus</em>, and experienced a
strong bottleneck. <em>M. guttatus</em> has very high levels of genetic
diversity (<span class="math inline">\(\pi=4\%\)</span> at synonymous
sites), but <em>M. nasutus</em> has lost much of this diversity (<span
class="math inline">\(\pi =1\%\)</span>). Looking along the genome,
between a pair of <em>M. guttatus</em> chromosomes, levels of diversity
are fairly uniformly high.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img src="illustration_images/Genetic_drift/Mimulus/Mimulus.png"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>But in comparing two <em>M. nasutus</em> chromosomes, diversity is
low because the pair of lineages generally coalesce recently. Yet in a
few places we see levels of diversity comparable to <em>M.
guttatus</em>; these regions correspond to genomic sites where our pair
of lineages fail to coalesce during the bottleneck and subsequently
coalesce much more deeply in the ancestral <em>M. guttatus</em>
population.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/genetic_drift/Maize_bottleneck/Wright_Tajima_D.pdf"
id="fig:maize_Tajimas_D" />
<figcaption aria-hidden="true">Data for polymorphism from Maize and
Teosinite: 774 loci from <span class="citation"
data-cites="Wright:05">@Wright:05</span>. <strong>Left)</strong> Genetic
diversity levels in maize and and teosinte samples at each of these
loci. Note how diversity levels are lower in maize than teosinte,
i.e. most points are below the red <span
class="math inline">\(x=y\)</span> line. <strong>Right)</strong> The
distribution of Tajima’s D in maize and teosinte, see how the maize
distribution is shifted towards positive values.</figcaption>
</figure>
</div>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Genetic_drift/teosinte/17960129408_21ad0a16f7_k.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Mutations that arise on deeper lineages will be at intermediate
frequency in our sample, and so mild bottlenecks can lead to an excess
of intermediate frequency alleles compared to the standard constant-size
model. This can skew Tajima’s D (see eqn <a href="#eqn_Tajimas_D"
data-reference-type="ref"
data-reference="eqn_Tajimas_D">[eqn_Tajimas_D]</a>) towards positive
values and away from its expectation of zero. One example of this skew
is shown in Figure <a href="#fig:maize_Tajimas_D"
data-reference-type="ref" data-reference="fig:maize_Tajimas_D">4.14</a>.
Maize (<em>Zea mays</em> subsp. <em>mays</em>) was domesticated from its
wild progenitor teosinte (<em>Zea mays subsp. parviglumis</em>) roughly
ten thousand years ago. We can see how the bottleneck associated with
domestication has resulted in a loss of genetic diversity in maize
compared to teosinte, and the polymorphism that remains is somewhat
skewed towards intermediate frequencies resulting in more positive
values of Tajima’s D.</p>
<p><span class="citation"
data-cites="voight2005interrogating">@voight2005interrogating</span>
sequenced 40 autosomal regions from 15 diploid samples of Hausa people
from Yaounde, Cameroon. The average length of locus they sequenced for
each region was <span class="math inline">\(2365\)</span>bp. They found
that the average number of segregating sites per locus was <span
class="math inline">\(S= 11.1\)</span> and the average <span
class="math inline">\(\pi = 0.0011\)</span> per base over the loci. Is
Tajima’s D positive or negative? Is a demographic model with a
bottleneck or growth more consistent with this result?</p>
<div class="ChapterSummary">
<p>Genetic drift is the random change in allele frequencies due to
alleles by chance leaving more or fewer copies of themselves to the next
generation. It is directionless, with alleles equally likely to go up or
down in frequency thanks to drift. Genetic drift occurs at a slower rate
in larger populations as there is a greater degree of averaging in
larger populations that reduces the impact of the randomness in
individuals’ reproduction.</p>
<p>On average genetic drift acts to remove genetic diversity (e.g.
heterozygosity) from the population. The rate at which neutral genetic
diversity is lost from the population is inversely proportional the
population size.</p>
<p>A balance of mutation and genetic drift can maintain an equilibrium
level of neutral genetic diversity in a population. This equilibrium
level is determined by the population-scaled mutation rate (<span
class="math inline">\(N \mu\)</span>).</p>
<p>In practice, genetic drift will rarely occur at the rate suggested by
the census population size, e.g. due to large variance in reproductive
success and short-term population size fluctuations. In many situations,
we can address this by using an effective population size in place of
the census population size. We can estimate this effective population
size by matching our observed rate of genetic drift to that expected in
an idealized population.</p>
<p>A key insight in thinking about patterns of neutral diversity is to
realize that neutral mutations do not alter the shape of the genetic
tree (or genealogy) relating individuals, and so it is often helpful to
think about the tree first and then think of neutral mutations scattered
on top of this tree.</p>
<p>Coalescent theory describes the properties of these trees, and the
mutational patterns generated, under a model of neutral evolution.</p>
<p>Long-term changes in population size alter the rate of coalescence in
a predictable way that impacts patterns of variation. These patterns can
be used to detect violations of a constant population model and to
estimate more complex demographic models.</p>
</div>
<p>Based on museum samples from <span class="math inline">\(\sim
1800\)</span>, you estimate that the average heterozygosity in Northern
Elephant Seals was <span class="math inline">\(0.0304\)</span> across
many loci. Based on further samples, you estimate that in <span
class="math inline">\(1960\)</span> this had dropped to <span
class="math inline">\(0.011\)</span>. Elephant Seals have a generation
time of <span class="math inline">\(8\)</span> years.<br />
What effective population size do you estimate is consistent with this
drop?</p>
<p><strong>A)</strong> Why are large populations expected to harbor more
neutral variation?<br />
<strong>B)</strong> What is the effective population size? Is it usually
higher or lower than the census population size?<br />
<strong>C)</strong> Why does the effective population size differ across
the autosomes, Y chromosome, and mtDNA?</p>
<p>You sequence a genomic region of a species of Baboon. Out of 100
thousand basepairs, on average, 200 differ between each pair of
sequences. Assume a per base mutation rate of <span
class="math inline">\(1 \times 10^{-8}\)</span> and a generation time of
ten years.<br />
<strong>A)</strong> What is the effective population size of these
Baboons?<br />
<strong>B)</strong> What is the average coalescent time (in years) of a
pair of sequences in this species?</p>
<h1
id="phenotypic-variation-and-the-resemblance-between-relatives.">Phenotypic
Variation and the Resemblance Between Relatives.</h1>
<p>is one of the most useful ideas in biology.<span class="citation"
data-cites="Johannsen:1911">[@Johannsen:1911]</span> The genotype of an
individual (the genome), for most purposes, is decided when the gametes
fuse to form a zygote (individual). The phenotype of an individual
represents any measurable aspect of an organism.</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="illustration_images/Quant_gen/Aspen_budset/Aspen_leaves.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Your height, the amount of RNA transcribed from a given gene, what
you ate last Tuesday: all of these are phenotypes. Nearly any phenotype
we can choose to measure about an organism represents the outcome of the
information encoded by their genome played out through an incredibly
complicated developmental, physiological and/or behavioural processes
that in turn interact with a myriad of environmental and stochastic
factors. Honestly it boggles the mind how organisms work as well as they
do, let alone that I managed to eat lunch last Tuesday.</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Wang_GWAS_poplar/Poplar_Aspen_budset_geno_pheno.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>There are many different ways to think about studying the path from
genotype through to phenotype. The one we will take here is to think
about how phenotypic variation among individuals in a population arises
as a result of genetic variation in the population. One simple way to
measure this genotype-phenotype relationship is to calculate the
phenotypic mean for each genotype at a locus. For example, <span
class="citation" data-cites="wang:18">@wang:18</span> explored the
genetic basis of budset time in European aspen (<em>Populus
tremula</em>); the effect of one specific SNP on that phenotype is shown
in in Figure <a href="#fig:Apsen_geno_pheno" data-reference-type="ref"
data-reference="fig:Apsen_geno_pheno">[fig:Apsen_geno_pheno]</a>. Budset
timing is a key trait underlying local adaptation to varying growing
season length. The associated SNP falls in a gene (<em>PtFT2</em>) that
is known to play a strong role in flowering time regulation in other
plants.</p>
<p>One way for us to assess the relationship between genotype and
phenotype is to find the best fitting linear line through the data,
i.e. fit a linear regression of phenotypes for our individuals on their
genotypes at a particular SNP (<span class="math inline">\(l\)</span>):
<span class="math display">\[X \sim \mu + a_l G_{l}\]</span></p>
<p>In the equation above, <span class="math inline">\(X\)</span> is a
vector of the phenotypes of a set of individuals and <span
class="math inline">\(G_{l}\)</span> is our vector of genotypes at locus
<span class="math inline">\(l\)</span>, with <span
class="math inline">\(G_{i,l}\)</span> taking the value 0, 1, or 2
depending on whether our individual <span
class="math inline">\(i\)</span> is homozygote, heterozygote, or the
alternate homozygote at our locus of interest. Here <span
class="math inline">\(\mu\)</span> is our phenotypic mean. The slope of
this regression line (<span class="math inline">\(a_l\)</span>) has the
interpretation of being the average effect of substituting a copy of
allele <span class="math inline">\(2\)</span> for a copy of allele <span
class="math inline">\(1\)</span>. In our aspen example the slope is
<span class="math inline">\(-13.6\)</span>, i.e. swapping a single <span
class="math inline">\(T\)</span> for a <span
class="math inline">\(G\)</span> allele moves the budset forward by
<span class="math inline">\(13.6\)</span> days, such that the <span
class="math inline">\(GG\)</span> homozygote is predicted to set buds
<span class="math inline">\(27.2\)</span> days earlier than the <span
class="math inline">\(TT\)</span> homozygote.</p>
<p>As a measure of the significance of this genotype-phenotype
relationship, we can calculate the p-value of our regression. To try to
identify loci that are associated with our trait genome-wide, we can
conduct this regression at each SNP we genotype in the genome. One
common way to display the results of such an analysis (called a
genome-wide association study or GWAS for short) is to plot the minus
logarithm of the p-value for each SNP along genome (a so-called
Manhattan plot). Here’s one from <span class="citation"
data-cites="wang:18">@wang:18</span> for their aspen budset
phenotype</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Wang_GWAS_poplar/Wang_Fig_just_Manhattan.pdf"
id="fig:Apsen_Manhattan" />
<figcaption aria-hidden="true">Manhattan plot of the p-value of the
linear association between genotype and budset in aspen. Each dot
represents the test at a single SNP, plotted at its physical coordinate
in the genome. Different chromosomes are plotted in alternating colours.
The SNPs surrounding the PtFT2 gene are shown in red. From <span
class="citation" data-cites="wang:18">@wang:18</span>, .</figcaption>
</figure>
</div>
<p>The SNP with the most significant p-value is SNP in <em>PtFT2</em>.
Note that other SNPs in the surrounding region also light up as showing
a significant association with budset timing. This is because loci that
are in linkage disequilibrium with a functional locus may in turn show
an association, not because they directly affect the phenotype, but
simply because the genotypes at the two loci are themselves non-randomly
associated. Below is a zoomed in version (Figure 2 in <span
class="citation" data-cites="wang:18">@wang:18</span>) with SNPs
coloured by the strength of their LD with the putatively functional
SNP.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Wang_GWAS_poplar/Wang_Fig_zoomed_Manhattan.pdf"
id="fig:Apsen_zoom_Manhattan" />
<figcaption aria-hidden="true">The Manhattan plot zoomed in on the
top-hit (red SNPs from Figure <a href="#fig:Apsen_Manhattan"
data-reference-type="ref" data-reference="fig:Apsen_Manhattan">5.1</a>).
SNPs are now coloured by their <span
class="math inline">\(D^{\prime}\)</span> value with the most
significant SNP. <span class="math inline">\(D^{\prime}\)</span> is the
LD covariance between a pair of loci (<span
class="math inline">\(D\)</span>, eqn <a href="#eqn:LD_def"
data-reference-type="eqref"
data-reference="eqn:LD_def">[eqn:LD_def]</a>) normalized by the largest
value <span class="math inline">\(D\)</span> can take given the allele
frequencies. Figure from <span class="citation"
data-cites="wang:18">@wang:18</span>, .</figcaption>
</figure>
</div>
<p>Note how SNPs in strong LD with the functional allele (redder points)
have more significant p-values.</p>
<p>Variation in some traits seems to have a relatively simple genetic
basis. In our aspen example there is one clear large-effect locus, which
explains 62% of the variation in budset. Note that even in this case,
where we have an allele with a very strong effect on a phenotype, this
is not an allele <em>for</em> budset, nor is <em>PtFT2</em> a gene
<em>for</em> budset. It is an allele that is associated with budset in
the sampled environments and populations. In a different set of
environments, this allele’s effects may be far smaller, and a different
set of alleles may contribute to phenotype variation. <em>PtFT2</em>,
the gene our focal SNP falls close to, is just one of many genes and
molecular pathways involved in budset. A mutant screen for budset may
uncover many genes with larger effects; this gene is just a locus that
happens to be polymorphic in this particular set of genotyped
individuals.</p>
<p>While phenotypic variation for some phenotypes has a relatively
simple genetic basis, many phenotypes are likely much more genetically
complex, involving the functional effect of many alleles at hundreds or
thousands of polymorphic loci. For example hundreds of small effect loci
affecting human height have been mapped in European populations to date.
Such genetically complex traits are called polygenic traits.</p>
<p>In this chapter, we will use our understanding of the sharing of
alleles between relatives to understand the phenotypic resemblance
between relatives in quantitative phenotypes. This will allow us to
understand the contribution of genetic variation to phenotypic
variation. In the next chapter, we will then use these results to
understand the evolutionary change in quantitative phenotypes in
response to selection.<br />
</p>
<h3 id="a-simple-additive-model-of-a-trait">A simple additive model of a
trait</h3>
<p>Let’s imagine that the genetic component of the variation in our
trait is controlled by <span class="math inline">\(L\)</span> autosomal
loci that act in an additive manner. The frequency of allele <span
class="math inline">\(1\)</span> at locus <span
class="math inline">\(l\)</span> is <span
class="math inline">\(p_l\)</span>, with each copy of allele <span
class="math inline">\(1\)</span> at this locus increasing your trait
value by <span class="math inline">\(a_l\)</span> above the population
mean. The phenotype of an individual, let’s call her <span
class="math inline">\(i\)</span>, is <span
class="math inline">\(X_i\)</span>. Her genotype at SNP <span
class="math inline">\(l\)</span> is <span
class="math inline">\(G_{i,l}\)</span>. Here <span
class="math inline">\(G_{i,l}=0,~1,\)</span> or <span
class="math inline">\(2\)</span>, representing the number of copies of
allele <span class="math inline">\(1\)</span> she has at this SNP. Her
expected phenotype, given her genotype at all <span
class="math inline">\(L\)</span> SNPs, is then <span
class="math display">\[\mathbb{E}(X_i | G_{i,1},\cdots,G_{i,L}) =\mu +
X_{A,i} = \mu+\sum_{l=1}^L G_{i,l} a_{l} \label{pheno_geno}\]</span>
where <span class="math inline">\(\mu\)</span> is the mean phenotype in
our population, and <span class="math inline">\(X_{A,i}\)</span> is the
deviation away from the mean phenotype due to her genotype. Now in
reality the phenotype is a function of the expression of those alleles
in a particular environment. Therefore, we can think of this expected
phenotype as being an average across a set of environments that occur in
the population.<br />
When we measure our individual’s observed phenotype we see <span
class="math display">\[X_i =   \mu+X_{A,i} + X_{E,i}
\label{pheno_geno_environ}\]</span> where <span
class="math inline">\(X_E\)</span> is the deviation from the mean
phenotype due to the environment. This <span
class="math inline">\(X_E\)</span> includes the systematic effects of
the environment our individual finds herself in and all of the noise
during development, growth, and the various random insults that life
throws at our individual. If a reasonable number of loci contribute to
variation in our trait then we can approximate the distribution of <span
class="math inline">\(X_{A,i}\)</span> by a normal distribution due to
the central limit theorem (see Figure <a href="#fig:QT1"
data-reference-type="ref" data-reference="fig:QT1">5.3</a>). Thus if we
can approximate the distribution of the effect of environmental
variation on our trait (<span class="math inline">\(X_{E,i}\)</span>)
also by a normal distribution, which is reasonable as there are many
small environmental effects, then the distribution of phenotypes within
the population (<span class="math inline">\(X_i\)</span>) will be
normally distributed (see Figure <a href="#fig:QT1"
data-reference-type="ref" data-reference="fig:QT1">5.3</a>).<br />
</p>
<div class="center">
<figure>
<img src="figures/QT1.png" id="fig:QT1"
alt="The convergence of the phenotypic distribution to a normal distribution. Each of the three histograms shows the distribution of the phenotype in a large sample, for increasingly large numbers of loci (L=1,~4, and 10, with the proportion of variance explained held at V_A=1). I have simulated each individual’s phenotype following equations [pheno_geno] and [pheno_geno_environ]. Specifically, we’ve simulated each individual’s biallelic genotype at L loci, assuming Hardy-Weinberg proportions and that the allele is at 50% frequency. We assume that all of the alleles have equal effects and combine them additively together. We then add an environmental contribution, which is normally distributed with mean zero and variance 0.05. Note that in the left two pictures you can see peaks corresponding to different genotypes due to our low environmental noise (in practice we can rarely see such peaks for real quantitative phenotypes)." />
<figcaption aria-hidden="true">The convergence of the phenotypic
distribution to a normal distribution. Each of the three histograms
shows the distribution of the phenotype in a large sample, for
increasingly large numbers of loci (<span
class="math inline">\(L=1,~4,\)</span> and <span
class="math inline">\(10\)</span>, with the proportion of variance
explained held at <span class="math inline">\(V_A=1\)</span>). I have
simulated each individual’s phenotype following equations <a
href="#pheno_geno" data-reference-type="ref"
data-reference="pheno_geno">[pheno_geno]</a> and <a
href="#pheno_geno_environ" data-reference-type="ref"
data-reference="pheno_geno_environ">[pheno_geno_environ]</a>.
Specifically, we’ve simulated each individual’s biallelic genotype at
<span class="math inline">\(L\)</span> loci, assuming Hardy-Weinberg
proportions and that the allele is at 50% frequency. We assume that all
of the alleles have equal effects and combine them additively together.
We then add an environmental contribution, which is normally distributed
with mean zero and variance <span class="math inline">\(0.05\)</span>.
Note that in the left two pictures you can see peaks corresponding to
different genotypes due to our low environmental noise (in practice we
can rarely see such peaks for real quantitative
phenotypes).</figcaption>
</figure>
</div>
<p>Note that as this is an additive model; we can decompose eqn. <a
href="#pheno_geno_environ" data-reference-type="ref"
data-reference="pheno_geno_environ">[pheno_geno_environ]</a> into the
effects of the two alleles at each locus and rewrite it as <span
class="math display">\[X_i = \mu + X_{iM}+X_{iP} +X_{iE}
\label{eqn:mum_and_pop_var}\]</span> where <span
class="math inline">\(X_{iM}\)</span> and <span
class="math inline">\(X_{iP}\)</span> are the contribution to the
phenotype of the alleles that our individual received from her mother
(maternal alleles) and father (paternal alleles) respectively. This will
come in handy in just a moment when we start thinking about the
phenotypic covariance of relatives.<br />
Now obviously this model seems silly at first sight as alleles don’t
only act in an additive manner, as they interact with alleles at the
same loci (dominance) and at different loci (epistasis). Later we’ll
relax this assumption, however, we’ll find that if we are interested in
evolutionary change over short time-scales it is actually only the
“additive component” of genetic variation that will (usually) concern
us. We will define this more formally later on, but for the moment we
can offer the intuition that parents only get to pass on a single allele
at each locus on to the next generation. As such, it is the effect of
these transmitted alleles, averaged over possible matings, that is an
individual’s average contribution to the next generation (i.e. the
additive effect of the alleles that their genotype consists of).</p>
<h3 id="additive-genetic-variance-and-heritability">Additive genetic
variance and heritability</h3>
<p>As we are talking about an additive genetic model, we’ll talk about
the additive genetic variance (<span
class="math inline">\(V_A\)</span>), the phenotypic variance due to the
additive effects of segregating genetic variation. This is a subset of
the total genetic variance if we allow for non-additive effects.<br />
The variance of our phenotype across individuals (<span
class="math inline">\(V_P\)</span>) we can write as <span
class="math display">\[V_P = Var(X)= Var(X_A) + Var(X_E) =
V_A+V_E\]</span> In writing the phenotypic variance as a sum of the
additive and environmental contributions, we are assuming that there is
no covariance between <span class="math inline">\(X_{G,i}\)</span> and
<span class="math inline">\(X_{E,i}\)</span> i.e. there is no covariance
between genotype and environment.<br />
Our additive genetic variance can be written as <span
class="math display">\[V_A =Var(X_A)  =\sum_{l=1}^L Var(G_{i,l}
a_{l})\]</span> where <span class="math inline">\(Var(G_{i,l}
a_{l})\)</span> is the contribution of locus <span
class="math inline">\(l\)</span> to the additive variance among
individuals. Assuming random mating, and that our loci are in linkage
equilibrium, we can write our additive genetic variance as <span
class="math display">\[V_A = \sum_{l=1}^L a_{l}^2 2
p_l(1-p_l)  \label{eqn:VA}\]</span> where the <span
class="math inline">\(2 p_l(1-p_l)\)</span> term follows from the
binomial sampling of two alleles per individual at each locus.<br />
<strong>Question</strong> You have two biallelic SNPs contributing to
variance in human height. At the first SNP you have an allele with an
additive effect of <span class="math inline">\(5\)</span>cm which is
found at a frequency of <span
class="math inline">\(\frac{1}{10,000}\)</span>. At the second SNP you
have an allele with an additive effect of <span
class="math inline">\(-0.5\)</span>cm segregating at 50% frequency.
Which SNP contributes more to the additive genetic variance? Explain the
intuition of your answer.</p>
Above in eqn<a href="#eqn:mum_and_pop_var" data-reference-type="eqref"
data-reference="eqn:mum_and_pop_var">[eqn:mum_and_pop_var]</a> we
decomposed the additive genetic component of <span
class="math inline">\(X_{A,i}\)</span> as <span
class="math inline">\(X_{M,i} + X_{P,i}\)</span> the additive
contributions of the maternal and paternal derived alleles in the <span
class="math inline">\(i^{th}\)</span> individual. Similarly we can
decompose the additive genetic variance <span
class="math inline">\(V_A\)</span> as $$
<span class="math display">\[\begin{aligned}
  V_A&amp;= Var(X_A) = Var(X_{M,i} ) + Var(X_{P,i} ) \\
       Var(X_{M,i} ) &amp; =Var(X_{P,i} ) =
\frac{V_{A}}{2}  \label{eqn:mat_pat_var}
\end{aligned}\]</span>
<p>$$ assuming that our individuals are mating at random and that
maternal and paternal alleles are equal in their effect in
offspring.</p>
<h5
id="an-example-of-the-additive-basis-of-variation-using-polygenic-scores.">An
example of the additive basis of variation using polygenic scores.</h5>
<p>Now we don’t usually get to see the individual loci contributing to
highly polygenic traits. Instead, we only get to see the distribution of
the trait in the population. However, with the advent of GWAS in human
genetics we can see some of the underlying genetics using the many
trait-associated loci identified to date. Using the estimated effect
sizes at each locus, each one of which is tiny, we can calculate the
weighted sum over an individual’s genotype as in equation <a
href="#pheno_geno" data-reference-type="ref"
data-reference="pheno_geno">[pheno_geno]</a>. This weighted sum is
called the individual’s polygenic score. To illustrate how polygenic
scores work, we can take a set of 1700 SNPs. The effects of these SNPs
are tiny; the median, absolute additive effect size is <span
class="math inline">\(0.07\)</span>cm. Figure <a
href="#fig:Biobank_height_PGS" data-reference-type="ref"
data-reference="fig:Biobank_height_PGS">5.4</a> shows the distribution
of a thousand individuals’ polygenic scores calculated using these 1700
SNPs (simulated genotypes using the UKBB frequencies). The standard
deviation of these polygenic scores <span class="math inline">\(\sim
2\)</span>cm. The individuals with higher polygenic scores for height
are predicted to be taller than the individuals with lower polygenic
scores.</p>
<div class="center">
<figure>
<embed src="figures/Biobank_height_dist.pdf"
id="fig:Biobank_height_PGS" />
<figcaption aria-hidden="true"><strong>Left)</strong> The distribution
of the number of height-increasing alleles that individuals carry at
1700 SNPs associated with height in the UK Biobank, for a sample of 1000
individuals. <strong>right)</strong> The distribution of the polygenic
scores for these 1000 individuals. Plotted on top is a normal
distribution with the same mean and variance. The empirical variance of
these polygenic scores is <span class="math inline">\(0.13\)</span>, the
additive genetic variance calculated by equation <a href="#eqn:VA"
data-reference-type="eqref" data-reference="eqn:VA">[eqn:VA]</a> is
<span class="math inline">\(0.135\)</span>, so the two are in good
agreement.</figcaption>
</figure>
</div>
<h5 id="the-narrow-sense-heritability">The narrow sense
heritability</h5>
<p>We would like a way to think about what proportion of the variation
in our phenotype across individuals is due to genetic differences as
opposed to environmental differences. Such a quantity will be key in
helping us think about the evolution of phenotypes. For example, if
variation in our phenotype had no genetic basis, then no matter how much
selection changes the mean phenotype within a generation the trait will
not change over generations.<br />
We’ll call the proportion of the variance that is genetic the
<em>heritability</em>, and denote it by <span
class="math inline">\(h^2\)</span>. We can then write heritability as
<span class="math display">\[h^2 = \frac{Var(X_A)}{V_P} =
\frac{V_A}{V_P}\]</span> Remember that we are thinking about a trait
where all of the alleles act in a perfectly additive manner. In this
case our heritability <span class="math inline">\(h^2\)</span> is
referred to as the <em>narrow sense heritability</em>, the proportion of
the variance explained by the additive effect of our loci. When we allow
dominance and epistasis into our model, we’ll also have to define the
<em>broad sense heritability</em> (the total proportion of the
phenotypic variance attributable to genetic variation).<br />
The narrow sense heritability of a trait is a useful quantity; indeed
we’ll see shortly that it is exactly what we need to understand the
evolutionary response to selection on a quantitative phenotype. We can
calculate the narrow sense heritability by using the resemblance between
relatives. For example, if the phenotypic differences between
individuals in our population were solely determined by environmental
differences experienced by these different individuals, we should not
expect relatives to resemble each other any more than random individuals
drawn from the population. Now the obvious caveat here is that relatives
also share an environment, so they may resemble each other due to shared
environmental effects.<br />
Note that the heritability is a property of a sample from the population
in a particular set of environments at a particular time. Changes in the
environment may change the phenotypic variance. Changes in the
environment may also change how our genetic alleles are expressed
through development and so change <span
class="math inline">\(V_A\)</span>. Thus estimates of heritability are
not transferable across environments or populations.</p>
<h3 id="the-covariance-between-relatives">The covariance between
relatives</h3>
<p>People have long been fascinated by the resemblance between
relatives, particularly twins (see Figure <a
href="#Fig:The_Cholmondeley_Ladies" data-reference-type="ref"
data-reference="Fig:The_Cholmondeley_Ladies">[Fig:The_Cholmondeley_Ladies]</a>).
Families hold a special place in quantitative genetics, as remarkably we
can use the resemblance between relatives to directly estimate the
heritability and covariance of traits. To see this we can calculate the
covariance in phenotype between pairs of individuals (<span
class="math inline">\(1\)</span> and <span
class="math inline">\(2\)</span>) who have phenotypes <span
class="math inline">\(X_1\)</span> and <span
class="math inline">\(X_2\)</span> respectively. To think about imagine
plotting the phenotypes of, say, sisters against each other. The x and y
coordinates of each point will be the, say, heights of the pair of
siblings. Do tall women tend to have tall sisters, do short women tend
to have short sisters? How much do their phenotypes covary? If some of
the variation in our phenotype is genetic we expect identical twins to
resemble each other more than full siblings, who in turn will resemble
each other more than half-sibs and so on out (see Figure <a
href="#fig:Varying_rellys_phenos" data-reference-type="ref"
data-reference="fig:Varying_rellys_phenos">[fig:Varying_rellys_phenos]</a>).
Under our simple additive model of phenotypes we can write the
covariance as <span class="math display">\[Cov(X_1,X_2) =
Cov\left( X_{1M}+X_{1P}+X_{1E},~ X_{2M}+X_{2P}+X_{2E} \right)\]</span>
We can expand this out in terms of the covariance between the various
components in these sums.<br />
</p>
<div class="figure*">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/Cholmondeley_Ladies/1024px-British_School_17th_century_-_The_Cholmondeley_Ladies_-_Google_Art_Project.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>To make our task easier, we will make two commonly made
assumptions:</p>
<ol type="1">
<li><p>We can ignore the covariance of the environments between
individuals (i.e. <span
class="math inline">\(Cov(X_{1E},X_{2E})=0\)</span>)</p></li>
<li><p>We can ignore the covariance between the environment of one
individual and the genetic variation in another individual (i.e. <span
class="math inline">\(Cov(X_{1E},(X_{2M}+X_{2P}))=0\)</span>).</p></li>
</ol>
<p>The failure of these assumptions to hold can undermine our estimates
of heritability, but we’ll return to that later. Moving forward with
these assumptions, we can simplify our original expression above and
write our phenotypic covariance between our pair of individuals as <span
class="math display">\[Cov(X_1,X_2) =
Cov(X_{1M},X_{2M})+Cov(X_{1M},X_{2P})+Cov(X_{1P},X_{2M})
+Cov(X_{1P},X_{2P}) \label{cov_rels_1}\]</span> This equation is saying
that, under our simple additive model, we can see the covariance in
phenotypes between individuals as the covariance between the maternal
and paternal allelic effects in our individuals. We can use our results
about the sharing of alleles between relatives to obtain these
covariance terms. But before we write down the general case, let’s
quickly work through some examples.<br />
</p>
<div class="figure*">
<div class="center">
<figure>
<embed src="figures/Varying_rellys_phenos.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h5 id="the-covariance-between-identical-twins">The covariance between
identical twins</h5>
<p>Let’s first consider the case of a pair of identical twins,
monzygotic (MZ) twins, from two unrelated parents. Our pair of twins
share their maternal and paternal allele identical by descent (<span
class="math inline">\(X_{1M}=X_{2M}\)</span> and <span
class="math inline">\(X_{1P}=X_{2P}\)</span>). As their maternal and
paternal alleles are not correlated draws from the population, i.e. have
no probability of being <span class="math inline">\(IBD\)</span> as
we’ve said the parents are unrelated, the covariance between their
effects on the phenotype is zero (i.e. <span
class="math inline">\(Cov(X_{1P},X_{2M})=Cov(X_{1M},X_{2P})=0\)</span>).
In that case, eqn. <a href="#cov_rels_1" data-reference-type="ref"
data-reference="cov_rels_1">[cov_rels_1]</a> is <span
class="math display">\[Cov(X_1,X_2) =
Cov(X_{1M},X_{2M})+Cov(X_{1P},X_{2P}) = Var(X_{1M}) +Var(X_{1P})
= V_A\]</span> where the middle step follows from the fact the maternal
(or similarly the paternal) allele in a pair of twins is the same allele
so <span
class="math inline">\(Cov(X_{1M},X_{2M})=Cov(X_{1M},X_{1M})=Var(X_{1M})\)</span>,
as the covariance of random variable with itself is just its variance,
and then the additive variance of the maternal allele contribution is
<span class="math inline">\(Var(X_{1M})=\frac{V_A}{2}\)</span> following
from eqn<a href="#eqn:mat_pat_var" data-reference-type="eqref"
data-reference="eqn:mat_pat_var">[eqn:mat_pat_var]</a>.<br />
To calculate the narrow sense heritability we could then in principal
divide the covariance of our pairs of MZ twins (MZ<span
class="math inline">\(_1\)</span> and MZ<span
class="math inline">\(_2\)</span>) by the trait variance to give <span
class="math display">\[h^2 = \frac{Cov(\text{MZ}_1, \text{MZ}_2) }{V_P}
=
\rho_{\text{MZ}}\]</span> where <span
class="math inline">\(\rho_{\text{MZ}}\)</span> is the correlation of
pairs of MZ twins (see Appendix eqn <a href="#eqn:def_corr"
data-reference-type="eqref"
data-reference="eqn:def_corr">[eqn:def_corr]</a> for more on
correlations). For example, we could estimate the heritability of a
measure of body from the MZ correlation in Figure <a
href="#fig:twins_body_fat" data-reference-type="ref"
data-reference="fig:twins_body_fat">[fig:twins_body_fat]</a>. In
general, this simple estimator isn’t great as the correlation of
identical twins includes the effects of the shared family environment of
the twins (i.e. <span
class="math inline">\(Cov(X_{1E},X_{2E})\)</span>).</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="Journal_figs/Quant_gen/twins_body_fat/twins_body_fat.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Moreover, it can be inflated by non-additive effects as identical
twins don’t just share alleles, they share their entire genotypes, and
thus resemble each other in phenotype also because of shared dominance
effects (we’ll discuss non-additive effects in Section <a
href="#section:nonAddVar" data-reference-type="ref"
data-reference="section:nonAddVar">5.0.4</a>). Better twin-based
estimates of heritability are commonly used based on the comparison of
MZ vs twins that bypass some of these issues.<br />
</p>
<h5 id="the-covariance-in-phenotype-between-parent-and-child">The
covariance in phenotype between parent and child</h5>
<p>Children resemble their biological parents because children inherit
their genome from their parents (putting aside shared environments for
the moment). If a mother and father are unrelated individuals, i.e. they
are two random draws from the population, then this mother and her child
share one allele IBD at each locus (i.e. <span
class="math inline">\(r_1=1\)</span> and <span
class="math inline">\(r_0=r_2=0\)</span>). Let’s assume that our mother
(ind 1) transmits her paternal allele to the child (ind 2), in which
case <span class="math inline">\(X_{P1}=X_{M2}\)</span>, and so <span
class="math inline">\(Cov(X_{P1},X_{M2})=Var(X_{P1})=\tfrac{1}{2}
V_A\)</span>, and all the other covariances in eqn. <a
href="#cov_rels_1" data-reference-type="ref"
data-reference="cov_rels_1">[cov_rels_1]</a> are zero. We’d also arrive
at this result if instead we had thought of the mother transmitting her
own maternal allele. Thus <span class="math inline">\(Cov(X_1,X_2) =
\tfrac{1}{2} V_A\)</span>, we can leverage this form of the covariance
to directly estimate <span class="math inline">\(h^2\)</span> by
regression.<br />
We can estimate the narrow sense heritability through the regression of
child’s phenotype on the parental mid-point phenotype. The parental
mid-point phenotype is simply the average of the mum and dad’s
phenotype. See Figure <a href="#fig:song_sparrow_herit"
data-reference-type="ref"
data-reference="fig:song_sparrow_herit">5.5</a> for an example from song
sparrows.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/song_sparrow_herit/song_sparrow_herit.pdf"
id="fig:song_sparrow_herit" />
<figcaption aria-hidden="true">Parent-midpoint offspring regression for
beak depth and tarsus length in song sparrows. The phenotypes have been
standardized to have mean <span class="math inline">\(0\)</span> and
variance <span class="math inline">\(1\)</span>. The red line shows the
best fitting slope, whose slope is reported on the graph. Note that
<span class="citation"
data-cites="smith1979heritability">@smith1979heritability</span>
regressed the average offspring phenotype for each family on parental
mid-point (<span class="math inline">\(X_{\textrm{avg.kid}}\sim
X_{\textrm{mid}}\)</span>), as they had multiple offspring per family.
However, this doesn’t change the slope of the regression from the form
given by eqn <a href="#eqn:par_off_herit" data-reference-type="eqref"
data-reference="eqn:par_off_herit">[eqn:par_off_herit]</a>. The grey
line is the <span class="math inline">\(x=y\)</span> line. Data from
<span class="citation"
data-cites="smith1979heritability">@smith1979heritability</span>,</figcaption>
</figure>
</div>
<p>We denote the child’s phenotype by <span
class="math inline">\(X_{kid}\)</span> and mid-point phenotype by <span
class="math inline">\(X_{\textrm{mid}}\)</span>, so that if we take the
regression <span class="math inline">\(X_{\textrm{kid}} \sim
X_{\textrm{mid}}\)</span> this regression has slope <span
class="math inline">\(\beta =
Cov(X_{\textrm{kid}},X_{\textrm{mid}})/Var(X_{\textrm{mid}})\)</span>.
The covariance of <span
class="math inline">\(Cov(X_{\textrm{kid}},X_{\textrm{mid}})=\tfrac{1}{2}
V_A\)</span>, and <span class="math inline">\(Var(X_{\textrm{mid}}) =
\tfrac{1}{2}V_P\)</span>, as by taking the average of the parents we
have halved the variance, such that the slope of the regression is <span
class="math display">\[\beta_{\textrm{mid}, \textrm{kid}}=
\frac{Cov(X_{\textrm{kid}},X_{\textrm{mid}})}{Var(X_{\textrm{mid}})}=
\frac{V_A}{V_P} = h^2 \label{eqn:par_off_herit}\]</span></p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/song_sparrow/birdbiographies00ball_0165.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>i.e. the regression of the child’s phenotype on the parental midpoint
phenotype is an estimate of the narrow sense heritability. If much of
the phenotypic variation is due to the (additive) differences in
genotypes among individuals (<span class="math inline">\(h^2 \approx
1\)</span>), then children will closely resemble their parents.
Conversely if much of the variation is environmental (<span
class="math inline">\(h^2 \approx 0\)</span>), and there is no shared
environment between parent and child, children will not resemble their
parents.</p>
<div class="center">
<figure>
<embed src="figures/QT2.pdf" id="fig:midpar" />
<figcaption aria-hidden="true">Regression of child’s phenotype of the
parental mid-point phenotype. The three panels show decreasing levels of
environmental variance (<span class="math inline">\(V_E\)</span>)
holding the additive genetic variance constant (<span
class="math inline">\(V_A=1\)</span>). In these figures, we simulate
<span class="math inline">\(100\)</span> loci, as described in the
caption of Figure <a href="#fig:QT1" data-reference-type="ref"
data-reference="fig:QT1">5.3</a>.We simulate the genotypes and
phenotypes of the two parents, and then simulate the child’s genotype
following mendelian transmission. The red line shows <span
class="math inline">\(x=y\)</span> and the blue line shows the best
fitting linear regression line.</figcaption>
</figure>
</div>
<p>Applying this heritability estimate to the Song sparrow sample we
find <span class="math inline">\(h^2=0.43\)</span> and <span
class="math inline">\(h^2=0.3\)</span> for beak depth and tarsus length
respectively from Figure <a href="#fig:song_sparrow_herit"
data-reference-type="ref"
data-reference="fig:song_sparrow_herit">5.5</a>. So in <span
class="citation"
data-cites="smith1979heritability">@smith1979heritability</span>
analysis, for example, <span class="math inline">\(30\%\)</span> of the
variance in tarsus length is atttributal to the additive effect of
genetic differences among individuals. <span class="citation"
data-cites="smith1979heritability">@smith1979heritability</span> also
regressed the average offspring phenotype against their fathers
<em>o</em>r mothers, giving a slope of <span
class="math inline">\(\beta_{\textrm{dad}, \textrm{avg.kid}}\)</span>
and <span class="math inline">\(\beta_{\textrm{mum},
\textrm{avg.kid}}\)</span>. For tarsus length, for example, they found
<span class="math inline">\(\beta_{\textrm{dad},  \textrm{avg.kid}}=
0.19\)</span> and <span
class="math inline">\(\beta_{\textrm{mum},  \textrm{avg.kid}}=
0.17\)</span>. Following a similar argument to that in eqn <a
href="#eqn:par_off_herit" data-reference-type="eqref"
data-reference="eqn:par_off_herit">[eqn:par_off_herit]</a> we find that
these slopes are <span class="math inline">\(\beta_{\textrm{dad},
\textrm{avg.kid}}= \frac{\frac{V_{A}}{2}}{V_P} = \frac{h^2}{2}\)</span>,
and the same for mums. Thus the regression of offspring’s phenotype on a
particular parent is an estimate of half the narrow-sense heritability,
in line with the reduced slopes found by <span class="citation"
data-cites="smith1979heritability">@smith1979heritability</span>, this
halfing of the slope is due to the fact that a single parent’s phenotype
is a noisier estimate of the parental mid-point and so less informative
about the child s phenotype. These parent specific estimates of
heritability are particularly useful as they allow us to investigate
sex-specific inheritance and sexual dimorphism (we’ll explore this in a
later section).</p>
<p>Estimating heritability by these various parent-offspring regression
have the issue of not controlling for environmental correlations between
parent and offspring, which can inflate our estimates of heritability
(as we will mistake environmentally mediated resemblance for genetics).
Despite its issues, this measure of heritability provides useful
intuition and is directly relevant to our discussion of the response to
selection in the next chapter. That’s because our regression allows us
to attempt to predict the phenotype of the child given the phenotypes of
the parents; how well we can do this depends on the slope. See Figure <a
href="#fig:midpar" data-reference-type="ref"
data-reference="fig:midpar">5.6</a> for examples. If the slope is close
to zero then the parental phenotypes hold little information about the
phenotype of the child, while if the slope is close to one then the
parental mid-point is a good predictor of the child’s phenotype. As we
will see, natural selection will only efficiently drive evolution if
children resemble their parents.<br />
Thinking about our prediction of child’s phenotpye more formally, the
expected phenotype of the child given the parental phenotypes is <span
class="math display">\[\mathbb{E}(X_{kid} | X_{mum},X_{dad}) = \mu +
\beta_{mid,kid}(X_{mid} - \mu) =\mu + h^2(X_{mid} -
\mu)  \label{predict_kid}\]</span> which follows from the definition of
linear regression. So to find the child’s predicted phenotype, we simply
take the mean phenotype and add on the difference between our parental
mid-point and the population mean, multiplied by our narrow sense
heritability.<br />
</p>
<h5 id="the-confounding-effect-of-the-shared-environment.">The
confounding effect of the shared environment.</h5>
<p>Children resemble their biological parents both because they share
alleles but also because they can share environments. For example,
mother birds who are lucky enough to live in a territory with good
nutrition available can in turn provide their offspring better
nutrition, while mothers born in poor territories will be able to
provide less nutrition to their offspring. This covariance in
environment will increase the covariance in traits among relatives, e.g.
weight between mother bird and child, and falsely bias our estimates of
heritability upward if we fail to account for the shared environment.
(Technically what is happening here is that the shared environment
causes <span class="math inline">\(Cov(X_{1E},X_{2E})&gt;0\)</span>
violating the assumption we made in deriving eqn <a href="#cov_rels_1"
data-reference-type="eqref"
data-reference="cov_rels_1">[cov_rels_1]</a>.)</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/song_sparrow_herit/song_sparrow_herit_foster.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Raising the organisms in the lab could remove much of the potential
for shared environment between parent and offspring, but it also removes
much of the environmental variation and we (as evolutionary geneticists)
are usually not primarily interested in knowing the heritability in the
lab bur rather in the field. In some organisms, notably plants, we can
begin to sidestep these issues by raising offspring in a common set of
randomized field conditions (a so called “common garden”). Another
option is cross-foster animals, for example <span class="citation"
data-cites="smith1980experimental">@smith1980experimental</span>
returned to the song sparrow population and swapped eggs between parents
nests. They found that the covariance between biological parents and
children was still high despite these children being raised in a
different nest, but that there was no significant covariance between
foster parents and their non-biological children (see Figure <a
href="#fig:song_sparrow_herit_foster" data-reference-type="ref"
data-reference="fig:song_sparrow_herit_foster">[fig:song_sparrow_herit_foster]</a>
for beak depth). This suggests that family environment is not
confounding the estimate of heritability in this song sparrow
sample.</p>
<div class="figure*">
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/hyena_mat_effects/hyena_mat_effects.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Spotted hyenas (<em>Crocuta crocuta</em>) live in female dominated
groups (‘clans’) that are highly social, with some clans having over a
hundred members. The social rank of a hyena determines many aspects of
their lives. Daughters usually achieve similar rank to their mothers,
i.e. mother and daughter ranks are highly correlated (Left panel, Figure
<a href="#fig:%20hyena_mat_effects" data-reference-type="ref"
data-reference="fig: hyena_mat_effects">[fig: hyena_mat_effects]</a>).
This could suggest that social rank has a high heritability, as could be
the case if rank is determined by body size and body size has a high
additive genetic variance.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img src="illustration_images/Quant_gen/Hyena/Hyena.png" alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Alternatively, social rank could be partially determined by daughters
sharing their social rank with their mothers through gaining access to
their mother’s social connections. To test between these ideas, <span
class="citation" data-cites="east2009maternal">@east2009maternal</span>
made use natural cases of adoption, when a female takes unrelated pups
into her family. <span class="citation"
data-cites="east2009maternal">@east2009maternal</span> found that in
such cases the daughter’s rank was uncorrelated with her biological
mother’s rank (middle panel, Figure <a href="#fig:%20hyena_mat_effects"
data-reference-type="ref" data-reference="fig: hyena_mat_effects">[fig:
hyena_mat_effects]</a>) but strongly correlated with the adopted
mother’s rank (right panel, Figure <a href="#fig:%20hyena_mat_effects"
data-reference-type="ref" data-reference="fig: hyena_mat_effects">[fig:
hyena_mat_effects]</a>). Thus the resemblance between mother and child
for non-adopted hyenas seems to be almost entirely due to the
non-genetic inheritance of social rank through the shared environment of
daughters with their mothers (biological or adopted).</p>
<p>However, such manipulations or common gardens are often impossible in
many systems, and issues of shared environmental covariance due to
maternal resources from egg (or seed) can still present in measures of
heritability from parents and child. Thus researchers often make use of
a variety of different biological relationships to estimate
heritability.</p>
<h5
id="the-covariance-between-general-pairs-of-relatives-under-an-additive-model">The
covariance between general pairs of relatives under an additive
model</h5>
<p>The above examples make clear that to understand the covariance
between phenotypes of relatives, we simply need to think about the
alleles they share IBD. Consider a pair of relatives (<span
class="math inline">\(1\)</span> and <span
class="math inline">\(2\)</span>) with a probability <span
class="math inline">\(r_0\)</span>, <span
class="math inline">\(r_1\)</span>, and <span
class="math inline">\(r_2\)</span> of sharing zero, one, or two alleles
IBD respectively. When they share zero alleles <span
class="math inline">\(Cov((X_{1M}+X_{1P}),(X_{2M}+X_{2P}))=0\)</span>,
when they share one allele <span
class="math inline">\(Cov((X_{1M}+X_{1P}),(X_{2M}+X_{2P}))=
Var(X_{1M})=\frac{1}{2}V_A\)</span>, and when they share two alleles
<span class="math inline">\(Cov((X_{1M}+X_{1P}),(X_{2M}+X_{2P}))=
V_A\)</span>. Therefore, the general covariance between two relatives is
<span class="math display">\[Cov(X_1,X_2) = r_0 \times 0 + r_1
\frac{1}{2}V_A + r_2  V_A =
2 F_{1,2} V_A  \label{additive_covar_general_rellys}\]</span><br />
where <span class="math inline">\(F_{1,2}\)</span> is our coefficient of
kinship, i.e. the probability that two alleles sampled at random from
our pair individuals <span class="math inline">\(1\)</span> and <span
class="math inline">\(2\)</span> are IBD (see eqn <a
href="#eqn:coeffkinship" data-reference-type="eqref"
data-reference="eqn:coeffkinship">[eqn:coeffkinship]</a>). So under a
simple additive model of the genetic basis of a phenotype, to measure
the narrow sense heritability we need to measure the covariance between
pairs of relatives (assuming that we can remove the effect of shared
environmental noise). From the covariance between relatives we can
calculate <span class="math inline">\(V_A\)</span>, and we can then
divide this by the total phenotypic variance to get <span
class="math inline">\(h^2\)</span>.<br />
<strong>Question</strong> <strong>A)</strong> In polygynous red-winged
blackbird populations (i.e. males mate with several females), paternal
half-sibs can be identified. Suppose that the covariance of tarsus
lengths among half-sibs is 0.25 <span
class="math inline">\(cm^2\)</span> and that the total phenotypic
variance is 4 <span class="math inline">\(cm^2\)</span>. Use these data
to estimate <span class="math inline">\(h^2\)</span> for tarsus length
in this population.<br />
<strong>B)</strong> Why might paternal half-sibs be preferable for
measuring heritability than maternal half-sibs?</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/red_winged_blackbirds/Red_wing_blackbirds.png"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h5
id="estimating-additive-genetic-variance-across-a-variety-of-different-relationships-the-animal-model.">Estimating
additive genetic variance across a variety of different relationships
(The animal model).</h5>
<p>In many natural populations we may have access to individuals with a
range of different relationships to each other (e.g. through monitoring
of the paternity of individuals), but relatively few pairs of
individuals for a specific relationship (e.g. sibs). We can try and use
this information on various relatives as fully as possible in a mixed
model framework. Building from equation <a href="#pheno_geno_environ"
data-reference-type="ref"
data-reference="pheno_geno_environ">[pheno_geno_environ]</a>, we can
write an individual’s phenotype <span class="math inline">\(X_i\)</span>
as <span class="math display">\[X_i =  \mu  + X_{A,i} + X_{E,i}\]</span>
where <span class="math inline">\(X_{E,i} \sim N(0,V_E)\)</span> and
<span class="math inline">\(X_{A,i}\)</span> is normally distributed
across individuals with covariance matrix <span
class="math inline">\(V_A A\)</span>, where the the entries for a pair
of individuals i and j are <span class="math inline">\(A_{ij}= 2
F_{i,j}\)</span> and <span class="math inline">\(A_{ii}= 1\)</span>.
Given the matrix <span class="math inline">\(A\)</span> we can estimate
<span class="math inline">\(V_A\)</span>. We can also add fixed effects
into this model to account for generation effects, additional mixed
effects could also be included to account for shared environments
between particular individuals (e.g. a shared nest). This approach is
sometimes called the “animal model”, and is widely used to in modern
quantitative genetics to estimate genetic variances and
heritabilities.</p>
<h3 id="section:nonAddVar">Non-additive variation.</h3>
<p>Up to now we’ve assumed that our alleles contribute to our phenotype
in an additive fashion. However, that does not have to be the case as
there may be non-additivity among the alleles present at a locus
(<em>dominance</em>) or among alleles at different loci
(<em>epistasis</em>). We can accommodate these complications into our
models. We do this by partitioning our total genetic variance into
independent variance components. We’ll see that dominant and
epistatically interacting loci can contribute to the additive genetic
variance. In constructing these variance components models we’ll assume
that we know the alleles contributing to variation in our trait and
their effects, but in reality we rarely know these. However, as we’ll
see we don’t need to know these details and we can partition our
variance and estimate additive variance and other forms of non-additive
variation using the resemblance between various types of relatives.</p>
<h5 id="dominance.">Dominance.</h5>
<p>To understand the effect of dominance, let’s consider how the allele
that a parent transmits influences their offspring’s phenotype. A parent
transmits one of their two alleles at a locus to their offspring.
Assuming that individuals mate at random, this allele is paired with
another allele drawn at random from the population. For example, assume
your mother transmitted an allele 1 to you: with probability <span
class="math inline">\(p\)</span> it would be paired with another allele
1, and you would be a homozygote; and with probability <span
class="math inline">\(q\)</span> it’s paired with a 2 allele and you’re
a heterozygote.</p>
<p>Now consider an autosomal biallelic locus <span
class="math inline">\(\ell\)</span>, with frequency <span
class="math inline">\(p\)</span> for allele 1, and genotypes <span
class="math inline">\(0\)</span>, <span
class="math inline">\(1\)</span>, and <span
class="math inline">\(2\)</span> corresponding to how many copies of
allele <span class="math inline">\(1\)</span> individuals carry. We’ll
denote the mean phenotype of an individual with genotype <span
class="math inline">\(0\)</span>, <span
class="math inline">\(1\)</span>, and <span
class="math inline">\(2\)</span> as <span
class="math inline">\(\overline{X}_{\ell,0}\)</span>, <span
class="math inline">\(\overline{X}_{\ell,1}\)</span>, <span
class="math inline">\(\overline{X}_{\ell,2}\)</span> respectively. This
mean is taking an average phenotype over all the environments and
genetic backgrounds the alleles are present on. We’ll mean center (MC)
these phenotypic values, setting <span
class="math inline">\(\overline{X}&#39;_{\ell,0} = \overline{X}_{\ell,0}
- \mu\)</span>, and likewise for the other genotypes.</p>
<p>We can think about the average (marginal) MC phenotype of an
individual who received an allele 1 from their parent as the average of
the MC phenotype for heterozygotes and 11 homozygotes, weighted by the
probability that the individual has these genotypes, i.e. the
probability they receive an additional allele <span
class="math inline">\(1\)</span> or an allele <span
class="math inline">\(2\)</span> from their other parent: <span
class="math display">\[a_{\ell, 1} = p\overline{X}&#39;_{\ell,2}  +
q\overline{X}&#39;_{\ell,1}, \label{eqn:add_effect1}\]</span> Similarly,
if your parent transmitted an 2 allele to you, your average MC phenotype
would be <span class="math display">\[~~ a_{\ell, 2} =
p\overline{X}&#39;_{\ell,1}  + q\overline{X}&#39;_{\ell,0}
\label{eqn:add_effect2}\]</span></p>
<p>Let’s now consider the average phenotype of an offspring by how many
copies of the allele <span class="math inline">\(1\)</span> they
carry</p>
<div class="center">
<table>
<tbody>
<tr class="odd">
<td style="text-align: center;">genotype:</td>
<td style="text-align: center;">0,</td>
<td style="text-align: center;">1,</td>
<td style="text-align: center;">2.</td>
</tr>
<tr class="even">
<td style="text-align: center;">additive genetic value:</td>
<td style="text-align: center;"><span class="math inline">\(a_{\ell,2}+
a_{\ell,2}\)</span>,</td>
<td style="text-align: center;"><span
class="math inline">\(a_{\ell,1}+a_{\ell,2}\)</span>,</td>
<td style="text-align: center;"><span
class="math inline">\(a_{\ell,1}+a_{\ell,1}\)</span> <span
id="add_values" label="add_values"></span></td>
</tr>
</tbody>
</table>
</div>
<p>i.e. the mean phenotype of each genotypes’ offspring averaged over
all possible matings to other individuals in the population (assuming
individuals mate at random). These are the additive MC genetic values
(breeding values) of our genotypes. Here we are simply adding up the
additive contributions of the alleles present in each genotype and
ignoring any non-additive effects of genotype.</p>
<div class="center">
<figure>
<embed src="figures/additive_effect.pdf" id="fig:add_dom" />
<figcaption aria-hidden="true">The average mean-centered (MC) phenotypes
plotted against the number of allele <span
class="math inline">\(1\)</span> carried (from <span
class="math inline">\(0\)</span> for <span
class="math inline">\(22\)</span> to <span
class="math inline">\(2\)</span> for <span
class="math inline">\(11\)</span>). <strong>Top Row:</strong> Additive
relationship between genotype and phenotype. <strong>Bottom
Row:</strong> Allele 1 is dominant over allele 2, such that the
heterozygote has the same phenotype as the <span
class="math inline">\(11\)</span> genotype. The area of each circle is
proportional to the fraction of the population in each genotypic class
(<span class="math inline">\(p^2\)</span>, <span
class="math inline">\(2pq\)</span>, and <span
class="math inline">\(q^2\)</span>). One the left column <span
class="math inline">\(p=0.1\)</span> and the right column is <span
class="math inline">\(p=0.9\)</span>. The additive genetic values of the
genotypes are shown as red dots. The regression between phenotype and
additive genotype is shown as a red line. The black vertical arrows show
the difference between the average MC phenotype and additive genetic
value for each genotype.</figcaption>
</figure>
</div>
<p>To illustrate this, in Figure <a href="#fig:add_dom"
data-reference-type="ref" data-reference="fig:add_dom">5.7</a> we plot
two different cases of dominance relationships; in the top row an
additive polymorphism and in the second row a fully dominant allele. The
additive genetic values of the genotypes are shown as red dots. Note
that the additive values of the genotypes line up with the observed MC
phenotypic means in the top row, when our alleles interact in a
completely additive manner. Our additive genetic values always fall
along a linear line (the red line in our figure). The additive values
are falling along the best fitting line of linear regression for our
population, when phenotype is regressed against the additive genotype
(<span class="math inline">\(0\)</span>, <span
class="math inline">\(1\)</span>, <span class="math inline">\(2\)</span>
copies of allele 1) across all individuals in our population. Note in
the dominant case the additive genetic values differ from the observed
phenotypic means, and are closer to the observed values for the
genotypes that are most common in the population.<br />
The difference in the additive effect of the two alleles <span
class="math inline">\(a_{\ell, 2}-a_{\ell, 1}\)</span> can be
interpreted as an average effect of swapping an allele 1 for an allele
2; we’ll call this difference <span
class="math inline">\(\alpha_{\ell}=a_{\ell, 2}-a_{\ell, 1}\)</span>.
Our <span class="math inline">\(\alpha_{\ell}\)</span> is also the slope
of the regression of phenotype against genotype (the red line in Figure
<a href="#fig:add_dom" data-reference-type="ref"
data-reference="fig:add_dom">5.7</a>). Note that the slope of our
regression of phenotype on genotype (<span
class="math inline">\(\alpha_{\ell}\)</span>) does not depend on the
population allele frequency for our completely additive locus (top row
of <a href="#fig:add_dom" data-reference-type="ref"
data-reference="fig:add_dom">5.7</a>). In contrast, when there is
dominance, the slope between genotype and phenotype (<span
class="math inline">\(\alpha_{\ell}\)</span>) is a function of allele
frequency (bottom row of <a href="#fig:add_dom"
data-reference-type="ref" data-reference="fig:add_dom">5.7</a>). When a
dominant allele (1) is rare there is a strong slope of phenotype on
genotype, bottom left Figure <a href="#fig:add_dom"
data-reference-type="ref" data-reference="fig:add_dom">5.7</a>. This
strong slope is because replacing a single copy of the 2 allele with a 1
allele in an individual has a big effect on average phenotype, as it
will most likely move an individual from being a 22 homozygote to being
a 12 heterozygote. In contrast, when the dominant allele (1) is common
in the population, replacing a 2 allele by a 1 allele in an individual
on average has little phenotypic effect, leading to a weak slope (bottom
right Figure <a href="#fig:add_dom" data-reference-type="ref"
data-reference="fig:add_dom">5.7</a>). This small effect is because as
we are mainly turning heterozygotes into homozygotes (11), who have the
same mean phenotype as each other.<br />
</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/Salmon/6918368208_5353868a88_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>As an example of how dominance and population allele frequencies can
change the additive effect of an allele, let’s consider the genetics of
the age of sexual maturity in Atlantic Salmon. A single allele of large
effect segregates in Atlantic Salmon that influences the sexual
maturation rate in salmon <span class="citation"
data-cites="ayllon2015vgll3 barson2015sex">[@ayllon2015vgll3;
@barson2015sex]</span>, and hence the timing of their return from the
sea to spawn (sea age). The allele falls close to the autosomal gene
VGLL3 <span class="citation"
data-cites="cousminer2013genome">[@cousminer2013genome variation at this
gene in humans also influences the timing of puberty]</span>. The left
side of Figure <a href="#fig:salmon_add_dom" data-reference-type="ref"
data-reference="fig:salmon_add_dom">5.8</a> shows the age at sexual
maturity in males. The L allele associated with slower sexual maturity
is recessive in males. While the LL homozygotes mature on average a
whole year later, the additive effect of the allele is weak while the L
allele is rare in the population. The right panel shows the effect of
the L allele in females. Note how the allele is much more dominant in
females, and has a much more pronounced additive effect. The dominance
of an allele is not a fixed property of the allele but rather a
statement of the relationship of genotype to phenotype, such that the
dominance relationship between alleles may vary across phenotypes and
contexts (e.g. sexes).</p>
<div class="center">
<figure>
<embed src="Journal_figs/Quant_gen/salmon_age/Salmon_age_dom.pdf"
id="fig:salmon_add_dom" />
<figcaption aria-hidden="true">The average age at sexual maturity for
each genotype, broken down by sex. The area of each circle is
proportional to the fraction of the population in each genotypic class.
The regression between phenotype and additive genotype is shown as a red
line. Data from <span class="citation"
data-cites="barson2015sex">@barson2015sex</span>.</figcaption>
</figure>
</div>
<p>The variance in the population phenotype due to these additive
breeding values at locus <span class="math inline">\(\ell\)</span>,
assuming HW proportions, is <span class="math display">\[\begin{aligned}
V_{A, \ell} &amp;= p^2 (2a_{\ell,2})^2 + 2pq (a_{\ell,1}+a_{\ell,1})^2 +
q^2
(2a_{\ell,0})^2 \nonumber \\
&amp; = 2(p a_{\ell, 1}^2 + q a_{\ell, 2}^2 ) \nonumber \\
&amp; = 2pq \alpha_{\ell}^2 \label{eqn:additive_var_additive_effect}
\end{aligned}\]</span> The total additive variance for the whole
genotype can be found by summing the individual additive genetic
variances over loci <span class="math display">\[V_A = \sum_{\ell=1}^{L}
V_{A, \ell} = \sum_{\ell=1}^{L}
2p_{\ell}q_{\ell} \alpha_{\ell}^2.\]</span></p>
<p>Having assigned the additive genetic variance to be the variance
explained by the additive contribution of the alleles at a locus, we
define the dominance variance as the population variance among genotypes
at a locus due to their deviation from additivity. We can calculate how
much each genotypic mean deviates away from its additive prediction at
locus <span class="math inline">\(\ell\)</span> (the length of the
arrows in Figure <a href="#fig:add_dom" data-reference-type="ref"
data-reference="fig:add_dom">5.7</a>). For example, the heterozygote
deviates <span class="math display">\[d_{\ell,1}
=\overline{X}&#39;_{\ell,1}  - (a_{\ell,1}+ a_{\ell,2})\]</span> away
from its additive genetic value, with similar expressions for each of
the homozygotes (<span class="math inline">\(d_{\ell,0}\)</span> and
<span class="math inline">\(d_{\ell,2}\)</span>). We can then write the
dominance variance at our locus as the genotype-frequency weighted sum
of our squared dominance deviations <span
class="math display">\[V_{D,\ell} = p^2 d_{\ell,0}^2+ 2pq d_{\ell,1}^2+
q^2 d_{\ell,2}^2.\]</span> Writing our total dominance variance as the
sum across loci <span class="math display">\[V_D =
\sum_{\ell=1}^{L}  V_{D,\ell}.\]</span> Having now partitioned all of
the genetic variance into additive and dominant terms, we can write our
total genetic variance as <span class="math display">\[V_{G} =
V_A+V_D.\]</span> We can do this because by construction the covariance
between our additive and dominant deviations for the genotypes is zero.
We can define the narrow sense heritability as before <span
class="math inline">\(h^2=V_A/V_P=V_A/(V_G+V_E)\)</span>, which is the
proportion of phenotypic variance due to additive genetic variance. We
can also define the total proportion of the phenotypic variance due to
genetic differences among individuals, as the broad-sense heritability
<span class="math inline">\(H^2 = V_G/(V_G+V_E)\)</span>.<br />
</p>
<div class="center">
<div id="table:domcovar">
<table>
<caption>Phenotypic covariance between some pairs of relatives, include
the dominance variation. <span
class="math inline">\(^{*}\)</span>Assuming this is the only
relationship the pair of individuals share (above that expected from
randomly sampling individuals from the population).</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Relationship (i,j)<span
class="math inline">\(^{*}\)</span></th>
<th style="text-align: center;"><span
class="math inline">\(Cov(X_i,X_j)\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">parent–child</td>
<td style="text-align: center;"><span class="math inline">\(\frac{1}{2}
V_A\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">full siblings</td>
<td style="text-align: center;"><span class="math inline">\(\frac{1}{2}
V_A +\frac{1}{4} V_D\)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">identical (monzygotic) twins</td>
<td style="text-align: center;"><span
class="math inline">\(V_A+V_D\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;"><span
class="math inline">\(1^{st}\)</span> cousins</td>
<td style="text-align: center;"><span class="math inline">\(\frac{1}{8}
V_A\)</span></td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The additive and dominance variance can be estimated by the
resemblance among relatives. When dominance is present in the loci
influencing our trait (<span class="math inline">\(V_D&gt;0\)</span>),
we need to modify our phenotype covariance among relatives to account
for this non-additivity. Specifically, our equation for the covariance
among a general pair of relatives (eqn. <a
href="#additive_covar_general_rellys" data-reference-type="ref"
data-reference="additive_covar_general_rellys">[additive_covar_general_rellys]</a>
for additive variation) becomes <span
class="math display">\[Cov(X_1,X_2) = 2 F_{1,2} V_A + r_2 V_D
\label{eqn:relly_covar_add_dom}\]</span> where <span
class="math inline">\(r_2\)</span> is the probability that the pair of
individuals share 2 alleles identical by descent, making the same
assumptions (other than additivity) that we made in deriving eqn. <a
href="#additive_covar_general_rellys" data-reference-type="ref"
data-reference="additive_covar_general_rellys">[additive_covar_general_rellys]</a>.
In table <a href="#table:domcovar" data-reference-type="ref"
data-reference="table:domcovar">5.1</a> we show the phenotypic
covariance for some common pairs of relatives. Importantly, in the
presence of dominance variance, the regression of offspring phenotype on
parental midpoint still has a slope <span
class="math inline">\(V_A/V_P\)</span>, as a parent and offspring share
precisely one of their autosomal alleles IBD but never their genotype
IBD (assuming no inbreeding).</p>
<p>Full sibs and parent-offspring have the same covariance if there is
no dominance variance (as they have the same kinship coefficient <span
class="math inline">\(F_{1,2}\)</span>). However, when dominance effects
are present (<span class="math inline">\(V_D&gt;0\)</span>), full-sibs
resemble each other more than parent-offspring pairs. That’s because
full-sibs can share both alleles (i.e. the full genotype at a locus)
identical by descent. We can attempt to estimate <span
class="math inline">\(V_D\)</span> by comparing different sets of
relationships. For example, non-identical twins (full sibs born at same
time) should have <span class="math inline">\(1/2\)</span> the
phenotypic covariance of identical twins if <span
class="math inline">\(V_D=0\)</span>. Therefore, we can attempt to
estimate <span class="math inline">\(V_D\)</span> by looking at whether
identical twins have more than twice the phenotypic covariance than
non-identical twins.<br />
The most important aspect of this discussion for thinking about
evolutionary genetics is that the parent-offspring covariance is still
only a function of <span class="math inline">\(V_A\)</span>. This is
because our parent (e.g. the mother) transmits only a single allele, at
each locus, to its offspring. The other allele the offspring receives is
random (assuming random mating), as it comes from the other unrelated
parent (the father). Therefore, the average effect on the child’s
phenotype of an allele the child receives from their mother is averaged
over all possible random alleles the child could receive from their
father (weighted by their frequency in the population). Thus we only
care about the additive effect of the allele, as parents transmit only
alleles (not genotypes) to their offspring. This means that the
short-term response to selection, as described by the breeder’s
equation, depends only on <span class="math inline">\(V_A\)</span> and
the additive effect of alleles. Therefore, if we can estimate the
narrow-sense heritability we can predict the short-term response.</p>
<p>While our <span class="math inline">\(V_A\)</span> predicts the short
term response to selection, if alleles display dominance, our value of
<span class="math inline">\(V_A\)</span> will change as alleles at our
loci change in frequency. For, example as dominant alleles become common
in the population their contribution to <span
class="math inline">\(V_A\)</span> decreases, we can see this in Figure
<a href="#fig:add_dom" data-reference-type="ref"
data-reference="fig:add_dom">5.7</a> our rare dominant allele (bottom
left) contributes to the additive variance far more than when it is at
high frequency (bottom right). So if selection favours higher values of
our trait, the response to selection will push the dominant allele to
higher frequency decreasing <span class="math inline">\(V_A\)</span>.
Therefore, if there is dominance our value of <span
class="math inline">\(V_A\)</span> will not be constant across
generations.</p>
<p>Up to this point we have only considered dominance and not epistasis.
However, we can include epistasis in a similar manner (for example among
pairs of loci). This gets a little tricky to think about, so we will
only briefly explain it. We can first estimate the additive effect of
the alleles by considering the effect of the alleles averaging over
their possible genetic backgrounds (including the other interacting
alleles they are possibly paired with), just as before. We can then
calculate the additive genetic variance from this. We can estimate the
dominance variance, by calculating the residual variance among genotypes
at a locus unexplained by the additive effect of the loci. We can then
estimate the epistatic variance by estimating the residual variance left
unexplained among the two locus genotypes after accounting for the
additive and dominant deviations calculated from each locus separately.
In practice these high variance components are hard to estimate, and
usually small as much of our variance is assigned to the additive
effect. Again we would find that we mostly care about <span
class="math inline">\(V_A\)</span> for predicting short-term evolution,
but that the contribution of loci to the additive genetic variance will
depend on the epistatic relationships among loci.</p>
<p>How could you use 1/2 sibs vs. full-sibs to estimate <span
class="math inline">\(V_D\)</span>? Why might this be difficult in
practice? Why are identical vs. non-identical twins better suited for
this?</p>
<div class="ChapterSummary">
<p>A key concern of quantitative genetics is how phenotypic variation
within populations is partitioned into environmental and genetic
components of the variance.</p>
<p>The additive genetic variance (<span
class="math inline">\(V_A\)</span>) is the component of phenotypic
variance that can be attributed to the additive action of alleles. The
proportion of phenotypic variance due to <span
class="math inline">\(V_A\)</span> is the narrow-sense heritability
<span class="math inline">\(h^2 = \frac{V_A}{V_P}\)</span>. These
quantities are both measurements of the contribution of the current
standing genetic variation in a particular set of environments and
should not be thought of as fixed quantities of the population or
trait.</p>
<p>We can estimate the additive genetic variance and the heritability by
using the resemblance of relatives, if we can experimentally remove or
statistically partition out the effect of the shared environment among
relatives.</p>
<p>The genetic basis of variation in traits can genetically covary due
to pleiotropy, assortative mating, and linkage. We can estimate the
genetic covariance between traits by using the covariance in different
traits among relatives.</p>
<p>Alleles with dominance and epistatic effects can and do contribute to
<span class="math inline">\(V_A\)</span> to the extent to which
transmitting an additional copy of the allele to an offspring changes
their expected phenotype. These alleles and combinations of alleles also
contribute to higher order genetic variance components, the dominance
and epistatic covariance.</p>
<p>The magnitude of the additive, dominance, and epistatic genetic
variance can change as allele frequency change and recombination changes
the context in which alleles are expressed.</p>
</div>
<p>The additive genetic variance for leg length on mice is <span
class="math inline">\(10mm^2\)</span>. What is the expected covariance
of mice who are first cousins? <strong>Question</strong> Can you
construct a case where <span class="math inline">\(V_A=0\)</span> and
<span class="math inline">\(V_D&gt;0\)</span>? You need just describe it
qualitatively; you don’t need to work out the math. (tricker
question).</p>
<h1 id="the-response-to-phenotypic-selection.">The Response to
Phenotypic Selection.</h1>
<p>Evolution by natural selection requires:</p>
<ol type="1">
<li><p>Variation in a phenotype</p></li>
<li><p>That survival and reproduction is non-random with respect to this
phenotypic variation.</p></li>
<li><p>That this variation is heritable.</p></li>
</ol>
<p>Points 1 and 2 encapsulate our idea of Natural Selection, but
evolution by natural selection will only occur if the 3rd condition is
also met. It is the heritable nature of variation that couples change
within a generation due to natural selection to change across
generations (evolutionary change).<br />
Let’s start by thinking about the change within a generation due to
directional selection, where selection acts to change the mean phenotype
within a generation. For example, a decrease in mean height within a
generation, due to taller organisms having a lower chance of surviving
to reproduction than shorter organisms. Specifically, we’ll denote our
mean phenotype at reproduction by <span
class="math inline">\(\mu_S\)</span>, i.e. after selection has acted,
and our mean phenotype before selection acts by <span
class="math inline">\(\mu_{BS}\)</span>. This second quantity may be
hard to measure, as obviously selection acts throughout the life-cycle,
so it might be easier to think of this as the mean phenotype if
selection hadn’t acted. So the change in mean phenotype within a
generation is <span class="math inline">\(\mu_{S} - \mu_{BS}=
S\)</span>, we’ll call <span class="math inline">\(S\)</span> the
selection differential.<br />
</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/Response_to_sel/QT3.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>We are interested in predicting the distribution of phenotypes in the
next generation. In particular, we are interested in the mean phenotype
in the next generation to understand how directional selection has
contributed to evolutionary change. We’ll denote the mean phenotype in
offspring, i.e. the mean phenotype in the next generation before
selection acts, as <span class="math inline">\(\mu_{NG}\)</span>. The
change across generations we’ll call the response to selection <span
class="math inline">\(R\)</span> and put this equal to <span
class="math inline">\(\mu_{NG}- \mu_{BS}\)</span>.<br />
The mean phenotype in the next generation is <span
class="math display">\[\mu_{NG} = \mathbb{E}\left( \mathbb{E}(X_{kid} |
X_{mum},X_{dad}) \right)\]</span> where the inner expectation is giving
us the expected phenotype of the child given the parents, and the outer
expectation is over possible pairings of parents formed by randomly
mating individuals who survive to reproduce. We can use eqn. <a
href="#predict_kid" data-reference-type="ref"
data-reference="predict_kid">[predict_kid]</a> to obtain an expression
for this expectation: <span class="math display">\[\mu_{NG} = \mu_{BS} +
\beta_{mid,kid} ( \mathbb{E}(X_{mid}) - \mu_{BS})\]</span></p>
<div class="marginfigure">
<div class="center">
<figure>
<embed src="figures/Response_to_sel/Breeders_eqn.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>So to obtain <span class="math inline">\(\mu_{NG}\)</span> we need to
compute <span class="math inline">\(\mathbb{E}(X_{mid})\)</span>, the
expected mid-point phenotype of pairs of individuals who survive to
reproduce. This is just the expected phenotype in the individuals who
survived to reproduce (<span class="math inline">\(\mu_{S}\)</span>), so
<span class="math display">\[\mu_{NG} = \mu_{BS} +
h^2 (\mu_S - \mu_{BS})\]</span> So we can write our response to
selection as <span class="math display">\[R = \mu_{NG} -\mu_{BS}  =
h^2 (\mu_S - \mu_{BS}) = h^2 S \label{breeders_eqn}\]</span> So our
response to selection is proportional to our selection differential, and
the constant of proportionality is the narrow sense heritability. This
equation is sometimes termed the Breeder’s equation. It is a statement
that the evolutionary change across generations (<span
class="math inline">\(R\)</span>) is proportional to the change caused
by directional selection within a generation (<span
class="math inline">\(S\)</span>), and that the strength of this
relationship is determined by the narrow sense heritability (<span
class="math inline">\(h^2\)</span>).<br />
</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Galen_flower_herit/Galen_corolla_flare.pdf"
id="fig:Galen_corolla" />
<figcaption aria-hidden="true">The relationship between maternal and
offspring corolla flare (flower width) in <em>P. viscosum</em>. From
<span class="citation" data-cites="galen:96">@galen:96</span>’s data the
covariance of mother and child is 1.3, while the variance of the mother
is 2.8. Data from <span class="citation"
data-cites="galen:96">@galen:96</span>.</figcaption>
</figure>
</div>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/Polemonium_viscosum_Galen/Polemonium_viscosum.jpg"
style="width:75.0%" alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p><span class="citation" data-cites="galen:96">@galen:96</span>
explored selection on flower shape in <em>Polemonium viscosum</em>. She
found that plants with larger corolla flare had more bumblebee visits,
which resulted in higher seed set and a <span
class="math inline">\(17\%\)</span> increase in corolla flare in the
plants contributing to the next generation. Based on the data in the
caption of Figure <a href="#fig:Galen_corolla" data-reference-type="ref"
data-reference="fig:Galen_corolla">6.1</a> what is the expected response
in the next generation?</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Illinois_long_term_selection_corn/Illinois_LTS_breeders_eq.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>If we know <span class="math inline">\(R\)</span> and <span
class="math inline">\(S\)</span> we can estimate <span
class="math inline">\(h^2\)</span>. Heritabilities estimated like this
are called ‘realized heritability’. Estimates of the ‘realized
heritability’ can readily be produced in artificial selection
experiments: <strong>Question</strong> From the experiment shown in
Figure <a href="#Fig:Illinois_LTS_breeders_eq" data-reference-type="ref"
data-reference="Fig:Illinois_LTS_breeders_eq">[Fig:Illinois_LTS_breeders_eq]</a>,
the mean corn oil content in 1897 was <span
class="math inline">\(4.78\)</span>, among the <span
class="math inline">\(24\)</span> individuals chosen to breed to for the
next generation the mean was <span class="math inline">\(5.2\)</span>.
The offspring of these individuals had a mean kernel oil content of
<span class="math inline">\(5.1\)</span>. What is the narrow sense
realized heritability?</p>
<p>To understand the genetic basis of the response to selection take a
look at Figure <a href="#Fig:Response_num_alleles"
data-reference-type="ref"
data-reference="Fig:Response_num_alleles">6.2</a>. The setup is the same
as in our previous simulation figures.</p>
<div class="center">
<figure>
<embed src="figures/QT3_w_genosums.pdf" id="Fig:Response_num_alleles" />
<figcaption aria-hidden="true"><strong>Top.</strong> Distribution of the
number of up alleles in the parental population prior to selection
(red), for the selected individuals in the top <span
class="math inline">\(10\%\)</span> phenotypic tail of the population
(blue) <strong>Bottom.</strong> The same distribution for the offspring
of the selected parents in the next generation (green).</figcaption>
</figure>
</div>
<p>The individuals who are selected to form our next generation carry
more alleles that increase the phenotype in the current range of
environments currently experienced by the population. The average
individual before selection carried 100 of these ‘up’ alleles, while the
average individual surviving selection carries 108 ‘up’ alleles.</p>
<p>As individuals faithfully transmit their alleles to the next
generation the average child of the selected parents carries <span
class="math inline">\(108\)</span> up alleles. Note that the variance
has changed little, the children have plenty of variation in their
genotype, such that selection can readily drive evolution in future
generations. The average frequency of an ‘up’ allele has changed from
<span class="math inline">\(50\%\)</span> to <span
class="math inline">\(54\%\)</span>. Gains due to selection will be
stably inherited to future generations and can be compounded on
generation after generation if selection pressures were to remain
constant.</p>
<h3 id="the-long-term-response-to-selection">The Long-Term Response to
Selection</h3>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Illinois_long_term_selection_corn/Illinois_LTS_means.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>If our selection pressure is sustained over many generations, we can
use our breeder’s equation to predict the response. If we are willing to
assume that our heritability does not change and we maintain a constant
selection differential (<span class="math inline">\(S\)</span>), then
after <span class="math inline">\(n\)</span> generations our phenotype
mean will have shifted <span class="math display">\[n h^2 S\]</span>
i.e. our population will keep up a linear response to selection.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Illinois_long_term_selection_corn/Illinois_LTS_ggridges_distribution.pdf"
id="Fig:Illinois_LTS_dists" />
<figcaption aria-hidden="true">Density plots showing the phenotypic
distributions of the up- and down-selection populations of the Illinois
long term selection experiment over time. Data available <a
href="https://www.ideals.illinois.edu/handle/2142/3525">here</a>,</figcaption>
</figure>
</div>
<p>Therefore, long-term, consistent selection can drive impressive
evolutionary change. One example of this comes from a field experiment
in Illinois, where plant breeders have systematically selected for
higher and lower oil content in corn (see our previous Figure <a
href="#Fig:Illinois_LTS_breeders_eq" data-reference-type="ref"
data-reference="Fig:Illinois_LTS_breeders_eq">[Fig:Illinois_LTS_breeders_eq]</a>
for one generation of up selection). For over a century, they have
taking seeds from the plants in the extremes of the distribution and
using them to form the next generation. They have achieved impressive
long-term responses, pushing the population distributions well beyond
their initial range (Figure <a href="#Fig:Illinois_LTS_dists"
data-reference-type="ref"
data-reference="Fig:Illinois_LTS_dists">6.3</a>. For example, the oil
up-selection line went from a mean oil content of <span
class="math inline">\(4.7\%\)</span> in 1896 to <span
class="math inline">\(22.1\%\)</span> in 2004. They’ve established two
secondary populations where the selection differential was reversed. In
the up-selection population they have maintained an impressively linear
increase in oil content, shown by red line in Figure <a
href="#Fig:Illinois_LTS_means" data-reference-type="ref"
data-reference="Fig:Illinois_LTS_means">[Fig:Illinois_LTS_means]</a>,
but while the response is linear at first in the down line but they
quickly reach very low oil content (limited by the physical boundary at
0% oil content).</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Genetic_drift/maize/7845339168_66aa3d8ccc_z.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p><strong>Question</strong> <span id="question:reddeer"
label="question:reddeer"></span> A population of red deer were trapped
on Jersey (an island off of England) during the last inter-glacial
period. From the fossil record we can see that the population rapidly
adapted to their new conditions, perhaps due to selection for shorter
reproductive times in the absence of predation <span class="citation"
data-cites="lister:89">[@lister:89]</span>. Within 6,000 years they
evolved from an estimated mean weight of the population of 200kg to an
estimated mean weight of 36kg (a 6 fold reduction)! You estimate that
the generation time of red deer is 5 years and, from a current day
population, that the narrow sense heritability of the phenotype is
0.5.<br />
<strong>A)</strong> Estimate the mean change per generation in the mean
body weight.<br />
<strong>B)</strong> Estimate the change in mean body weight caused by
selection within a generation. State your assumptions.<br />
<strong>C)</strong> Assuming we only have fossils from the founding
population and the population after 6000 years, should we assume that
the calculations accurately reflect what actually occurred within our
population?</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="illustration_images/Quant_gen/dwarf_elephant/M_exilis_skeletal.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Darwins_Finches_unpred/Darwins_Finches_unpred.pdf"
id="fig:Darwins_Finches_unpred" />
<figcaption aria-hidden="true"><strong>Top)</strong> Mean body size of
the medium ground-finch population measured each year. The 1973 <span
class="math inline">\(95\%\)</span> confidence intervals are shown as
horizontal bars. <strong>Bottom)</strong> Standardized selection
differentials on body size. The statistical significance of the
selection differentials is shown, black points are <span
class="math inline">\(p&lt;0.001\)</span> and grey <span
class="math inline">\(p&lt;0.05\)</span>. Data from <span
class="citation"
data-cites="grant2002unpredictable">@grant2002unpredictable</span></figcaption>
</figure>
</div>
<p>In wild populations, selection pressures are likely rarely sustained
for large numbers of generations. For example, the Grants’ have measured
phenotypic selection in Darwin’s Finches over multiple decades on the
island of Daphne Major. They have seen that selection pressures in the
Medium ground-finch (<em>Geospiza fortis</em>) have reversed a number of
times over the years (Figure <a href="#fig:Darwins_Finches_unpred"
data-reference-type="ref"
data-reference="fig:Darwins_Finches_unpred">6.4</a>).</p>
<h5 id="patterns-of-long-term-phenotypic-change-in-the-wild.">Patterns
of long-term phenotypic change in the wild.</h5>
<p>Looking across the diversity of plants and animals we see huge
changes in size and form, can the strengths of selection we can observe
over short time periods possibly explain these changes?</p>
<p>To compare phenotypic changes over various time periods we need some
measure of the rate of phenotypic change. <span class="citation"
data-cites="haldane1949suggestions">@haldane1949suggestions</span>
proposed the rate of change from <span
class="math inline">\(X_1\)</span> to <span
class="math inline">\(X_2\)</span> in time interval <span
class="math inline">\(\Delta t\)</span>, measured in millions of years,
be quantified as <span class="math display">\[\frac{\log
\left(\frac{X_2}{X_1} \right) }{\Delta t}  = \frac{\log
  \left(X_2 \right) -\log \left(X_1 \right)  }{\Delta t}\]</span> by
expressing this the log of the ratio, we are looking at the proportional
fold change, which makes sense as a evolutionary change of 1cm in length
is more impressive if you’re a mouse than an elephant. By putting this
on a <span class="math inline">\(\log\)</span>-scale we are looking at
the fold relative change. <span class="citation"
data-cites="haldane1949suggestions">@haldane1949suggestions</span>
called the units of this measure <em>`</em>the Darwin’, with a one
Darwin change corresponding to a <span class="math inline">\(e\approx
2.71\)</span> fold change in a million years, a two Darwin change
corresponding to a <span class="math inline">\(e^2\approx 7.34\)</span>
fold change in a million years and so on.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img src="illustration_images/Quant_gen/dog_whelks/dog_whelks.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>Calculate the rate of change in body size in the Jersey red deer from
Question <a href="#question:reddeer" data-reference-type="ref"
data-reference="question:reddeer">[question:reddeer]</a> in Darwins. Do
the same for the total change in corn oil content in the up lines in
Figure <a href="#Fig:Illinois_LTS_means" data-reference-type="ref"
data-reference="Fig:Illinois_LTS_means">[Fig:Illinois_LTS_means]</a>.</p>
<p><span class="citation"
data-cites="gingerich1983rates">@gingerich1983rates</span> examined the
absolute rate of phenotypic change in field study data and the fossil
record, a dataset considerably expanded by <span class="citation"
data-cites="uyeda2011million">@uyeda2011million</span>. In Figure <a
href="#fig:uyeda_gingerich" data-reference-type="ref"
data-reference="fig:uyeda_gingerich">6.5</a> each point is an
observation of phenotype evolution. The <em>x</em>-axis shows the time
period in years over which the evolutionary change was observed, the
<em>x</em>-axis is plotted on a <span
class="math inline">\(\log_{10}\)</span> scale. The <em>y</em>-axis
shows absolute rate of phenotypic change, measured in Darwins, again on
a <span class="math inline">\(\log_{10}\)</span> scale.</p>
<p>Over short timescales we see incredibly rapid evolution, note the
high rates on the left of Figure <a href="#fig:uyeda_gingerich"
data-reference-type="ref" data-reference="fig:uyeda_gingerich">6.5</a>.
For example, the first black dot from the left is a case of evolution
over decades in dog whelks. The invasion the green crab (<em>Carcinus
maenas</em>) drove the evolution of more robust shells in Atlantic dog
whelk (<em>Nucella lapillus</em>) in response to predation along the
North American coast <span class="citation"
data-cites="vermeij1982phenotypic">[@vermeij1982phenotypic]</span>. The
shell lip thickness of dog whelks in the St. Andrews, New Brunswick
population had changed from 0.94mm to 1.44mm in just 25 years. That’s a
50% increase, and a rate of 17060 Darwins.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Uyeda_evol_rates/Uyeda_evol_rates.pdf"
id="fig:uyeda_gingerich" />
<figcaption aria-hidden="true">The absolute rate of phenotypic
evolution, measured in Darwins, plotted against the time interval over
which the evolution was observed. The green points show direct
observations of phenotypic change in historical and contemporary
populations. The orange dots give changes observed in the fossil record.
The three black dots left to right give examples from dog whelks, our
red deer example, and <em>Triceratops</em>. Based on an original plot by
<span class="citation"
data-cites="gingerich1983rates">@gingerich1983rates</span> using an
expanded dataset from <span class="citation"
data-cites="uyeda2011million">@uyeda2011million</span>.</figcaption>
</figure>
</div>
<p>However, when we observe phenotypic evolution over longer time
periods it is usually much slower. For example, the rightmost black dot
in Figure <a href="#fig:uyeda_gingerich" data-reference-type="ref"
data-reference="fig:uyeda_gingerich">6.5</a> shows the phenotypic
evolution along the lineage leading to <em>Triceratops</em>.
<em>Triceratops</em> measured in an impressive 25.9–29.5 ft in length.
They evolved from a close relative of <em>Protoceratops</em>, which was
a bit bigger than a sheep at <span
class="math inline">\(\sim\)</span><!-- -->5.9 ft in about 7.5 million
years <span class="citation"
data-cites="colbert1948evolution">[@colbert1948evolution]</span>.
However, that’s only a phenotypic change of <span
class="math inline">\(0.143\)</span> Darwins, its only a roughly four
fold change in millions of years. These rates of change in dinosaurs
have nothing on our dog whelks, or many other examples of evolution on
short time scales. Thus evolutionary changes we can observe over short
timescales readily explain long term changes in quantitative
phenotypes.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/Triceratops/Triceratops_phylo.png"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h2 id="fitness-and-the-breeders-equation.">Fitness and the Breeder’s
Equation.</h2>
<p>So directional evolution occurs as selection drives a change in the
mean phenotype within a generation. But precisely how does this relate
to the natural-selection requirement that organisms vary in their
fitness? Some different ways of formulating the Breeder’s equation give
us insight into the conditions for directional selection and the
relationship to fitness landscapes.</p>
<h3
id="directional-selection-as-the-covariance-between-fitness-and-phenotype.">Directional
selection as the covariance between fitness and phenotype.</h3>
<p>To think more carefully about this change within a generation, let’s
think about a simple fitness model where our phenotype affects the
viability of our organisms (i.e. the probability they survive to
reproduce). The probability that an individual has a phenotype <span
class="math inline">\(X\)</span> before selection is <span
class="math inline">\(p(X=x)\)</span>, so that the mean phenotype before
selection is <span class="math display">\[\mu_{BS} = \mathbb{E}[X]
=  \int_{-\infty}^{\infty} x p(x) dx\]</span> The probability that an
organism with a phenotype <span class="math inline">\(X\)</span>
survives to reproduce is <span class="math inline">\(w(X)\)</span>, and
we’ll think about this as the fitness of our organism. The probability
distribution of phenotypes in those who do survive to reproduce is <span
class="math display">\[\mathbb{P}(X | \textrm{survive}) =  \frac{p(x)
w(x)}{
\int_{-\infty}^{\infty} p(x) w(x) dx}.\]</span> where the denominator is
a normalization constant which ensures that our phenotypic distribution
integrates to one. The denominator also has the interpretation of being
the mean fitness of the population, which we’ll call <span
class="math inline">\(\overline{w}\)</span>, i.e. <span
class="math display">\[\overline{w}=  \int_{-\infty}^{\infty} p(x) w(x)
dx. \label{eqn:pheno_mean_fitness}\]</span> Therefore, we can write the
mean phenotype in those who survive to reproduce as <span
class="math display">\[\mu_S =
\frac{1}{\overline{w}}\int_{-\infty}^{\infty} x p(x) w(x)
dx\]</span></p>
<div class="marginfigure">
<div class="center">
<figure>
<img src="illustration_images/Quant_gen/red_deer/Red_deer.png"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>If we mean center the distribution of phenotypes in our population,
i.e. set the phenotype before selection to zero, then <span
class="math display">\[S=\mu_S=
\frac{1}{\overline{w}}\int_{-\infty}^{\infty} x p(x) w(x) dx =
\frac{1}{\overline{w}}\mathbb{E}\left (X
  w(X) \right)\]</span> where the final part follows from the fact that
the integral is taking the mean of <span class="math inline">\(X
w(X)\)</span> over the population.</p>
<p>As our phenotype is mean centered (<span
class="math inline">\(\mathbb{E}(X)=0\)</span>), we can see that <span
class="math inline">\(S\)</span> has the form of a covariance between
our phenotype <span class="math inline">\(X\)</span> and our relative
fitness <span class="math inline">\(\frac{w(X)}{\overline{w}}\)</span>.
<span class="math display">\[S =  \mathbb{E}\left (X
  \frac{w(X)}{\overline{w}} \right) =Cov \left(X,
\frac{w(X)}{\overline{w}} \right) \label{S_covar}\]</span></p>
<p>Thus our change in mean phenotype is directly a measure of the
covariance of our phenotype and our fitness. Rewriting our breeder’s
equation using this observation we see <span class="math display">\[R =
\frac{V_A}{V_P}  Cov \left(X, \frac{w(X)}{\overline{w}} \right)\]</span>
we see that the response to selection is due to the fact that our
fitness (viability) of our organisms/parents covaries with our
phenotype, and that our child’s phenotype covaries with our parent’s
phenotype.</p>
<h5 id="fitness-gradients-and-linear-regressions">Fitness Gradients and
linear regressions</h5>
<p>To understand this in more detail let imagine that we calculate the
linear regression of an individual <span
class="math inline">\(i\)</span>’s mean-centered phenotype (<span
class="math inline">\(X_i\)</span>) on fitness (<span
class="math inline">\(W_i\)</span>), i.e. <span
class="math display">\[W_i \sim \beta X_i +
\overline{w}\label{fitness_regression}\]</span> the best fitting slope
of this regression (<span class="math inline">\(\beta\)</span>), we’ll
call it the ‘fitness gradient’, is given by <span
class="math display">\[\beta = Cov(X, \frac{w(X)}{\overline{w}} )/
V_P  \label{beta_covar}\]</span> i.e. the fitness gradient is the
phenotype-fitness covariance divided by the phenotypic variance. Using
this result we can rewrite the breeder’s equation as <span
class="math display">\[R= V_A \beta \label{eqn:R_beta}\]</span>
i.e. we’ll see a directional response to selection if there is a linear
relationship of phenotype on fitness, and if there is additive genetic
variance for the phenotype. As one example of a fitness gradient, in
Figure <a href="#fig:red_deer_fitness_grad" data-reference-type="ref"
data-reference="fig:red_deer_fitness_grad">[fig:red_deer_fitness_grad]</a>
the lifetime reproductive success (LRS) of male Red Deer is plotted
against the weight of their antlers. The red line gives the linear
regression of fitness (LRS) on antler mass and the slope of this line is
the fitness gradient (<span class="math inline">\(\beta\)</span>).</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/red_deer_selection_gradient/selection_grad_deer.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h5 id="fishers-fundamental-theorem-of-natural-selection">Fisher’s
fundamental theorem of natural selection</h5>
<p>Finally how does the mean fitness of our population evolve? If we
choose relative fitness to be our phenotype (<span
class="math inline">\(X=\frac{w(X)}{\overline{w}}\)</span>), then the
response in fitness is <span class="math display">\[\begin{aligned}
  R &amp;= \frac{V_A}{V_P}  Cov \left(\frac{w(X)}{\overline{w}} ,
  \frac{w(X)}{\overline{w}} \right) = \frac{V_A}{V_P} V_P \nonumber\\
  &amp;=V_A
\end{aligned}\]</span> i.e. the response to selection is equal to the
additive genetic variance for relative fitness. Or as Fisher put it</p>
<blockquote>
<p>“The rate of increase in fitness of any organism at any time is equal
to its genetic variance in fitness at that time.” -<span
class="citation" data-cites="fisher1930">@fisher1930</span> (pg 37)</p>
</blockquote>
<p>Fisher called this ‘the fundamental theorem of natural selection’.
Our proof here is just a sketch, and more formal approaches are needed
to show it in generality. There has been much gnashing of teeth over
exactly how broadly this result holds, and exactly what Fisher meant
<span class="citation" data-cites="ewens2010gene">[see @ewens2010gene
for a recent overview]</span>.</p>
<h3 id="section:pheno_fitness_landscapes">Directional Selection on
Fitness Landscapes.</h3>
<div class="figure*">
<div class="center">
<figure>
<embed src="figures/Response_to_sel/fitness_landscape_1D.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>One common metaphor when we talk about evolution is that of a
population exploring an adaptive landscape with natural selection
pushing a population towards higher fitness states corresponding to
peaks in this landscape (see e.g. Figure <a
href="#fig:fitness_landscape_1D" data-reference-type="ref"
data-reference="fig:fitness_landscape_1D">[fig:fitness_landscape_1D]</a>).
<span class="citation"
data-cites="lande1976natural">@lande1976natural</span> found an
evocative formulation of the Breeder’s equation which aids our intuition
of phenotypic fitness landscapes. <span class="citation"
data-cites="lande1976natural">@lande1976natural</span> showed that, if
the phenotype is normally distributed, the response to selection (<span
class="math inline">\(R\)</span>) could be written in terms of the
gradient (derivative) of the mean fitness (<span
class="math inline">\(\overline{w}\)</span>) of the population as a
function of the mean phenotype: <span class="math display">\[R =
\frac{V_A}{\overline{w}} \frac{\partial \overline{w}}{\partial
\bar{x}}  \label{eqn:pheno_fitness_landscape}  %V_A
% \frac{\partial \log \left(\wbar \right)}{\partial
\bar{z}}\]</span></p>
<p>What does this mean? Well <span
class="math inline">\(\frac{V_A}{\overline{w}}\)</span> is always
positive, so the direction our population responds to selection is
predicted by the sign of the derivative (see Appendix Section <a
href="#section:calculus" data-reference-type="ref"
data-reference="section:calculus">[section:calculus]</a> for more on
derivatives). If increasing the mean phenotype of the population
slightly would increase mean fitness (<span
class="math inline">\(\frac{\partial \overline{w}}{\partial \bar{x}}
&gt;0\)</span>) our population will respond that generation by evolving
toward higher values of the trait (<span
class="math inline">\(R&gt;0\)</span>), left panel of Figure <a
href="#fig:fitness_landscape_1D_w_wbar" data-reference-type="ref"
data-reference="fig:fitness_landscape_1D_w_wbar">[fig:fitness_landscape_1D_w_wbar]</a>.
Conversely, if decreasing the population mean phenotype slightly would
increase the mean fitness (<span class="math inline">\(\frac{\partial
\overline{w}}{\partial \bar{x}} &lt;0\)</span>) the population will that
generation evolve towards lower values of the phenotype (middle panel of
Figure <a href="#fig:fitness_landscape_1D_w_wbar"
data-reference-type="ref"
data-reference="fig:fitness_landscape_1D_w_wbar">[fig:fitness_landscape_1D_w_wbar]</a>).
Thus, if selection pressures remain constant, we can think of the
population as evolving on an adaptive landscape where the elevation is
given by the population mean fitness. Natural selection operates on the
basis of individual-level fitness, but as a result of this our
population is increasing in its average fitness, i.e. our population is
becoming better adapted. We’ll discuss the caveats of this hill-climbing
interpretation below.</p>
<div class="figure*">
<div class="center">
<figure>
<embed src="figures/Response_to_sel/fitness_landscape_1D_w_wbar.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>What happens when it reaches the top of a peak? Well at the top of a
peak <span class="math inline">\(\frac{\partial  \overline{w}}{\partial
\bar{x}}=0\)</span>, as it is a local maximum, and so <span
class="math inline">\(R=0\)</span>. Assuming that the relationship
between fitness and phenotype stays constant, our population will stay
at the top of the fitness peak. This view of natural selection does not
imply that the population is evolving to the best possible state. Our
population is just marching up the hill of mean fitness (end panel
Figure <a href="#fig:fitness_landscape_1D_w_wbar"
data-reference-type="ref"
data-reference="fig:fitness_landscape_1D_w_wbar">[fig:fitness_landscape_1D_w_wbar]</a>).
However, this peak isn’t necessarily the highest fitness peak but simply
whichever peak was closest. So our population can become trapped on a
local, but not global peak of fitness (see, for example Figure <a
href="#fig:fitness_landscape_1D" data-reference-type="ref"
data-reference="fig:fitness_landscape_1D">[fig:fitness_landscape_1D]</a>).</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Stickleback_fossil_traj/Stickleback_fossil_traj.pdf" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>One dramatic example documenting adaptive evolution to a new fitness
optimum is offered by a remarkable time-series of stickleback evolution
from a fossil lake-bed in Nevada <span class="citation"
data-cites="bell2006inferring">[@bell2006inferring]</span>. In this lake
the layers of sediment are laid down each year allowing a very detailed
time series with over five thousand fossils measured. The time-series
documents the evolution towards a new set of optimum phenotypes in the
fifteen thousand years after the initial invasion of the lake by a
heavily armoured stickleback species. In Figure <a
href="#fig:Stickleback_fossil_traj" data-reference-type="ref"
data-reference="fig:Stickleback_fossil_traj">[fig:Stickleback_fossil_traj]</a>
the population mean number of touching pterygiophores, the bones
supporting the dorsal spines, through the fossil record (Figure <a
href="#fig:Stickleback_fossil" data-reference-type="ref"
data-reference="fig:Stickleback_fossil">[fig:Stickleback_fossil]</a>).
Note how quickly the species evolves toward its new value, presumably a
fitness optimum in their new environment, and the long subsequent time
interval over which the population mean phenotype fluctuates about its
new value.</p>
<p><span class="citation"
data-cites="hunt2008evolution">@hunt2008evolution</span> fitted a model
of a population adapting to a fitness landscape, with a single peak, to
these time-series data. Their fitted fitness surface is shown in the
lower panel of Figure <a href="#fig:Stickleback_fossil_traj"
data-reference-type="ref"
data-reference="fig:Stickleback_fossil_traj">[fig:Stickleback_fossil_traj]</a>
. The arrows show the moves that the population mean phenotype is making
on this inferred fitness surface. The population initially takes large
steps up toward the peak of this surface and subsequently fluctuates
around the peak. Under the interpretation that there is a single
stationary peak these fluctuations represent genetic drift randomly
knocking the population off its optimum, with selection acting to
restore the population towards this local optimum.</p>
<div class="marginfigure">
<div class="center">
<figure>
<embed
src="illustration_images/Quant_gen/Fossil_stickleback/journal.pbio.1001466.g003.eps" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<h5 id="issues-with-the-interpretation-of-fitness-landscapes.">Issues
with the interpretation of fitness landscapes.</h5>
<p>In practice, fitness landscapes may not be constant. The environment
may be constantly changing so our population is constantly forced to
change to keep up with the fitness peak. Indeed our environment may
change so quickly that our population cannot keep up with the peak. Our
population is still trying to increase its mean fitness, to ‘adapt’, but
the landscape itself is evolving. In the case of very rapid
environmental change our population may slide further and further away
the peak, and as a consequence its mean fitness decreases which may
drive the population to extinction if our population drops below <span
class="math inline">\(\overline{w}&lt;1\)</span> for long enough. The
conditions for extinction are an active area of research in the field of
‘Evolutionary rescue’. More generally, for our fitness landscape result
(eqn <a href="#eqn:pheno_fitness_landscape" data-reference-type="eqref"
data-reference="eqn:pheno_fitness_landscape">[eqn:pheno_fitness_landscape]</a>)
to hold, and for us to be able to talk of our population attempting to
evolve to higher mean fitness states, we need the fitness of our
phenotypes to be independent of the frequency of other phenotypes in the
population. (This independence allows us to assume that the fitness of
individuals is not a function of the mean phenotype, as needed in eqn <a
href="#eqn:proof_landscape" data-reference-type="eqref"
data-reference="eqn:proof_landscape">[eqn:proof_landscape]</a>). The
assumption of frequency independence may not hold when there is
competition between individuals, e.g. for resources or mates, as then
the fitness of an individual depends on the strategies pursued by other
individuals in the populations.</p>
<h3 id="stabilizing-and-disruptive-selection">Stabilizing and Disruptive
selection</h3>
<p>Up to now we have just looked at directional selection, where
selection acts to change the mean phenotype. However, we can also use
quantitative genetic models to describe other modes of selection,
extending from effects on the population mean the next natural step is
to think about selection which acts on the population variance.
Selection might act more strongly against individuals in the tails of
the distribution, with those closer to the mean phenotype having higher
fitness, which lowers the variance. Selection could also disfavour
individuals close to the population mean, with individuals with extreme
phenotypes having higher fitness, which acts to increase the variance of
the population.</p>
<p>Directional selection occurs because of the covariance between our
phenotype and fitness, eqn <a href="#S_covar"
data-reference-type="eqref" data-reference="S_covar">[S_covar]</a>. Just
as expressing directional selection as a covariance allowed us to
characterize directional selection as the linear relationship between
fitness and phenotype, <span class="math inline">\(\beta\)</span>, we
can summarize the variance reducing selection by including a quadratic
term in the regression of fitness on phenotype <span
class="math display">\[w_i \sim \beta x_i + \frac{1}{2}  \gamma x_i^2  +
\overline{w}\label{fitness_regression_stab}\]</span> This <span
class="math inline">\(\gamma\)</span>, the coefficient of the quadratic
term in our model, is the quadratic selection gradient: the covariance
of fitness and the squared deviation from the phenotypic mean (<span
class="math inline">\(\mu_{BS}\)</span>), i.e. <span
class="math display">\[\gamma = \frac{Cov\left(w(X), (X-\mu_{BS})^2
\right)}{V^2}\]</span> Our <span class="math inline">\(\gamma\)</span>
describes the curvature of the fitness surface around the mean. Values
of <span class="math inline">\(\gamma&lt;0\)</span> are consistent with
stabilizing selection, reducing the variance. While values of <span
class="math inline">\(\gamma&gt;0\)</span> are consistent with
disruptive selection, increasing the variance.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/birth_weight/Karn_Penrose_birth_weight.pdf"
id="fig:Birth_weight" />
<figcaption aria-hidden="true">Bars show the total number of births with
different birth weights (left axis) Dots show the mortality probability
for different birth-weight bins (right axis), the red line shows a
fitted quadratic model to mortality. Data from <span class="citation"
data-cites="karn1951birth">@karn1951birth</span> Table 2, collapsing
male and female births,</figcaption>
</figure>
</div>
<p>Under stabilizing selection the individuals with extreme phenotypes
in either tail have lower fitness, the result of which is to reduce the
phenotypic variance within a generation. A classic case of stabilizing
selection is birth weight in humans <span class="citation"
data-cites="karn1951birth">[@karn1951birth]</span>. Mary Karn collected
data for nearly fourteen thousand pregnancies from 1935-46 for birth
weight and mortality. These data are replotted in Figure <a
href="#fig:Birth_weight" data-reference-type="ref"
data-reference="fig:Birth_weight">6.6</a>. The variance of all births is
<span class="math inline">\(1.575\)</span>lb<span
class="math inline">\(^2\)</span>, while in live births this was reduced
to <span class="math inline">\(1.26\)</span>lb<span
class="math inline">\(^2\)</span>, a 20% reduction in variance due to
stabilizing selection. It is worth noting that this selection pressure
has been greatly reduced over the decades in societies with access to
good prenatal care <span class="citation"
data-cites="ulizzi1992natural">[@ulizzi1992natural]</span>.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/Pyrenestes_seedcracker/Pyrenestes_minor.jpg"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>In Central Africa, Black-bellied seedcrackers ( <em>Pyrenestes
ostrinus</em>) show disruptive selection on a remarkable beak-size
polymorphism (Figure <a href="#Black_bellied_seedcrackers_beaks"
data-reference-type="ref"
data-reference="Black_bellied_seedcrackers_beaks">6.7</a>). The
small-beaked individuals feed on soft seeds from one species of marsh
sedge while the big-beaked individuals feed on hard seeds from another
sedge, which requires ten times the force to crack. <span
class="citation"
data-cites="smith1993disruptive">@smith1993disruptive</span> recorded
the fates of hundreds of juveniles, and found that individuals with
intermediate beak sizes survived at much lower rates (Figure <a
href="#Black_bellied_seedcrackers_beaks" data-reference-type="ref"
data-reference="Black_bellied_seedcrackers_beaks">6.7</a>) because they
were not well adapted to either seed resource. Break length is subject
to disruptive selection, as can also be seen by the significant positive
quadratic term in the regression of survival probability on break
length. The variance of mandible length in the total sample of
individuals was <span class="math inline">\(0.5\)</span>mm<span
class="math inline">\(^2\)</span> in the survivors this variance
increased by a factor of <span class="math inline">\(2.5\)</span> to
<span class="math inline">\(1.3\)</span>mm<span
class="math inline">\(^2\)</span>.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Smith_black_bellied_seed_cracker/Smith_black_bellied.pdf"
id="Black_bellied_seedcrackers_beaks" />
<figcaption aria-hidden="true"><strong>Left</strong> An illustration of
the the remarkable variation in beak size within Black-bellied
seedcrackers (<em>P. ostrinus</em>). <strong>Right</strong> A histogram
of a beak size measurement in Black-bellied seedcrackers. All juveniles
are shown in grey, while the black bars show the survivors. The red
curve shows the best fitting linear and quadratic model to the
probability of survival, fitted using a binomial generalized linear
model with a logit link function.</figcaption>
</figure>
</div>
<p>To illustrate how directional selection and quadratic terms play off
during adaptation, let’s consider the goldenrod gall fly (<em>Eurosta
solidaginis</em>), aka the goldenrod ball gallmaker. See Figure <a
href="#gall_size_stab" data-reference-type="ref"
data-reference="gall_size_stab">6.8</a>. As it’s wonderful name implies
this insect lays its eggs in goldenrod plants, and the larvae release
chemicals forcing the plant to form a gall that forms a home for the
larvae as they develop. While this seems like a pretty sweet deal for
the larvae, it is not without its perils.</p>
<div class="marginfigure">
<div class="center">
<figure>
<img
src="illustration_images/Quant_gen/goldenrod_ball_gall_maker/goldenrod_ball_gall_maker.png"
alt="image" />
<figcaption aria-hidden="true">image</figcaption>
</figure>
</div>
</div>
<p>When the small, ball galls fall prey to parasitism from parasitoid
wasps. When all the ball galls are small in the population selection
drives strong positive directional selection on gall size, with little
stabilizing selection. Notice in the left panel of Figure <a
href="#gall_size_stab" data-reference-type="ref"
data-reference="gall_size_stab">6.8</a> the good agreement between the
linear selection gradient and the fit including a linear and quadratic
term. However, bigger galls fall under the pall of predation from downy
woodpeckers and black-capped chickadees, who seek out the tasty larvae.
Thus intermediate size galls are favoured, a fitness peak that the
population quickly reaches. Once on this peak, as shown in the right
panel of Figure <a href="#gall_size_stab" data-reference-type="ref"
data-reference="gall_size_stab">6.8</a> there is no directional
selection, i.e. no linear slope, but there is strong stabilizing
selection, i.e. a quadratic term. Thus the population will be maintained
at this fitness peak indefinitely if the environment remains
unchanged.</p>
<div class="center">
<figure>
<embed
src="Journal_figs/Quant_gen/Weis_Gorman_gall_size_stablizing_sel/gall_size.pdf"
id="gall_size_stab" />
<figcaption aria-hidden="true">Fitness surface for gall diameter in
goldenrod ball gallmakers. The dots are the measured survival
probabilities of bins of different sized galls.The solid line is a
fitted individual fitness surface (<span
class="math inline">\(w(~)\)</span>). Dotted line is <span
class="math inline">\(\overline{w}\)</span> plotted as a function of the
population mean assuming a normal distribution with a standard deviation
of <span class="math inline">\(2\)</span>mm. Data from <span
class="citation"
data-cites="weis1990measuring">@weis1990measuring</span>,</figcaption>
</figure>
</div>
<div class="ChapterSummary">
<p>Phenotypic natural selection requires variation in phenotypes impacts
fitness in a non-random way. For evolution by natural selection to occur
we need this phenotypic variation to be heritable.</p>
<p>A simple model for the response to directional selection on a
phenotype is given by the breeder’s equation. The expected response to
selection between adjacent generations (<span
class="math inline">\(R\)</span>) is proportional to the response to
selection within a generation, the selection differential (<span
class="math inline">\(S\)</span>), where the constant of proportionality
is the narrow-sense heritability (<span
class="math inline">\(h^2\)</span>) of the trait. Thus we expect to see,
and indeed do see, strong responses to selection when selection causes
large changes within a generation in heritable phenotypes.</p>
<p>If selection pressures and heritability remain constant we expect a
linear response to selection across many generations. We can often see
this in selection experiments, but in the wild selection pressures often
fluctuate from generation to generation. The large changes in phenotype
we see in the fossil record are easily explained by the strengths of
selection we see acting over short time-scales.</p>
<p>There are two other common ways to write the breeder’s equation. The
first uses the selection gradient (<span
class="math inline">\(\beta\)</span>), the regression of fitness on
phenotype. The second, the fitness landscape interpretation relies on
writing this selection gradient as the derivative of mean fitness by
phenotype. The fitness landscape form to the breeder’s equation helps us
to understand how and when we can expect selection to act to increase
the mean fitness of the population.</p>
<p>We can understand other forms of selection on our phenotype that are
expected to act on the phenotypic variance rather than the mean, such as
disruptive and stabilizing selection, by extensions of the breeder’s
equation to include terms for quadratic selection gradients.</p>
</div>
<p>You are studying the rapid evolution of light organ size in fireflies
(<em>Photinus pyralis</em>) in response to light pollution on a prairie
in Ohio. In January of 1985, a highway was constructed through the
prairie with bright streetlights. Since fireflies use light signals to
locate mates, individuals with smaller, and thus less visible, light
organs were less successful at mating in these new light conditions. You
know the light organ was, on average, 4mm long prior to the construction
of the highway. In 2005, the average light organ size in this population
before mating was 6mm. If this firefly has 1 generation per year and the
narrow sense heritability is 0.1, what was the mean light organ length
of successfully reproducing individuals in 1985 (the first year of
selection)?</p>
<p>You are a rabbit breeder, and you decide that you want rabbits with
long fur. The phenotypic variance is <span
class="math inline">\(4\textrm{cm}^2\)</span>. The covariance of fur
length between between full sibs is <span
class="math inline">\(1\textrm{cm}^2\)</span>. The mean fur length in
the initial population is <span
class="math inline">\(3\textrm{cm}\)</span>. You choose to breed the top
<span class="math inline">\(\frac{1}{3}\)</span> of the population with
the longest fur, and their mean fur length is <span
class="math inline">\(5\)</span>cm.<br />
Assuming that the covariance between sibs is due to only additive
genetic variance, how many generations of this selection regime will it
take for the fur length to be <span class="math inline">\(10cm\)</span>
in the population?</p>
</body>
</html>
